<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Olivier Gimenez">
<meta name="author" content="Maëlis Kervellec">
<meta name="author" content="Jean-Baptiste Fanjul">
<meta name="author" content="Anna Chaine">
<meta name="author" content="Lucile Marescot">
<meta name="author" content="Yoann Bollet">
<meta name="author" content="Christophe Duchamp">
<meta name="dcterms.date" content="2022-04-22">
<meta name="keywords" content="computer-vision, deep-learning, species-distribution-modeling, ecological-statistics">

<title>Trade-off between deep learning for species identification and inference about predator-prey co-occurrence</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="published-202204-gimenez-lynx_files/libs/clipboard/clipboard.min.js"></script>
<script src="published-202204-gimenez-lynx_files/libs/quarto-html/quarto.js"></script>
<script src="published-202204-gimenez-lynx_files/libs/quarto-html/popper.min.js"></script>
<script src="published-202204-gimenez-lynx_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="published-202204-gimenez-lynx_files/libs/quarto-html/anchor.min.js"></script>
<link href="published-202204-gimenez-lynx_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="published-202204-gimenez-lynx_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="published-202204-gimenez-lynx_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="published-202204-gimenez-lynx_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="published-202204-gimenez-lynx_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<style>
    .quarto-title-block .quarto-title-banner {
      color: #FFFFFF;
background: #034E79;
    }
    </style>

<script src="published-202204-gimenez-lynx_files/libs/kePrint/kePrint.js"></script>
<link href="published-202204-gimenez-lynx_files/libs/lightable/lightable.css" rel="stylesheet">

  <script>window.backupDefine = window.define; window.define = undefined;</script><script src="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.js"></script>
  <script>document.addEventListener("DOMContentLoaded", function () {
 var mathElements = document.getElementsByClassName("math");
 var macros = [];
 for (var i = 0; i < mathElements.length; i++) {
  var texText = mathElements[i].firstChild;
  if (mathElements[i].tagName == "SPAN") {
   katex.render(texText.data, mathElements[i], {
    displayMode: mathElements[i].classList.contains('display'),
    throwOnError: false,
    macros: macros,
    fleqn: false
   });
}}});
  </script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.css">

<meta name="citation_title" content="Trade-off between deep learning for species identification and inference about predator-prey co-occurrence">
<meta name="citation_abstract" content="Deep learning is used in computer vision problems with important applications in several scientific fields. In ecology for example, there is a growing interest in deep learning for automatizing repetitive analyses on large amounts of images, such as animal species identification. However, there are challenging issues toward the wide adoption of deep learning by the community of ecologists. First, there is a programming barrier as most algorithms are written in `Python` while most ecologists are versed in `R`. Second, recent applications of deep learning in ecology have focused on computational aspects and simple tasks without addressing the underlying ecological questions or carrying out the statistical data analysis to answer these questions. Here, we showcase a reproducible `R` workflow integrating both deep learning and statistical models using predator-prey relationships as a case study. We illustrate deep learning for the identification of animal species on images collected with camera traps, and quantify spatial co-occurrence using multispecies occupancy models. Despite average model classification performances, ecological inference was similar whether we analysed the ground truth dataset or the classified dataset. This result calls for further work on the trade-offs between time and resources allocated to train models with deep learning and our ability to properly address key ecological questions with biodiversity monitoring. We hope that our reproducible workflow will be useful to ecologists and applied statisticians.
">
<meta name="citation_keywords" content="computer-vision,deep-learning,species-distribution-modeling,ecological-statistics">
<meta name="citation_author" content="Olivier Gimenez">
<meta name="citation_author" content="Maëlis Kervellec">
<meta name="citation_author" content="Jean-Baptiste Fanjul">
<meta name="citation_author" content="Anna Chaine">
<meta name="citation_author" content="Lucile Marescot">
<meta name="citation_author" content="Yoann Bollet">
<meta name="citation_author" content="Christophe Duchamp">
<meta name="citation_publication_date" content="2022-04-22">
<meta name="citation_cover_date" content="2022-04-22">
<meta name="citation_year" content="2022">
<meta name="citation_online_date" content="2022-04-22">
<meta name="citation_fulltext_html_url" content="https://computo.sfds.asso.fr/published-202204-deeplearning-occupancy-lynx">
<meta name="citation_doi" content="10.57750/yfm2-5f45">
<meta name="citation_issn" content="2824-7795">
<meta name="citation_language" content="en">
<meta name="citation_journal_title" content="Computo">
<meta name="citation_publisher" content="Société Française de Statistique">
<meta name="citation_reference" content="citation_title=Uncovering Ecological Patterns with Convolutional Neural Networks;,citation_author=Philip G. Brodrick;,citation_author=Andrew B. Davies;,citation_author=Gregory P. Asner;,citation_publication_date=2019-08;,citation_cover_date=2019-08;,citation_year=2019;,citation_fulltext_html_url=https://linkinghub.elsevier.com/retrieve/pii/S0169534719300862;,citation_issue=8;,citation_doi=10.1016/j.tree.2019.03.006;,citation_issn=01695347;,citation_volume=34;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Deep learning for environmental conservation;,citation_author=Aakash Lamba;,citation_author=Phillip Cassey;,citation_author=Ramesh Raja Segaran;,citation_author=Lian Pin Koh;,citation_publication_date=2019-10;,citation_cover_date=2019-10;,citation_year=2019;,citation_fulltext_html_url=https://linkinghub.elsevier.com/retrieve/pii/S0960982219310322;,citation_issue=19;,citation_doi=10.1016/j.cub.2019.08.016;,citation_issn=09609822;,citation_volume=29;,citation_journal_title=Current Biology;">
<meta name="citation_reference" content="citation_title=Species distribution modeling based on the automated identification of citizen observations;,citation_author=Christophe Botella;,citation_author=Alexis Joly;,citation_author=Pierre Bonnet;,citation_author=Pascal Monestiez;,citation_author=François Munoz;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=2;,citation_volume=6;,citation_journal_title=Applications in Plant Sciences;">
<meta name="citation_reference" content="citation_title=A computer vision for animal ecology;,citation_author=Ben G. Weinstein;,citation_editor=Laura Prugh;,citation_publication_date=2018-05;,citation_cover_date=2018-05;,citation_year=2018;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/1365-2656.12780;,citation_issue=3;,citation_doi=10.1111/1365-2656.12780;,citation_issn=00218790;,citation_volume=87;,citation_journal_title=Journal of Animal Ecology;">
<meta name="citation_reference" content="citation_title=Energy and policy considerations for deep learning in NLP;,citation_author=Emma Strubell;,citation_author=Ananya Ganesh;,citation_author=Andrew McCallum;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=http://arxiv.org/abs/1906.02243;,citation_journal_title=arXiv:1906.02243;">
<meta name="citation_reference" content="citation_title=Recognition in Terra Incognita;,citation_author=Sara Beery;,citation_author=Grant Horn;,citation_author=Pietro Perona;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=http://arxiv.org/abs/1807.04975;,citation_journal_title=arXiv:1807.04975;">
<meta name="citation_reference" content="citation_title=Identifying animal species in camera trap images using deep learning and citizen science;,citation_author=Marco Willi;,citation_author=Ross T. Pitman;,citation_author=Anabelle W. Cardoso;,citation_author=Christina Locke;,citation_author=Alexandra Swanson;,citation_author=Amy Boyer;,citation_author=Marten Veldthuis;,citation_author=Lucy Fortson;,citation_editor=Oscar Gaggiotti;,citation_publication_date=2019-01;,citation_cover_date=2019-01;,citation_year=2019;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/2041-210X.13099;,citation_issue=1;,citation_doi=10.1111/2041-210X.13099;,citation_issn=2041-210X, 2041-210X;,citation_volume=10;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Machine learning to classify animal species in camera trap images: Applications in ecology;,citation_author=Michael A. Tabak;,citation_author=Mohammad S. Norouzzadeh;,citation_author=David W. Wolfson;,citation_author=Steven J. Sweeney;,citation_author=Kurt C. Vercauteren;,citation_author=Nathan P. Snow;,citation_author=Joseph M. Halseth;,citation_author=Paul A. Di Salvo;,citation_author=Jesse S. Lewis;,citation_author=Michael D. White;,citation_author=Ben Teton;,citation_author=James C. Beasley;,citation_author=Peter E. Schlichting;,citation_author=Raoul K. Boughton;,citation_author=Bethany Wight;,citation_author=Eric S. Newkirk;,citation_author=Jacob S. Ivan;,citation_author=Eric A. Odell;,citation_author=Ryan K. Brook;,citation_author=Paul M. Lukacs;,citation_author=Anna K. Moeller;,citation_author=Elizabeth G. Mandeville;,citation_author=Jeff Clune;,citation_author=Ryan S. Miller;,citation_editor=Theoni Photopoulou;,citation_publication_date=2019-04;,citation_cover_date=2019-04;,citation_year=2019;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/2041-210X.13120;,citation_issue=4;,citation_doi=10.1111/2041-210X.13120;,citation_issn=2041-210X, 2041-210X;,citation_volume=10;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Past, present and future approaches using computer vision for animal re‐identification from camera trap data;,citation_author=Stefan Schneider;,citation_author=Graham W. Taylor;,citation_author=Stefan Linquist;,citation_author=Stefan C. Kremer;,citation_editor=Robert B. O’Hara;,citation_publication_date=2019-04;,citation_cover_date=2019-04;,citation_year=2019;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/2041-210X.13133;,citation_issue=4;,citation_doi=10.1111/2041-210X.13133;,citation_issn=2041-210X, 2041-210X;,citation_volume=10;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Applications for deep learning in ecology;,citation_author=Sylvain Christin;,citation_author=Éric Hervet;,citation_author=Nicolas Lecomte;,citation_editor=Hao Ye;,citation_publication_date=2019-10;,citation_cover_date=2019-10;,citation_year=2019;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/2041-210X.13256;,citation_issue=10;,citation_doi=10.1111/2041-210X.13256;,citation_issn=2041-210X, 2041-210X;,citation_volume=10;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Deep learning‐based methods for individual recognition in small birds;,citation_author=André C. Ferreira;,citation_author=Liliana R. Silva;,citation_author=Francesco Renna;,citation_author=Hanja B. Brandl;,citation_author=Julien P. Renoult;,citation_author=Damien R. Farine;,citation_author=Rita Covas;,citation_author=Claire Doutrelant;,citation_editor=Edward Codling;,citation_publication_date=2020-09;,citation_cover_date=2020-09;,citation_year=2020;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/2041-210X.13436;,citation_issue=9;,citation_doi=10.1111/2041-210X.13436;,citation_issn=2041-210X, 2041-210X;,citation_volume=11;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=A deep active learning system for species identification and counting in camera trap images;,citation_author=Mohammad Sadegh Norouzzadeh;,citation_author=Dan Morris;,citation_author=Sara Beery;,citation_author=Neel Joshi;,citation_author=Nebojsa Jojic;,citation_author=Jeff Clune;,citation_editor=Matthew Schofield;,citation_publication_date=2021-01;,citation_cover_date=2021-01;,citation_year=2021;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/2041-210X.13504;,citation_issue=1;,citation_doi=10.1111/2041-210X.13504;,citation_issn=2041-210X, 2041-210X;,citation_volume=12;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Applying deep learning to right whale photo identification;,citation_abstract=Photo identification is an important tool for estimating abundance and monitoring population trends over time. However, manually matching photographs to known individuals is time-consuming. Motivated by recent developments in image recognition, we hosted a data science challenge on the crowdsourcing platform Kaggle to automate the identification of endangered North Atlantic right whales (Eubalaena glacialis). The winning solution automatically identified individual whales with 87% accuracy with a series of convolutional neural networks to identify the region of interest on an image, rotate, crop, and create standardized photographs of uniform size and orientation and then identify the correct individual whale from these passport-like photographs. Recent advances in deep learning coupled with this fully automated workflow have yielded impressive results and have the potential to revolutionize traditional methods for the collection of data on the abundance and distribution of wild populations. Presenting these results to a broad audience should further bridge the gap between the data science and conservation science communities.;,citation_author=Robert Bogucki;,citation_author=Marek Cygan;,citation_author=Christin Brangwynne Khan;,citation_author=Maciej Klimek;,citation_author=Jan Kanty Milczek;,citation_author=Marcin Mucha;,citation_publication_date=2019-06;,citation_cover_date=2019-06;,citation_year=2019;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/cobi.13226;,citation_issue=3;,citation_doi=10.1111/cobi.13226;,citation_issn=0888-8892, 1523-1739;,citation_volume=33;,citation_journal_title=Conservation Biology;">
<meta name="citation_reference" content="citation_title=A study on giant panda recognition based on images of a large proportion of captive pandas;,citation_author=Peng Chen;,citation_author=Pranjal Swarup;,citation_author=Wojciech Michal Matkowski;,citation_author=Adams Wai Kin Kong;,citation_author=Su Han;,citation_author=Zhihe Zhang;,citation_author=Hou Rong;,citation_publication_date=2020-04;,citation_cover_date=2020-04;,citation_year=2020;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1002/ece3.6152;,citation_issue=7;,citation_doi=10.1002/ece3.6152;,citation_issn=2045-7758, 2045-7758;,citation_volume=10;,citation_journal_title=Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Improving the accessibility and transferability of machine learning algorithms for identification of animals in camera trap images: MLWIC2;,citation_author=Michael A. Tabak;,citation_author=Mohammad S. Norouzzadeh;,citation_author=David W. Wolfson;,citation_author=Erica J. Newton;,citation_author=Raoul K. Boughton;,citation_author=Jacob S. Ivan;,citation_author=Eric A. Odell;,citation_author=Eric S. Newkirk;,citation_author=Reesa Y. Conrey;,citation_author=Jennifer Stenglein;,citation_author=Fabiola Iannarilli;,citation_author=John Erb;,citation_author=Ryan K. Brook;,citation_author=Amy J. Davis;,citation_author=Jesse Lewis;,citation_author=Daniel P. Walsh;,citation_author=James C. Beasley;,citation_author=Kurt C. VerCauteren;,citation_author=Jeff Clune;,citation_author=Ryan S. Miller;,citation_publication_date=2020-10;,citation_cover_date=2020-10;,citation_year=2020;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1002/ece3.6692;,citation_issue=19;,citation_doi=10.1002/ece3.6692;,citation_issn=2045-7758, 2045-7758;,citation_volume=10;,citation_journal_title=Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Educating the future generation of researchers: A cross-disciplinary survey of trends in analysis methods;,citation_abstract=Methods for data analysis in the biomedical, life, and social (BLS) sciences are developing at a rapid pace. At the same time, there is increasing concern that education in quantitative methods is failing to adequately prepare students for contemporary research. These trends have led to calls for educational reform to undergraduate and graduate quantitative research method curricula. We argue that such reform should be based on data-driven insights into within- and cross-disciplinary use of analytic methods. Our survey of peer-reviewed literature analyzed approximately 1.3 million openly available research articles to monitor the cross-disciplinary mentions of analytic methods in the past decade. We applied data-driven text mining analyses to the “Methods” and “Results” sections of a large subset of this corpus to identify trends in analytic method mentions shared across disciplines, as well as those unique to each discipline. We found that the t test, analysis of variance (ANOVA), linear regression, chi-squared test, and other classical statistical methods have been and remain the most mentioned analytic methods in biomedical, life science, and social science research articles. However, mentions of these methods have declined as a percentage of the published literature between 2009 and 2020. On the other hand, multivariate statistical and machine learning approaches, such as artificial neural networks (ANNs), have seen a significant increase in the total share of scientific publications. We also found unique groupings of analytic methods associated with each BLS science discipline, such as the use of structural equation modeling (SEM) in psychology, survival models in oncology, and manifold learning in ecology. We discuss the implications of these findings for education in statistics and research methods, as well as within- and cross-disciplinary collaboration.;,citation_author=Taylor Bolt;,citation_author=Jason S. Nomi;,citation_author=Danilo Bzdok;,citation_author=Lucina Q. Uddin;,citation_editor=Ulrich Dirnagl;,citation_publication_date=2021-07;,citation_cover_date=2021-07;,citation_year=2021;,citation_fulltext_html_url=https://dx.plos.org/10.1371/journal.pbio.3001313;,citation_issue=7;,citation_doi=10.1371/journal.pbio.3001313;,citation_issn=1545-7885;,citation_volume=19;,citation_journal_title=PLOS Biology;">
<meta name="citation_reference" content="citation_title=Revisiting animal photo‐identification using deep metric learning and network analysis;,citation_author=Vincent Miele;,citation_author=Gaspard Dussert;,citation_author=Bruno Spataro;,citation_author=Simon Chamaillé‐Jammes;,citation_author=Dominique Allainé;,citation_author=Christophe Bonenfant;,citation_editor=Robert Freckleton;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/2041-210X.13577;,citation_issue=5;,citation_doi=10.1111/2041-210X.13577;,citation_issn=2041-210X, 2041-210X;,citation_volume=12;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=finFindR: Automated recognition and identification of marine mammal dorsal fins using residual convolutional neural networks;,citation_abstract=Photographic identification is an essential research and management tool for marine mammal scientists. However, manual identification of individuals is time-consuming. To shorten processing times, we developed finFindR, an opensource application that uses a series of neural networks to autonomously locate dorsal fins in unedited field images, quantify an individual’s unique fin characteristics, and match them to an existing photograph catalog. During a blind test comparing manual searching to finFindR for common bottlenose dolphin (Tursiops Tursiops truncatus) photographs, experienced photo-identification technicians achieved similar match rates but examined an order of magnitude fewer photographs using finFindR (an average of 10 required with finFindR versus 124 with manual search). In those tests, the correct identity was ranked in the first position in 88% of cases and was within the top 50 ranked positions in 97% of cases. Our observations suggest that finFindR’s matching capabilities are robust to moderate variation in image quality and fin distinctiveness. Importantly, finFindR allows users to build a catalog of known individuals through time and match an unlimited number of individuals instead of being restricted to a predefined set. finFindR’s convolutional neural networks could be re-trained to identify members of many marine mammal species without altering finFindR’s inherent structure.;,citation_author=Jaime W. Thompson;,citation_author=Victoria H. Zero;,citation_author=Lori H. Schwacke;,citation_author=Todd R. Speakman;,citation_author=Brian M. Quigley;,citation_author=Jeanine S. Morey;,citation_author=Trent L. McDonald;,citation_publication_date=2021-07;,citation_cover_date=2021-07;,citation_year=2021;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/mms.12849;,citation_doi=10.1111/mms.12849;,citation_issn=0824-0469, 1748-7692;,citation_journal_title=Marine Mammal Science;">
<meta name="citation_reference" content="citation_title=A Comprehensive Overview of Technologies for Species and Habitat Monitoring and Conservation;,citation_abstract=The range of technologies currently used in biodiversity conservation is staggering, with innovative uses often adopted from other disciplines and being trialed in the field. We provide the first comprehensive overview of the current (2020) landscape of conservation technology, encompassing technologies for monitoring wildlife and habitats, as well as for on-the-ground conservation management (e.g., fighting illegal activities). We cover both established technologies (routinely deployed in conservation, backed by substantial field experience and scientific literature) and novel technologies or technology applications (typically at trial stage, only recently used in conservation), providing examples of conservation applications for both types. We describe technologies that deploy sensors that are fixed or portable, attached to vehicles (terrestrial, aquatic, or airborne) or to animals (biologging), complemented with a section on wildlife tracking. The last two sections cover actuators and computing (including web platforms, algorithms, and artificial intelligence).;,citation_author=José J Lahoz-Monfort;,citation_author=Michael J L Magrath;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://academic.oup.com/bioscience/advance-article/doi/10.1093/biosci/biab073/6322306;,citation_doi=10.1093/biosci/biab073;,citation_issn=0006-3568, 1525-3244;,citation_journal_title=BioScience;">
<meta name="citation_reference" content="citation_title=ALPACA: A fast and accurate approach for automated landmarking of three-dimensional biological structures;,citation_abstract=Landmark-based geometric morphometrics has emerged as an essential discipline for the quantitative analysis of size and shape in ecology and evolution. With the ever-increasing density of digitized landmarks, the possible development of a fully automated method of landmark placement has attracted considerable attention. Despite the recent progress in image registration techniques, which could provide a pathway to automation, three-dimensional morphometric data is still mainly gathered by trained experts. For the most part, the large infrastructure requirements necessary to perform image-based registration, together with its system-speciﬁcity and its overall speed have prevented wide dissemination.;,citation_author=Arthur Porto;,citation_author=Sara M. Rolfe;,citation_author=A. Murat Maga;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=http://biorxiv.org/lookup/doi/10.1101/2020.09.18.303891;,citation_doi=10.1101/2020.09.18.303891;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Images, écologie et deep learning;,citation_author=Vincent Miele;,citation_author=Stéphane Dray;,citation_author=Olivier O. Gimenez;,citation_publication_date=2021-02;,citation_cover_date=2021-02;,citation_year=2021;,citation_fulltext_html_url=https://hal.archives-ouvertes.fr/hal-03142486;,citation_journal_title=Regards sur la biodiversité;,citation_publisher=Société Française d’Écologie et d’Évolution;">
<meta name="citation_reference" content="citation_title=Machine learning for image based species identification;,citation_author=Jana Wäldchen;,citation_author=Patrick Mäder;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=11;,citation_volume=9;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=unmarked: An R package for fitting hierarchical models of wildlife occurrence and abundance;,citation_author=Ian Fiske;,citation_author=Richard Chandler;,citation_publication_date=2011;,citation_cover_date=2011;,citation_year=2011;,citation_issue=10;,citation_volume=43;,citation_journal_title=Journal of Statistical Software;">
<meta name="citation_reference" content="citation_title=A multispecies occupancy model for two or more interacting species;,citation_author=Christopher T. Rota;,citation_author=Marco A. R. Ferreira;,citation_author=Roland W. Kays;,citation_author=Tavis D. Forrester;,citation_author=Elizabeth L. Kalies;,citation_author=William J. McShea;,citation_author=Arielle W. Parsons;,citation_author=Joshua J. Millspaugh;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=10;,citation_volume=7;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Deep learning;,citation_author=Yann LeCun;,citation_author=Yoshua Bengio;,citation_author=Geoffrey Hinton;,citation_publication_date=2015-05;,citation_cover_date=2015-05;,citation_year=2015;,citation_fulltext_html_url=http://www.nature.com/articles/nature14539;,citation_issue=7553;,citation_doi=10.1038/nature14539;,citation_issn=0028-0836, 1476-4687;,citation_volume=521;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=The science of deep learning;,citation_author=Richard Baraniuk;,citation_author=David Donoho;,citation_author=Matan Gavish;,citation_publication_date=2020-12;,citation_cover_date=2020-12;,citation_year=2020;,citation_fulltext_html_url=http://www.pnas.org/lookup/doi/10.1073/pnas.2020596117;,citation_issue=48;,citation_doi=10.1073/pnas.2020596117;,citation_issn=0027-8424, 1091-6490;,citation_volume=117;,citation_journal_title=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Deep Learning for Computer Vision: A Brief Review;,citation_abstract=Over the last years deep learning methods have been shown to outperform previous state-of-the-art machine learning techniques in several fields, with computer vision being one of the most prominent cases. This review paper provides a brief overview of some of the most significant deep learning schemes used in computer vision problems, that is, Convolutional Neural Networks, Deep Boltzmann Machines and Deep Belief Networks, and Stacked Denoising Autoencoders. A brief account of their history, structure, advantages, and limitations is given, followed by a description of their applications in various computer vision tasks, such as object detection, face recognition, action and activity recognition, and human pose estimation. Finally, a brief overview is given of future directions in designing deep learning schemes for computer vision problems and the challenges involved therein.;,citation_author=Athanasios Voulodimos;,citation_author=Nikolaos Doulamis;,citation_author=Anastasios Doulamis;,citation_author=Eftychios Protopapadakis;,citation_editor=Diego Andina;,citation_publication_date=2018-02;,citation_cover_date=2018-02;,citation_year=2018;,citation_fulltext_html_url=https://doi.org/10.1155/2018/7068349;,citation_doi=10.1155/2018/7068349;,citation_issn=1687-5265;,citation_volume=2018;,citation_journal_title=Computational Intelligence and Neuroscience;">
<meta name="citation_reference" content="citation_title=ImageNet classification with deep convolutional neural networks;,citation_author=Alex Krizhevsky;,citation_author=Ilya Sutskever;,citation_author=Geoffrey E. Hinton;,citation_editor=F. Pereira;,citation_editor=C. J. C. Burges;,citation_editor=L. Bottou;,citation_editor=K. Q. Weinberger;,citation_publication_date=2012;,citation_cover_date=2012;,citation_year=2012;,citation_inbook_title=Advances in neural information processing systems 25;">
<meta name="citation_reference" content="citation_title=Statistical ecology comes of age;,citation_abstract=The desire to predict the consequences of global environmental change has been the driver towards more realistic models embracing the variability and uncertainties inherent in ecology. Statistical ecology has gelled over the past decade as a discipline that moves away from describing patterns towards modelling the ecological processes that generate these patterns. Following the fourth International Statistical Ecology Conference (1–4 July 2014) in Montpellier, France, we analyse current trends in statistical ecology. Important advances in the analysis of individual movement, and in the modelling of population dynamics and species distributions, are made possible by the increasing use of hierarchical and hidden process models. Exciting research perspectives include the development of methods to interpret citizen science data and of efficient, flexible computational algorithms for model fitting. Statistical ecology has come of age: it now provides a general and mathematically rigorous framework linking ecological theory and empirical data.;,citation_author=Olivier Gimenez;,citation_author=Stephen T. Buckland;,citation_author=Byron J. T. Morgan;,citation_author=Nicolas Bez;,citation_author=Sophie Bertrand;,citation_author=Rémi Choquet;,citation_author=Stéphane Dray;,citation_author=Marie-Pierre Etienne;,citation_author=Rachel Fewster;,citation_author=Frédéric Gosselin;,citation_author=Bastien Mérigot;,citation_author=Pascal Monestiez;,citation_author=Juan M. Morales;,citation_author=Frédéric Mortier;,citation_author=François Munoz;,citation_author=Otso Ovaskainen;,citation_author=Sandrine Pavoine;,citation_author=Roger Pradel;,citation_author=Frank M. Schurr;,citation_author=Len Thomas;,citation_author=Wilfried Thuiller;,citation_author=Verena Trenkel;,citation_author=Perry Valpine;,citation_author=Eric Rexstad;,citation_publication_date=2014-12;,citation_cover_date=2014-12;,citation_year=2014;,citation_fulltext_html_url=https://royalsocietypublishing.org/doi/10.1098/rsbl.2014.0698;,citation_issue=12;,citation_doi=10.1098/rsbl.2014.0698;,citation_issn=1744-9561, 1744-957X;,citation_volume=10;,citation_journal_title=Biology Letters;">
<meta name="citation_reference" content="citation_title=Identification of 100 fundamental ecological questions;,citation_author=William J. Sutherland;,citation_author=Robert P. Freckleton;,citation_author=H. Charles J. Godfray;,citation_author=Steven R. Beissinger;,citation_author=Tim Benton;,citation_author=Duncan D. Cameron;,citation_author=Yohay Carmel;,citation_author=David A. Coomes;,citation_author=Tim Coulson;,citation_author=Mark C. Emmerson;,citation_author=Rosemary S. Hails;,citation_author=Graeme C. Hays;,citation_author=Dave J. Hodgson;,citation_author=Michael J. Hutchings;,citation_author=David Johnson;,citation_author=Julia P. G. Jones;,citation_author=Matt J. Keeling;,citation_author=Hanna Kokko;,citation_author=William E. Kunin;,citation_author=Xavier Lambin;,citation_author=Owen T. Lewis;,citation_author=Yadvinder Malhi;,citation_author=Nova Mieszkowska;,citation_author=E. J. Milner-Gulland;,citation_author=Ken Norris;,citation_author=Albert B. Phillimore;,citation_author=Drew W. Purves;,citation_author=Jane M. Reid;,citation_author=Daniel C. Reuman;,citation_author=Ken Thompson;,citation_author=Justin M. J. Travis;,citation_author=Lindsay A. Turnbull;,citation_author=David A. Wardle;,citation_author=Thorsten Wiegand;,citation_editor=David Gibson;,citation_publication_date=2013-01;,citation_cover_date=2013-01;,citation_year=2013;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1111/1365-2745.12025;,citation_issue=1;,citation_doi=10.1111/1365-2745.12025;,citation_issn=00220477;,citation_volume=101;,citation_journal_title=Journal of Ecology;">
<meta name="citation_reference" content="citation_title=Responsible AI for conservation;,citation_author=Oliver R. Wearn;,citation_author=Robin Freeman;,citation_author=David M. P. Jacoby;,citation_publication_date=2019-02;,citation_cover_date=2019-02;,citation_year=2019;,citation_fulltext_html_url=http://www.nature.com/articles/s42256-019-0022-7;,citation_issue=2;,citation_doi=10.1038/s42256-019-0022-7;,citation_issn=2522-5839;,citation_volume=1;,citation_journal_title=Nature Machine Intelligence;">
<meta name="citation_reference" content="citation_title=Evaluating the popularity of R in ecology;,citation_abstract=The programming language R is widely used in many ﬁelds. We explored the extent of reported R use in the ﬁeld of ecology using the Web of Science and text mining. We analyzed the frequencies of R packages reported in more than 60,000 peer-reviewed articles published in 30 ecology journals during a 10-yr period ending in 2017. The number of studies reported using R as their primary tool in data analysis increased linearly from 11.4% in 2008 to 58.0% in 2017. The top 10 packages reported were lme4, vegan, nlme, ape, MuMIn, MASS, mgcv, ade4, multcomp, and car. The increasing popularity of R has most likely furthered open science in ecological research because it can improve reproducibility of analyses and captures workﬂows when scripts and codes are included and shared. These ﬁndings may not be entirely unique to R because there are other programming languages used by ecologists, but they do strongly suggest that given the relatively high frequency of reported use of R, it is a signiﬁcant component of contemporary analytics in the ﬁeld of ecology.;,citation_author=Jiangshan Lai;,citation_author=Christopher J. Lortie;,citation_author=Robert A. Muenchen;,citation_author=Jian Yang;,citation_author=Keping Ma;,citation_publication_date=2019-01;,citation_cover_date=2019-01;,citation_year=2019;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/10.1002/ecs2.2567;,citation_issue=1;,citation_doi=10.1002/ecs2.2567;,citation_issn=2150-8925, 2150-8925;,citation_volume=10;,citation_journal_title=Ecosphere;">
<meta name="citation_reference" content="citation_title=Reticulate: R interface to python;,citation_author=JJ Allaire;,citation_author=Kevin Ushey;,citation_author=Yuan Tang;,citation_author=Dirk Eddelbuettel;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_fulltext_html_url=https://github.com/rstudio/reticulate;">
<meta name="citation_reference" content="citation_title=Large predators in the Alps: The fall and rise of man’s competitors;,citation_abstract=The brown bear Ursus arctos, wolf Canis lupus, and Eurasian lynx Lynx lynx vanished during the 18th and 19th centuries from all regions of high human activity in Europe because of direct persecution and environmental changes. Bear, wolf, and lynx were vulnerable in different ways to deforestation and the destruction of wild ungulate populations. Analysing the ecological factors responsible for the fall of the large carnivores can help to prepare their recovery. The return of large predators into semi-natural areas such as the Alps is possible, as the forests have expanded, and the wild ungulate populations increased. Lynx reintroduction in the Alps started in the 1970s. Wolves returned to the south-western Alps from the central Italian population in the early 1990s. The brown bear is recolonising the Austrian Alps from Slovenia. However, the modern protective legislation is not backed by a cooperative attitude among the affected people. In rural areas, large carnivores are still regarded as unrestrained killers of wildlife and livestock. Ecological conditions and husbandry in the Alps have been altered substantially since the large carnivores were eradicated, and the potential for conflicts has diminished. But stockmen have lost any remaining tradition of coexistence with large predators, and sheep are again very abundant in the Swiss Alps. The return of the large predators will not be possible without changing the system of sheep-husbandry. The rural people are not yet willing to do so. They generally object to any change in their lifestyle induced from outside, and the large predators become a negative symbol for restrictive conservation measures considered to hinder economic development. Nature conservation, including the reintegration of large predators, must be integrated into rural development; local people must be much more involved in this process.;,citation_author=Urs Breitenmoser;,citation_publication_date=1998-03;,citation_cover_date=1998-03;,citation_year=1998;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S0006320797000840;,citation_issue=3;,citation_doi=10.1016/S0006-3207(97)00084-0;,citation_issn=0006-3207;,citation_volume=83;,citation_journal_title=Biological Conservation;,citation_series_title=Conservation Biology and Biodiversity Strategies;">
<meta name="citation_reference" content="citation_title=Distribution trend of the Eurasian lynx Lynx lynx populations in France;,citation_abstract=The changes in the distribution of the lynx populations in France were analysed with 3,760 data gathered between 1974 and 2002. Maps were drawn for successive three-year periods on a fixed 3 km × 3 km grid. Lynx were assumed “present” in every 3 × 3 km grid with one or several data, as well as in the eight surrounding grids, i.e. in a total area of 81 km2. In 20002002, the total and permanent lynx range covered, respectively, 8,622 km2 and 5,823 km2 in the French Jura mountains (11,500 km2 and at least 6,000 km2 for the whole French and Swiss Jura Massif). Almost all forested mountain massifs are already occupied by lynx in that region. The total and permanent lynx range covered 3,159 km2 and 1,962 km2 in the Vosges Massif. This small range and still precarious status of the population 20 years after the first lynx re-introductions, should be related to the high mortality rate of the released lynx and low number of founders. Exchanges between the Jura and the Vosges lynx populations could however be expected in the near future. The total lynx range covered 4,365 km2 in the Alps but no permanent area of presence was noticed. The regular increase in the number of islets of presence and the progressive colonization from the north towards the extreme south of the Alps suggested an insufficient observation pressure rather than the absence of any established population in the French Alps. In the future, the status of these three reintroduced populations should probably be comforted by exchanges between the Jura and the Vosges mountains and the Jura and the northern Alps.;,citation_author=Jean-Michel Vandel;,citation_author=Philippe Stahl;,citation_publication_date=2005-01;,citation_cover_date=2005-01;,citation_year=2005;,citation_fulltext_html_url=https://www.degruyter.com/document/doi/10.1515/mamm.2005.013/html;,citation_issue=2;,citation_doi=10.1515/mamm.2005.013;,citation_issn=0025-1461, 1864-1547;,citation_volume=69;,citation_journal_title=Mammalia;">
<meta name="citation_reference" content="citation_title=Variation in diet, prey selectivity and home-range size of Eurasian lynx Lynx lynx in Switzerland;,citation_abstract=To analyse the factors responsible for the interplay of Eurasian lynx Lynx lynx predation and home-range size, we reviewed patterns of lynx predation in Switzerland by comparing the prey spectrum of lynx in five studies performed in the following study areas: the northwestern Alps, where lynx were studied both in the 1980s and 1990s, the central Alps, the Jura Mountains, and northeastern Switzerland. We then compared home-range size of female lynx with two indirect measures of prey abundance, roe deer Capreolus capreolus and chamois Rupicapra rupicapra harvested per km2 and habitat suitability for roe deer and chamois as derived from a GIS model. Lynx diets were similar among sites. Roe deer and chamois made up 90% of prey items in all five studies. Comparing the proportion of roe deer and chamois in the diet with availability, Manly’s preference indices indicated selective predation in all studies. Roe deer were preferred over chamois in all areas except in the Jura Mountains where relatively few chamois were present. Predation was least selective in northeastern Switzerland, where the initial phase of recolonisation by lynx was studied. Variation in prey availability is often identified as an important factor explaining intraspecific variation in home-range size. Due to differences in roe deer and chamois abundance from one study area to another, we expected female lynx home ranges to decrease with increasing prey abundance. The predictors for Minimum Convex Polygon (MCP) and Kernel home-range estimators differed. MCP home-range sizes were best explained by the interactions of study with the number of locations per lynx, roe deer harvested per km2, and good roe deer habitat, whereas Kernel home-range sizes were best explained by the interactions of study with good roe deer habitat, good chamois habitat, and the interaction of good roe deer and chamois habitat plus an additive effect of the study. Contrary to our expectations, there was no simple correlation of prime roe deer and chamois habitat nor between the number of roe deer and chamois harvested per km2 and the size of female lynx home ranges. The comparison of the five studies suggested that this expectation may only be valid if lynx populations are close to carrying capacity (e.g. the Jura Mountains and the northwestern Alps in the 1990s). For predictions of home-range size both habitat (spatial factor) and the status and dynamic of the predator/prey populations (temporal factor) need to be taken into account.;,citation_author=Anja Molinari-Jobin;,citation_author=Fridolin Zimmermann;,citation_author=Andreas Ryser;,citation_author=Christine Breitenmoser-Würsten;,citation_author=Simon Capt;,citation_author=Urs Breitenmoser;,citation_author=Paolo Molinari;,citation_author=Heinrich Haller;,citation_author=Roman Eyholzer;,citation_publication_date=2007-12;,citation_cover_date=2007-12;,citation_year=2007;,citation_fulltext_html_url=https://bioone.org/journals/wildlife-biology/volume-13/issue-4/0909-6396_2007_13_393_VIDPSA_2.0.CO_2/Variation-in-diet-prey-selectivity-and-home-range-size-of/10.2981/0909-6396(2007)13[393:VIDPSA]2.0.CO;2.full;,citation_issue=4;,citation_doi=10.2981/0909-6396(2007)13[393:VIDPSA]2.0.CO;2;,citation_issn=0909-6396, 1903-220X;,citation_volume=13;,citation_journal_title=Wildlife Biology;">
<meta name="citation_reference" content="citation_title=Spatial density estimates of Eurasian lynx (Lynx lynx) in the French Jura and Vosges Mountains;,citation_abstract=Obtaining estimates of animal population density is a key step in providing sound conservation and management strategies for wildlife. For many large carnivores however, estimating density is difficult because these species are elusive and wide-ranging. Here, we focus on providing the first density estimates of the Eurasian lynx (Lynx lynx) in the French Jura and Vosges mountains. We sampled a total of 413 camera trapping sites (with two cameras per site) between January 2011 and April 2016 in seven study areas across seven counties of the French Jura and Vosges mountains. We obtained 592 lynx detections over 19,035 trap days in the Jura mountains and 0 detection over 6,804 trap days in the Vosges mountains. Based on coat patterns, we identified a total number of 92 unique individuals from photographs, including 16 females, 13 males, and 63 individuals of unknown sex. Using spatial capture–recapture (SCR) models, we estimated abundance in the study areas between 5 (SE = 0.1) and 29 (0.2) lynx and density between 0.24 (SE = 0.02) and 0.91 (SE = 0.03) lynx per 100 km2. We also provide a comparison with nonspatial density estimates and discuss the observed discrepancies. Our study is yet another example of the advantage of combining SCR methods and noninvasive sampling techniques to estimate density for elusive and wide-ranging species, like large carnivores. While the estimated densities in the French Jura mountains are comparable to other lynx populations in Europe, the fact that we detected no lynx in the Vosges mountains is alarming. Connectivity should be encouraged between the French Jura mountains, the Vosges mountains, and the Palatinate Forest in Germany where a reintroduction program is currently ongoing. Our density estimates will help in setting a baseline conservation status for the lynx population in France.;,citation_author=Olivier Gimenez;,citation_author=Sylvain Gatti;,citation_author=Christophe Duchamp;,citation_author=Estelle Germain;,citation_author=Alain Laurent;,citation_author=Fridolin Zimmermann;,citation_author=Eric Marboutin;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.5668;,citation_issue=20;,citation_doi=10.1002/ece3.5668;,citation_issn=2045-7758;,citation_volume=9;,citation_journal_title=Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Optimizing the size of the area surveyed for monitoring a Eurasian lynx (Lynx lynx) population in the Swiss Alps by means of photographic capture-recapture;,citation_abstract=We studied the influence of surveyed area size on density estimates by means of camera-trapping in a low-density felid population (1-2 individuals/100 km(2) ). We applied non-spatial capture-recapture (CR) and spatial CR (SCR) models for Eurasian lynx during winter 2005/2006 in the northwestern Swiss Alps by sampling an area divided into 5 nested plots ranging from 65 to 760 km(2) . CR model density estimates (95% CI) for models M0 and Mh decreased from 2.61 (1.55-3.68) and 3.6 (1.62-5.57) independent lynx/100 km(2) , respectively, in the smallest to 1.20 (1.04-1.35) and 1.26 (0.89-1.63) independent lynx/100 km(2) , respectively, in the largest area surveyed. SCR model density estimates also decreased with increasing sampling area but not significantly. High individual range overlaps in relatively small areas (the edge effect) is the most plausible reason for this positive bias in the CR models. Our results confirm that SCR models are much more robust to changes in trap array size than CR models, thus avoiding overestimation of density in smaller areas. However, when a study is concerned with monitoring population changes, large spatial efforts (area surveyed ≥760 km(2) ) are required to obtain reliable and precise density estimates with these population densities and recapture rates.;,citation_author=Fridolin Zimmermann;,citation_author=Christine Breitenmoser-Würsten;,citation_author=Anja Molinari-Jobin;,citation_author=Urs Breitenmoser;,citation_publication_date=2013-09;,citation_cover_date=2013-09;,citation_year=2013;,citation_issue=3;,citation_doi=10.1111/1749-4877.12017;,citation_issn=1749-4877;,citation_pmid=24020463;,citation_volume=8;,citation_journal_title=Integrative Zoology;">
<meta name="citation_reference" content="citation_title=A multispecies occupancy model for two or more interacting species;,citation_abstract=Summary Species occurrence is influenced by environmental conditions and the presence of other species. Current approaches for multispecies occupancy modelling are practically limited to two interacting species and often require the assumption of asymmetric interactions. We propose a multispecies occupancy model that can accommodate two or more interacting species. We generalize the single-species occupancy model to two or more interacting species by assuming the latent occupancy state is a multivariate Bernoulli random variable. We propose modelling the probability of each potential latent occupancy state with both a multinomial logit and a multinomial probit model and present details of a Gibbs sampler for the latter. As an example, we model co-occurrence probabilities of bobcat (Lynx rufus), coyote (Canis latrans), grey fox (Urocyon cinereoargenteus) and red fox (Vulpes vulpes) as a function of human disturbance variables throughout 6 Mid-Atlantic states in the eastern United States. We found evidence for pairwise interactions among most species, and the probability of some pairs of species occupying the same site varied along environmental gradients; for example, occupancy probabilities of coyote and grey fox were independent at sites with little human disturbance, but these two species were more likely to occur together at sites with high human disturbance. Ecological communities are composed of multiple interacting species. Our proposed method improves our ability to draw inference from such communities by permitting modelling of detection/non-detection data from an arbitrary number of species, without assuming asymmetric interactions. Additionally, our proposed method permits modelling the probability two or more species occur together as a function of environmental variables. These advancements represent an important improvement in our ability to draw community-level inference from multiple interacting species that are subject to imperfect detection.;,citation_author=Christopher T. Rota;,citation_author=Marco A. R. Ferreira;,citation_author=Roland W. Kays;,citation_author=Tavis D. Forrester;,citation_author=Elizabeth L. Kalies;,citation_author=William J. McShea;,citation_author=Arielle W. Parsons;,citation_author=Joshua J. Millspaugh;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_fulltext_html_url=https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.12587;,citation_issue=10;,citation_doi=https://doi.org/10.1111/2041-210X.12587;,citation_volume=7;,citation_journal_title=Methods in Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Multivariate Bernoulli distribution;,citation_author=Bin Dai;,citation_author=Shilin Ding;,citation_author=Grace Wahba;,citation_publication_date=2013-09;,citation_cover_date=2013-09;,citation_year=2013;,citation_fulltext_html_url=https://projecteuclid.org/journals/bernoulli/volume-19/issue-4/Multivariate-Bernoulli-distribution/10.3150/12-BEJSP10.full;,citation_issue=4;,citation_doi=10.3150/12-BEJSP10;,citation_issn=1350-7265;,citation_volume=19;,citation_journal_title=Bernoulli;">
<meta name="citation_reference" content="citation_title=unmarked: An R package for fitting hierarchical models of wildlife occurrence and abundance;,citation_author=Ian Fiske;,citation_author=Richard Chandler;,citation_publication_date=2011;,citation_cover_date=2011;,citation_year=2011;,citation_fulltext_html_url=https://www.jstatsoft.org/v43/i10/;,citation_issue=10;,citation_volume=43;,citation_journal_title=Journal of Statistical Software;">
<meta name="citation_reference" content="citation_title=Transfer Learning for Visual Categorization: A Survey;,citation_abstract=Regular machine learning and data mining techniques study the training data for future inferences under a major assumption that the future data are within the same feature space or have the same distribution as the training data. However, due to the limited availability of human labeled training data, training data that stay in the same feature space or have the same distribution as the future data cannot be guaranteed to be sufficient enough to avoid the over-fitting problem. In real-world applications, apart from data in the target domain, related data in a different domain can also be included to expand the availability of our prior knowledge about the target future data. Transfer learning addresses such cross-domain learning problems by extracting useful information from data in a related domain and transferring them for being used in target tasks. In recent years, with transfer learning being applied to visual categorization, some typical problems, e.g., view divergence in action recognition tasks and concept drifting in image classification tasks, can be efficiently solved. In this paper, we survey state-of-the-art transfer learning algorithms in visual categorization applications, such as object recognition, image classification, and human action recognition.;,citation_author=Ling Shao;,citation_author=Fan Zhu;,citation_author=Xuelong Li;,citation_publication_date=2015-05;,citation_cover_date=2015-05;,citation_year=2015;,citation_issue=5;,citation_doi=10.1109/TNNLS.2014.2330900;,citation_issn=2162-2388;,citation_volume=26;,citation_journal_title=IEEE Transactions on Neural Networks and Learning Systems;">
<meta name="citation_reference" content="citation_title=A survey on Image Data Augmentation for Deep Learning;,citation_abstract=Deep convolutional neural networks have performed remarkably well on many Computer Vision tasks. However, these networks are heavily reliant on big data to avoid overfitting. Overfitting refers to the phenomenon when a network learns a function with very high variance such as to perfectly model the training data. Unfortunately, many application domains do not have access to big data, such as medical image analysis. This survey focuses on Data Augmentation, a data-space solution to the problem of limited data. Data Augmentation encompasses a suite of techniques that enhance the size and quality of training datasets such that better Deep Learning models can be built using them. The image augmentation algorithms discussed in this survey include geometric transformations, color space augmentations, kernel filters, mixing images, random erasing, feature space augmentation, adversarial training, generative adversarial networks, neural style transfer, and meta-learning. The application of augmentation methods based on GANs are heavily covered in this survey. In addition to augmentation techniques, this paper will briefly discuss other characteristics of Data Augmentation such as test-time augmentation, resolution impact, final dataset size, and curriculum learning. This survey will present existing methods for Data Augmentation, promising developments, and meta-level decisions for implementing Data Augmentation. Readers will understand how Data Augmentation can improve the performance of their models and expand limited datasets to take advantage of the capabilities of big data.;,citation_author=Connor Shorten;,citation_author=Taghi M. Khoshgoftaar;,citation_publication_date=2019-07;,citation_cover_date=2019-07;,citation_year=2019;,citation_fulltext_html_url=https://doi.org/10.1186/s40537-019-0197-0;,citation_issue=1;,citation_doi=10.1186/s40537-019-0197-0;,citation_issn=2196-1115;,citation_volume=6;,citation_journal_title=Journal of Big Data;">
<meta name="citation_reference" content="citation_title=Deep Residual Learning for Image Recognition;,citation_abstract=Deeper neural networks are more difficult to train. We present a residual learning framework to ease the training of networks that are substantially deeper than those used previously. We explicitly reformulate the layers as learning residual functions with reference to the layer inputs, instead of learning unreferenced functions. We provide comprehensive empirical evidence showing that these residual networks are easier to optimize, and can gain accuracy from considerably increased depth. On the ImageNet dataset we evaluate residual nets with a depth of up to 152 layers - 8× deeper than VGG nets [40] but still having lower complexity. An ensemble of these residual nets achieves 3.57% error on the ImageNet test set. This result won the 1st place on the ILSVRC 2015 classification task. We also present analysis on CIFAR-10 with 100 and 1000 layers. The depth of representations is of central importance for many visual recognition tasks. Solely due to our extremely deep representations, we obtain a 28% relative improvement on the COCO object detection dataset. Deep residual nets are foundations of our submissions to ILSVRC &amp;amp;amp; COCO 2015 competitions1, where we also won the 1st places on the tasks of ImageNet detection, ImageNet localization, COCO detection, and COCO segmentation.;,citation_author=Kaiming He;,citation_author=Xiangyu Zhang;,citation_author=Shaoqing Ren;,citation_author=Jian Sun;,citation_publication_date=2016-06;,citation_cover_date=2016-06;,citation_year=2016;,citation_doi=10.1109/CVPR.2016.90;,citation_conference_title=2016 IEEE Conference on Computer Vision and Pattern Recognition (CVPR);">
<meta name="citation_reference" content="citation_title=PyTorch: An imperative style, high-performance deep learning library;,citation_author=Adam Paszke;,citation_author=Sam Gross;,citation_author=Francisco Massa;,citation_author=Adam Lerer;,citation_author=James Bradbury;,citation_author=Gregory Chanan;,citation_author=Trevor Killeen;,citation_author=Zeming Lin;,citation_author=Natalia Gimelshein;,citation_author=Luca Antiga;,citation_author=Alban Desmaison;,citation_author=Andreas Kopf;,citation_author=Edward Yang;,citation_author=Zachary DeVito;,citation_author=Martin Raison;,citation_author=Alykhan Tejani;,citation_author=Sasank Chilamkurthy;,citation_author=Benoit Steiner;,citation_author=Lu Fang;,citation_author=Junjie Bai;,citation_author=Soumith Chintala;,citation_editor=H. Wallach;,citation_editor=H. Larochelle;,citation_editor=A. Beygelzimer;,citation_editor=F. dAlché-Buc;,citation_editor=E. Fox;,citation_editor=R. Garnett;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=http://papers.neurips.cc/paper/9015-pytorch-an-imperative-style-high-performance-deep-learning-library.pdf;,citation_inbook_title=Advances in neural information processing systems 32;">
<meta name="citation_reference" content="citation_title=How transferable are features in deep neural networks?;,citation_abstract=Many deep neural networks trained on natural images exhibit a curious phenomenon in common: on the first layer they learn features similar to Gabor filters and color blobs. Such first-layer features appear not to be specific to a particular dataset or task, but general in that they are applicable to many datasets and tasks. Features must eventually transition from general to specific by the last layer of the network, but this transition has not been studied extensively. In this paper we experimentally quantify the generality versus specificity of neurons in each layer of a deep convolutional neural network and report a few surprising results. Transferability is negatively affected by two distinct issues: (1) the specialization of higher layer neurons to their original task at the expense of performance on the target task, which was expected, and (2) optimization difficulties related to splitting networks between co-adapted neurons, which was not expected. In an example network trained on ImageNet, we demonstrate that either of these two issues may dominate, depending on whether features are transferred from the bottom, middle, or top of the network. We also document that the transferability of features decreases as the distance between the base task and target task increases, but that transferring features even from distant tasks can be better than using random features. A final surprising result is that initializing a network with transferred features from almost any number of layers can produce a boost to generalization that lingers even after fine-tuning to the target dataset.;,citation_author=Jason Yosinski;,citation_author=Jeff Clune;,citation_author=Yoshua Bengio;,citation_author=Hod Lipson;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_conference_title=Proceedings of the 27th international conference on neural information processing systems - volume 2;,citation_conference=MIT Press;,citation_series_title=NIPS’14;">
<meta name="citation_reference" content="citation_title=An approach to rapid processing of camera trap images with minimal human input;,citation_abstract=Abstract Camera traps have become an extensively utilized tool in ecological research, but the manual processing of images created by a network of camera traps rapidly becomes an overwhelming task, even for small camera trap studies. We used transfer learning to create convolutional neural network (CNN) models for identification and classification. By utilizing a small dataset with an average of 275 labeled images per species class, the model was able to distinguish between species and remove false triggers. We trained the model to detect 17 object classes with individual species identification, reaching an accuracy up to 92% and an average F1 score of 85%. Previous studies have suggested the need for thousands of images of each object class to reach results comparable to those achieved by human observers; however, we show that such accuracy can be achieved with fewer images. With transfer learning and an ongoing camera trap study, a deep learning model can be successfully created by a small camera trap study. A generalizable model produced from an unbalanced class set can be utilized to extract trap events that can later be confirmed by human processors.;,citation_author=Matthew T. Duggan;,citation_author=Melissa F. Groleau;,citation_author=Ethan P. Shealy;,citation_author=Lillian S. Self;,citation_author=Taylor E. Utter;,citation_author=Matthew M. Waller;,citation_author=Bryan C. Hall;,citation_author=Chris G. Stone;,citation_author=Layne L. Anderson;,citation_author=Timothy A. Mousseau;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.7970;,citation_doi=https://doi.org/10.1002/ece3.7970;,citation_journal_title=Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=A penalized likelihood for multi-species occupancy models improves predictions of species interactions;,citation_author=Hannah L. Clipp;,citation_author=Amber L. Evans;,citation_author=Brin E. Kessinger;,citation_author=K. Kellner;,citation_author=Christopher T. Rota;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Improving occupancy estimation when two types of observational error occur: Non-detection and species misidentification;,citation_abstract=Efforts to draw inferences about species occurrence frequently account for false negatives, the common situation when individuals of a species are not detected even when a site is occupied. However, recent studies suggest the need to also deal with false positives, which occur when species are misidentified so that a species is recorded as detected when a site is unoccupied. Bias in estimators of occupancy, colonization, and extinction can be severe when false positives occur. Accordingly, we propose models that simultaneously account for both types of error. Our approach can be used to improve estimates of occupancy for study designs where a subset of detections is of a type or method for which false positives can be assumed to not occur. We illustrate properties of the estimators with simulations and data for three species of frogs. We show that models that account for possible misidentification have greater support (lower AIC for two species) and can yield substantially different occupancy estimates than those that do not. When the potential for misidentification exists, researchers should consider analytical techniques that can account for this source of error, such as those presented here.;,citation_author=David A. Miller;,citation_author=James D. Nichols;,citation_author=Brett T. McClintock;,citation_author=Evan H. Campbell Grant;,citation_author=Larissa L. Bailey;,citation_author=Linda A. Weir;,citation_publication_date=2011;,citation_cover_date=2011;,citation_year=2011;,citation_fulltext_html_url=https://esajournals.onlinelibrary.wiley.com/doi/abs/10.1890/10-1396.1;,citation_issue=7;,citation_doi=https://doi.org/10.1890/10-1396.1;,citation_volume=92;,citation_journal_title=Ecology;">
<meta name="citation_reference" content="citation_title=Two-species occupancy modelling accounting for species misidentification and non-detection;,citation_abstract=Abstract In occupancy studies, species misidentification can lead to false-positive detections, which can cause severe estimator biases. Currently, all models that account for false-positive errors only consider omnibus sources of false detections and are limited to single-species occupancy. However, false detections for a given species often occur because of the misidentification with another, closely related species. To exploit this explicit source of false-positive detection error, we develop a two-species occupancy model that accounts for misidentifications between two species of interest. As with other false-positive models, identifiability is greatly improved by the availability of unambiguous detections at a subset of site x occasions. Here, we consider the case where some of the field observations can be confirmed using laboratory or other independent identification methods (“confirmatory data”). We performed three simulation studies to (1) assess the model’s performance under various realistic scenarios, (2) investigate the influence of the proportion of confirmatory data on estimator accuracy and (3) compare the performance of this two-species model with that of the single-species false-positive model. The model shows good performance under all scenarios, even when only small proportions of detections are confirmed (e.g. 5%). It also clearly outperforms the single-species model. We illustrate application of this model using a 4-year dataset on two sympatric species of lungless salamanders: the US federally endangered Shenandoah salamander Plethodon shenandoah, and its presumed competitor, the red-backed salamander Plethodon cinereus. Occupancy of red-backed salamanders appeared very stable across the 4&nbsp;years of study, whereas the Shenandoah salamander displayed substantial turnover in occupancy of forest habitats among years. Given the extent of species misidentification issues in occupancy studies, this modelling approach should help improve the reliability of estimates of species distribution, which is the goal of many studies and monitoring programmes. Further developments, to account for different forms of state uncertainty, can be readily undertaken under our general approach.;,citation_author=Thierry Chambert;,citation_author=Evan H. Campbell Grant;,citation_author=David A. W. Miller;,citation_author=James D. Nichols;,citation_author=Kevin P. Mulder;,citation_author=Adrianne B. Brand;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.12985;,citation_issue=6;,citation_doi=https://doi.org/10.1111/2041-210X.12985;,citation_volume=9;,citation_journal_title=Methods in Ecology and Evolution;">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title"><a href="https://computo.sfds.asso.fr">
        <img src="https://computo.sfds.asso.fr/assets/img/logo_notext_white.png" height="60px">
      </a> &nbsp; Trade-off between deep learning for species identification and inference about predator-prey co-occurrence</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> source</button></div></div>
            <p class="subtitle lead">Reproducible R workflow integrating models in computer vision and ecological statistics</p>
            <p><a href="http://creativecommons.org/licenses/by/4.0/"><img src="https://i.creativecommons.org/l/by/4.0/80x15.png" alt="Creative Commons BY License"></a>
ISSN 2824-7795</p>
                </div>
  </div>
    
    <div class="quarto-title-meta-author">
      <div class="quarto-title-meta-heading">Authors</div>
      <div class="quarto-title-meta-heading">Affiliations</div>
          
          <div class="quarto-title-meta-contents">
        <a href="https://oliviergimenez.github.io/">Olivier Gimenez</a> 
      </div>
          
          <div class="quarto-title-meta-contents">
              <p class="affiliation">
                  <a href="https://www.cefe.cnrs.fr/">
                  CEFE, Univ Montpellier, CNRS, EPHE, IRD, Montpellier, France
                  </a>
                </p>
            </div>
            <div class="quarto-title-meta-contents">
        Maëlis Kervellec 
      </div>
          
          <div class="quarto-title-meta-contents">
              <p class="affiliation">
                  <a href="https://www.cefe.cnrs.fr/">
                  CEFE, Univ Montpellier, CNRS, EPHE, IRD, Montpellier, France
                  </a>
                </p>
            </div>
            <div class="quarto-title-meta-contents">
        Jean-Baptiste Fanjul 
      </div>
          
          <div class="quarto-title-meta-contents">
              <p class="affiliation">
                  <a href="https://www.chasseurdujura.com/">
                  Fédérations Départementales des Chasseurs du Jura, Arlay, France
                  </a>
                </p>
            </div>
            <div class="quarto-title-meta-contents">
        Anna Chaine 
      </div>
          
          <div class="quarto-title-meta-contents">
              <p class="affiliation">
                  <a href="https://www.cefe.cnrs.fr/">
                  CEFE, Univ Montpellier, CNRS, EPHE, IRD, Montpellier, France
                  </a>
                </p>
            </div>
            <div class="quarto-title-meta-contents">
        Lucile Marescot 
      </div>
          
          <div class="quarto-title-meta-contents">
              <p class="affiliation">
                  <a href="https://www6.montpellier.inrae.fr/cbgp">
                  CBGP, Univ Montpellier, CIRAD, INRAE, Institut Agro, IRD, Montpellier, France
                  </a>
                </p>
            </div>
            <div class="quarto-title-meta-contents">
        Yoann Bollet 
      </div>
          
          <div class="quarto-title-meta-contents">
              <p class="affiliation">
                  <a href="http://www.fdcain.com/">
                  Fédération Départementale des Chasseurs de l’Ain, Bourg-en-Bresse, France
                  </a>
                </p>
            </div>
            <div class="quarto-title-meta-contents">
        Christophe Duchamp 
      </div>
          
          <div class="quarto-title-meta-contents">
              <p class="affiliation">
                  <a href="https://www.ofb.gouv.fr/">
                  Office Français de la Biodiversité, Unité Espèces à enjeux, Gap, France
                  </a>
                </p>
            </div>
        </div>
                    
  <div class="quarto-title-meta">
                                
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 22, 2022</p>
      </div>
    </div>
                                    
      <div>
      <div class="quarto-title-meta-heading">Modified</div>
      <div class="quarto-title-meta-contents">
        <p class="date-modified">February 7, 2023</p>
      </div>
    </div>
      
                  
      <div>
      <div class="quarto-title-meta-heading">Keywords</div>
      <div class="quarto-title-meta-contents">
        <p class="date">computer-vision, deep-learning, species-distribution-modeling, ecological-statistics</p>
      </div>
    </div>
    
    <div>
      <div class="quarto-title-meta-heading">Status</div>
      <div class="quarto-title-meta-contents">
              <a href="https://github.com/computorg/published-202204-deeplearning-occupancy-lynx"><img src="https://github.com/computorg/published-202204-deeplearning-occupancy-lynx/workflows/build/badge.svg" alt="build status"></a>
                    <p class="date"></p>
        <a href="https://github.com/computorg/published-202204-deeplearning-occupancy-lynx/issues/1"><img src="https://img.shields.io/badge/review-report%201-blue" alt="review 1"></a>
        <a href="https://github.com/computorg/published-202204-deeplearning-occupancy-lynx/issues/2"><img src="https://img.shields.io/badge/review-report%202-blue" alt="review 2"></a>
            </div>
    </div>

  </div>
                                                
  <div>
    <div class="abstract">
    <div class="abstract-title">Abstract</div>
      <p>Deep learning is used in computer vision problems with important applications in several scientific fields. In ecology for example, there is a growing interest in deep learning for automatizing repetitive analyses on large amounts of images, such as animal species identification. However, there are challenging issues toward the wide adoption of deep learning by the community of ecologists. First, there is a programming barrier as most algorithms are written in <code>Python</code> while most ecologists are versed in <code>R</code>. Second, recent applications of deep learning in ecology have focused on computational aspects and simple tasks without addressing the underlying ecological questions or carrying out the statistical data analysis to answer these questions. Here, we showcase a reproducible <code>R</code> workflow integrating both deep learning and statistical models using predator-prey relationships as a case study. We illustrate deep learning for the identification of animal species on images collected with camera traps, and quantify spatial co-occurrence using multispecies occupancy models. Despite average model classification performances, ecological inference was similar whether we analysed the ground truth dataset or the classified dataset. This result calls for further work on the trade-offs between time and resources allocated to train models with deep learning and our ability to properly address key ecological questions with biodiversity monitoring. We hope that our reproducible workflow will be useful to ecologists and applied statisticians.</p>
    </div>
  </div>

  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="toc-section-number">1</span>  Introduction</a></li>
  <li><a href="#collecting-images-with-camera-traps" id="toc-collecting-images-with-camera-traps" class="nav-link" data-scroll-target="#collecting-images-with-camera-traps"><span class="toc-section-number">2</span>  Collecting images with camera traps</a></li>
  <li><a href="#deep-learning-for-species-identification" id="toc-deep-learning-for-species-identification" class="nav-link" data-scroll-target="#deep-learning-for-species-identification"><span class="toc-section-number">3</span>  Deep learning for species identification</a>
  <ul class="collapse">
  <li><a href="#training---jura-study-site" id="toc-training---jura-study-site" class="nav-link" data-scroll-target="#training---jura-study-site"><span class="toc-section-number">3.1</span>  Training - Jura study site</a></li>
  <li><a href="#transferability---ain-study-site" id="toc-transferability---ain-study-site" class="nav-link" data-scroll-target="#transferability---ain-study-site"><span class="toc-section-number">3.2</span>  Transferability - Ain study site</a></li>
  </ul></li>
  <li><a href="#spatial-co-occurrence" id="toc-spatial-co-occurrence" class="nav-link" data-scroll-target="#spatial-co-occurrence"><span class="toc-section-number">4</span>  Spatial co-occurrence</a></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion"><span class="toc-section-number">5</span>  Discussion</a></li>
  <li><a href="#appendix-reproducible-example-of-species-identification-on-camera-trap-images-with-cpu" id="toc-appendix-reproducible-example-of-species-identification-on-camera-trap-images-with-cpu" class="nav-link" data-scroll-target="#appendix-reproducible-example-of-species-identification-on-camera-trap-images-with-cpu"><span class="toc-section-number">6</span>  Appendix: Reproducible example of species identification on camera trap images with CPU</a>
  <ul class="collapse">
  <li><a href="#training-and-validation-datasets" id="toc-training-and-validation-datasets" class="nav-link" data-scroll-target="#training-and-validation-datasets"><span class="toc-section-number">6.1</span>  Training and validation datasets</a></li>
  <li><a href="#transfer-learning" id="toc-transfer-learning" class="nav-link" data-scroll-target="#transfer-learning"><span class="toc-section-number">6.2</span>  Transfer learning</a></li>
  <li><a href="#transferability" id="toc-transferability" class="nav-link" data-scroll-target="#transferability"><span class="toc-section-number">6.3</span>  Transferability</a></li>
  </ul></li>
  <li><a href="#session-information" id="toc-session-information" class="nav-link" data-scroll-target="#session-information">Session information</a></li>
  <li><a href="#acknowledgments" id="toc-acknowledgments" class="nav-link" data-scroll-target="#acknowledgments">Acknowledgments</a></li>
  <li><a href="#bibliography" id="toc-bibliography" class="nav-link" data-scroll-target="#bibliography">Bibliography</a></li>
  </ul>
</nav>
</div>
<main class="content quarto-banner-title-block page-columns page-full" id="quarto-document-content">



<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>Computer vision is a field of artificial intelligence in which a machine is taught how to extract and interpret the content of an image <span class="citation" data-cites="NIPS2012_4824">(<a href="#ref-NIPS2012_4824" role="doc-biblioref">Krizhevsky, Sutskever, and Hinton 2012</a>)</span>. Computer vision relies on deep learning that allows computational models to learn from training data – a set of manually labelled images – and make predictions on new data – a set of unlabelled images <span class="citation" data-cites="baraniuk_science_2020 lecun_deep_2015">(<a href="#ref-baraniuk_science_2020" role="doc-biblioref">Baraniuk, Donoho, and Gavish 2020</a>; <a href="#ref-lecun_deep_2015" role="doc-biblioref">LeCun, Bengio, and Hinton 2015</a>)</span>. With the growing availability of massive data, computer vision with deep learning is being increasingly used to perform tasks such as object detection, face recognition, action and activity recognition or human pose estimation in fields as diverse as medicine, robotics, transportation, genomics, sports and agriculture <span class="citation" data-cites="andina_deep_2018">(<a href="#ref-andina_deep_2018" role="doc-biblioref">Voulodimos et al. 2018</a>)</span>.</p>
<p>In ecology in particular, there is a growing interest in deep learning for automatizing repetitive analyses on large amounts of images, such as identifying plant and animal species, distinguishing individuals of the same or different species, counting individuals or detecting relevant features <span class="citation" data-cites="christin_applications_2019 lamba_deep_2019 weinstein_computer_2018">(<a href="#ref-christin_applications_2019" role="doc-biblioref">Christin, Hervet, and Lecomte 2019</a>; <a href="#ref-lamba_deep_2019" role="doc-biblioref">Lamba et al. 2019</a>; <a href="#ref-weinstein_computer_2018" role="doc-biblioref">Weinstein 2018</a>)</span>. By saving hours of manual data analyses and tapping into massive amounts of data that keep accumulating with technological advances, deep learning has the potential to become an essential tool for ecologists and applied statisticians.</p>
<p>Despite the promising future of computer vision and deep learning, there are challenging issues toward their wide adoption by the community of ecologists <span class="citation" data-cites="wearn_responsible_2019">(e.g. <a href="#ref-wearn_responsible_2019" role="doc-biblioref">Wearn, Freeman, and Jacoby 2019</a>)</span>. First, there is a programming barrier as most algorithms are written in the <code>Python</code> language (but see <a href="https://mxnet.apache.org/versions/1.6/api/r/docs/index.html">MXNet in R</a> and the <a href="https://keras.rstudio.com/index.html">R interface to Keras</a>) while most ecologists are versed in <code>R</code> <span class="citation" data-cites="lai_evaluating_2019">(<a href="#ref-lai_evaluating_2019" role="doc-biblioref">Lai et al. 2019</a>)</span>. If ecologists are to use computer vision in routine, there is a need for bridges between these two languages (through, e.g., the <code>reticulate</code> package <span class="citation" data-cites="reticulate_ref">Allaire et al. (<a href="#ref-reticulate_ref" role="doc-biblioref">2017</a>)</span> or the <code>shiny</code> package <span class="citation" data-cites="tabak_improving_2020">Tabak et al. (<a href="#ref-tabak_improving_2020" role="doc-biblioref">2020</a>)</span>). Second, ecologists may be reluctant to develop deep learning algorithms that require large amounts of computation time and consequently come with an environmental cost due to carbon emissions <span class="citation" data-cites="strubell2019energy">(<a href="#ref-strubell2019energy" role="doc-biblioref">Strubell, Ganesh, and McCallum 2019</a>)</span>. Third, recent applications of computer vision via deep learning in ecology have focused on computational aspects and simple tasks without addressing the underlying ecological questions <span class="citation" data-cites="sutherland_identification_2013">(<a href="#ref-sutherland_identification_2013" role="doc-biblioref">Sutherland et al. 2013</a>)</span>, or carrying out statistical data analysis to answer these questions <span class="citation" data-cites="gimenez_statistical_2014">(<a href="#ref-gimenez_statistical_2014" role="doc-biblioref">Gimenez et al. 2014</a>)</span>. Although perfectly understandable given the challenges at hand, we argue that a better integration of the <em>why</em> (ecological questions), the <em>what</em> (automatically labelled images) and the <em>how</em> (statistics) would be beneficial to computer vision for ecology <span class="citation" data-cites="weinstein_computer_2018">(see also <a href="#ref-weinstein_computer_2018" role="doc-biblioref">Weinstein 2018</a>)</span>.</p>
<p>Here, we showcase a full why-what-how workflow in <code>R</code> using a case study on the structure of an ecological community (a set of co-occurring species) composed of the Eurasian lynx (<em>Lynx lynx</em>) and its two main preys. First, we introduce the case study and motivate the need for deep learning. Second we illustrate deep learning for the identification of animal species in large amounts of images, including model training and validation with a dataset of labelled images, and prediction with a new dataset of unlabelled images. Last, we proceed with the quantification of spatial co-occurrence using statistical models.</p>
</section>
<section id="collecting-images-with-camera-traps" class="level1 page-columns page-full" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Collecting images with camera traps</h1>
<p>Lynx (<em>Lynx lynx</em>) went extinct in France at the end of the 19th century due to habitat degradation, human persecution and decrease in prey availability <span class="citation" data-cites="vandel_distribution_2005">(<a href="#ref-vandel_distribution_2005" role="doc-biblioref">Vandel and Stahl 2005</a>)</span>. The species was reintroduced in Switzerland in the 1970s <span class="citation" data-cites="breitenmoser_large_1998">(<a href="#ref-breitenmoser_large_1998" role="doc-biblioref">Breitenmoser 1998</a>)</span>, then re-colonised France through the Jura mountains in the 1980s <span class="citation" data-cites="vandel_distribution_2005">(<a href="#ref-vandel_distribution_2005" role="doc-biblioref">Vandel and Stahl 2005</a>)</span>. The species is listed as endangered under the 2017 IUCN Red list and is of conservation concern in France due to habitat fragmentation, poaching and collisions with vehicles. The Jura holds the bulk of the French lynx population.</p>
<p>To better understand its distribution, we need to quantify its interactions with its main preys, roe deer (<em>Capreolus capreolus</em>) and chamois (<em>Rupicapra rupicapra</em>) <span class="citation" data-cites="molinari-jobin_variation_2007">(<a href="#ref-molinari-jobin_variation_2007" role="doc-biblioref">Molinari-Jobin et al. 2007</a>)</span>, two ungulate species that are also hunted. To assess the relative contribution of predation and hunting to the community structure and dynamics, a predator-prey program was set up jointly by the French Office for Biodiversity, the Federations of Hunters from the Jura, Ain and Haute-Savoie counties and the French National Centre for Scientific Research. Animal detections were made using a set of camera traps in the Jura mountains that were deployed in the Jura and Ain counties (see <a href="#fig-area">Figure&nbsp;1</a>). Altitude in the Jura site ranges from 520m to 1150m, and from 400m to 950m for the Ain site. Woodland areas cover 69% of the Ain site, with deciduous forests (63%) followed by coniferous (19.5%) and mixed forest (12.5%). In the Jura site, woodland areas cover 62% of the area, with mixed forests (46.6%), deciduous forests (37.3%) and coniferous (14%). In both sites, the remaining habitat is meadows used by cattle.</p>
<p>We divided the two study areas into grids of 2.7 <span class="math inline">\times</span> 2.7 km cells or sites hereafter <span class="citation" data-cites="zimmermann_optimizing_2013">(<a href="#ref-zimmermann_optimizing_2013" role="doc-biblioref">Zimmermann et al. 2013</a>)</span> in which we set two camera traps per site (Xenon white flash with passive infrared trigger mechanisms, model Capture, Ambush and Attack; Cuddeback), with 18 sites in the Jura study area, and 11 in the Ain study area that were active over the study period (from February 2016 to October 2017 for the Jura county, and from February 2017 to May 2019 for the Ain county). The location of camera traps was chosen to maximise lynx detection. More precisely, camera traps were set up along large paths in the forest, on each side of the path at 50cm high. Camera traps were checked weekly to change memory cards, batteries and to remove fresh snow after heavy snowfall.</p>
<div class="column-page">
<div class="cell">
<div class="cell-output-display">
<div id="fig-area" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="published-202204-gimenez-lynx_files/figure-html/fig-area-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;1: Study area, grid and camera trap locations.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
<div class="cell">

</div>
<p>In total, 45563 and 18044 pictures were considered in the Jura and Ain sites respectively after manually droping empty pictures and pictures with unidentified species. Note that classifying empty images could be automatised with deep learning <span class="citation" data-cites="norouzzadeh_deep_2021 tabak_improving_2020">(<a href="#ref-norouzzadeh_deep_2021" role="doc-biblioref">Norouzzadeh et al. 2021</a>; <a href="#ref-tabak_improving_2020" role="doc-biblioref">Tabak et al. 2020</a>)</span>. We identified the species present on all images by hand (see <a href="#tbl-jura">Table&nbsp;1</a>) using <code>digiKam</code> a free open-source digital photo management application (<a href="https://www.digikam.org/" class="uri">https://www.digikam.org/</a>). This operation took several weeks of labor full time, which is often identified as a limitation of camera trap studies. To expedite this tedious task, computer vision with deep learning has been identified as a promising approach <span class="citation" data-cites="norouzzadeh_deep_2021 tabak_machine_2019 willi_identifying_2019">(<a href="#ref-norouzzadeh_deep_2021" role="doc-biblioref">Norouzzadeh et al. 2021</a>; <a href="#ref-tabak_machine_2019" role="doc-biblioref">Tabak et al. 2019</a>; <a href="#ref-willi_identifying_2019" role="doc-biblioref">Willi et al. 2019</a>)</span>. <!-- Labelled images for the Ain county are available from <https://mycore.core-cloud.net/index.php/s/1DHbKdQzPKuDbxl> and that for the Jura county are available from <https://mycore.core-cloud.net/index.php/s/8GsYfAoyeDGzuNY>.  --></p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-jura" class="anchored">

<table class="table" style="margin-left: auto; margin-right: auto;"><caption>Table&nbsp;1:  Species identified in the Jura and Ain study sites with samples size (n). Only first 10 species with most images are shown. </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> Species in Jura study site </th>
   <th style="text-align:right;"> n </th>
   <th style="text-align:left;"> Species in Ain study site </th>
   <th style="text-align:right;"> n </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> human </td>
   <td style="text-align:right;"> 31644 </td>
   <td style="text-align:left;"> human </td>
   <td style="text-align:right;"> 4946 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> vehicule </td>
   <td style="text-align:right;"> 5637 </td>
   <td style="text-align:left;"> vehicule </td>
   <td style="text-align:right;"> 4454 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> dog </td>
   <td style="text-align:right;"> 2779 </td>
   <td style="text-align:left;"> dog </td>
   <td style="text-align:right;"> 2310 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> fox </td>
   <td style="text-align:right;"> 2088 </td>
   <td style="text-align:left;"> fox </td>
   <td style="text-align:right;"> 1587 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> chamois </td>
   <td style="text-align:right;"> 919 </td>
   <td style="text-align:left;"> rider </td>
   <td style="text-align:right;"> 1025 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> wild boar </td>
   <td style="text-align:right;"> 522 </td>
   <td style="text-align:left;"> roe deer </td>
   <td style="text-align:right;"> 860 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> badger </td>
   <td style="text-align:right;"> 401 </td>
   <td style="text-align:left;"> chamois </td>
   <td style="text-align:right;"> 780 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> roe deer </td>
   <td style="text-align:right;"> 368 </td>
   <td style="text-align:left;"> hunter </td>
   <td style="text-align:right;"> 593 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> cat </td>
   <td style="text-align:right;"> 343 </td>
   <td style="text-align:left;"> wild boar </td>
   <td style="text-align:right;"> 514 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> lynx </td>
   <td style="text-align:right;"> 302 </td>
   <td style="text-align:left;"> badger </td>
   <td style="text-align:right;"> 461 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="deep-learning-for-species-identification" class="level1 page-columns page-full" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Deep learning for species identification</h1>
<p>Using the images we obtained with camera traps (Table 1), we trained a model for identifying species using the Jura study site as a calibration dataset. We then assessed this model’s ability to automatically identify species on a new dataset, also known as transferability, using the Ain study site as an evaluation dataset. Even though in the present work we quantified co-occurrence between lynx and its prey, we included other species in the training to investigate the structure and dynamics of the entire community in future work. Also, the use of specific species categories instead of just a “other” category besides the focal species should help the algorithm to determine with better confidence when a picture does not contain a focal species in situations where there is no doubt that this is another species (think of a vehicle for example), or where a species is detected with which a focal species can be confused, e.g.&nbsp;lynx with fox.</p>
<section id="training---jura-study-site" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="training---jura-study-site"><span class="header-section-number">3.1</span> Training - Jura study site</h2>
<p>We selected at random 80% of the annotated images for each species in the Jura study site for training, and 20% for testing. We applied various transformations (flipping, brightness and contrast modifications following <span class="citation" data-cites="shorten_survey_2019">Shorten and Khoshgoftaar (<a href="#ref-shorten_survey_2019" role="doc-biblioref">2019</a>)</span>) to improve training (see Appendix). To reduce model training time and overcome the small number of images, we used transfer learning <span class="citation" data-cites="Yosinski2014 shao_transfer_2015">(<a href="#ref-Yosinski2014" role="doc-biblioref">Yosinski et al. 2014</a>; <a href="#ref-shao_transfer_2015" role="doc-biblioref">Shao, Zhu, and Li 2015</a>)</span> and considered a pre-trained model as a starting point. Specifically, we trained a deep convolutional neural network (ResNet-50) architecture <span class="citation" data-cites="he_deep_2016">(<a href="#ref-he_deep_2016" role="doc-biblioref">He et al. 2016</a>)</span> using the <code>fastai</code> library (<a href="https://docs.fast.ai/" class="uri">https://docs.fast.ai/</a>) that implements the <code>PyTorch</code> library <span class="citation" data-cites="NEURIPS2019_9015">(<a href="#ref-NEURIPS2019_9015" role="doc-biblioref">Paszke et al. 2019</a>)</span>. Interestingly, the <code>fastai</code> library comes with an <code>R</code> interface (<a href="https://eagerai.github.io/fastai/" class="uri">https://eagerai.github.io/fastai/</a>) that uses the <code>reticulate</code> package to communicate with <code>Python</code>, therefore allowing <code>R</code> users to access up-to-date deep learning tools. We trained models on the Montpellier Bioinformatics Biodiversity platform using a GPU machine (Titan Xp nvidia) with 16Go of RAM. We used 20 epochs which took approximately 10 hours. The computational burden prevented us from providing a full reproducible analysis, but we do so with a subsample of the dataset in the Appendix. All trained models are available from <a href="https://doi.org/10.5281/zenodo.5164796" class="uri">https://doi.org/10.5281/zenodo.5164796</a>.</p>
<p>Using the testing dataset, we calculated three metrics to evaluate our model performance at correctly identifying species <span class="citation" data-cites="Duggan2021">(e.g. <a href="#ref-Duggan2021" role="doc-biblioref">Duggan et al. 2021</a>)</span>. Specifically, we relied on <em>accuracy</em> the ratio of correct predictions to the total number of predictions, <em>recall</em> a measure of false negatives (FN; e.g.&nbsp;an image with a lynx for which our model predicts another species) with recall = TP / (TP + FN) where TP is for true positives, and <em>precision</em> a measure of false positives (FP; e.g.&nbsp;an image with any species but a lynx for which our model predicts a lynx) with precision = TP / (TP + FP). In camera trap studies, a strategy <span class="citation" data-cites="Duggan2021">(<a href="#ref-Duggan2021" role="doc-biblioref">Duggan et al. 2021</a>)</span> consists in optimizing precision if the focus is on rare species (lynx), while recall should be optimized if the focus is on commom species (chamois and roe deer).</p>
<p>We achieved 85% accuracy during training. Our model had good performances for the three classes we were interested in, with 87% precision for lynx and 81% recall for both roe deer and chamois (<a href="#tbl-deer">Table&nbsp;2</a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-deer" class="anchored">

<table class="table" style="margin-left: auto; margin-right: auto;"><caption>Table&nbsp;2:  Model performance metrics. Images from the Jura study site were used for training. </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> species </th>
   <th style="text-align:left;"> precision </th>
   <th style="text-align:left;"> recall </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;font-weight: bold;"> badger </td>
   <td style="text-align:left;"> 0.78 </td>
   <td style="text-align:left;"> 0.88 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> red deer </td>
   <td style="text-align:left;"> 0.67 </td>
   <td style="text-align:left;"> 0.21 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> chamois </td>
   <td style="text-align:left;"> 0.86 </td>
   <td style="text-align:left;"> 0.81 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> cat </td>
   <td style="text-align:left;"> 0.89 </td>
   <td style="text-align:left;"> 0.78 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> roe deer </td>
   <td style="text-align:left;"> 0.67 </td>
   <td style="text-align:left;"> 0.81 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> dog </td>
   <td style="text-align:left;"> 0.78 </td>
   <td style="text-align:left;"> 0.84 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> human </td>
   <td style="text-align:left;"> 0.99 </td>
   <td style="text-align:left;"> 0.79 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> hare </td>
   <td style="text-align:left;"> 0.32 </td>
   <td style="text-align:left;"> 0.52 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> lynx </td>
   <td style="text-align:left;"> 0.87 </td>
   <td style="text-align:left;"> 0.95 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> fox </td>
   <td style="text-align:left;"> 0.85 </td>
   <td style="text-align:left;"> 0.90 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> wild boar </td>
   <td style="text-align:left;"> 0.93 </td>
   <td style="text-align:left;"> 0.88 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> vehicule </td>
   <td style="text-align:left;"> 0.95 </td>
   <td style="text-align:left;"> 0.98 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="transferability---ain-study-site" class="level2 page-columns page-full" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="transferability---ain-study-site"><span class="header-section-number">3.2</span> Transferability - Ain study site</h2>
<p>We evaluated transferability for our trained model by predicting species on images from the Ain study site which were not used for training. Precision was 77% for lynx, and while we achieved 86% recall for roe deer, our model performed poorly for chamois with 8% recall (<a href="#tbl-chamois">Table&nbsp;3</a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-chamois" class="anchored">

<table class="table" style="margin-left: auto; margin-right: auto;"><caption>Table&nbsp;3:  Model transferability performance. Images from the Ain study site were used for assessing transferability. </caption>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:right;"> precision </th>
   <th style="text-align:right;"> recall </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;font-weight: bold;"> badger </td>
   <td style="text-align:right;"> 0.71 </td>
   <td style="text-align:right;"> 0.89 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> rider </td>
   <td style="text-align:right;"> 0.79 </td>
   <td style="text-align:right;"> 0.92 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> red deer </td>
   <td style="text-align:right;"> 0.00 </td>
   <td style="text-align:right;"> 0.00 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> chamois </td>
   <td style="text-align:right;"> 0.82 </td>
   <td style="text-align:right;"> 0.08 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> hunter </td>
   <td style="text-align:right;"> 0.17 </td>
   <td style="text-align:right;"> 0.11 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> cat </td>
   <td style="text-align:right;"> 0.46 </td>
   <td style="text-align:right;"> 0.59 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> roe deer </td>
   <td style="text-align:right;"> 0.67 </td>
   <td style="text-align:right;"> 0.86 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> dog </td>
   <td style="text-align:right;"> 0.77 </td>
   <td style="text-align:right;"> 0.35 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> human </td>
   <td style="text-align:right;"> 0.51 </td>
   <td style="text-align:right;"> 0.93 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> hare </td>
   <td style="text-align:right;"> 0.37 </td>
   <td style="text-align:right;"> 0.35 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> lynx </td>
   <td style="text-align:right;"> 0.77 </td>
   <td style="text-align:right;"> 0.89 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> marten </td>
   <td style="text-align:right;"> 0.05 </td>
   <td style="text-align:right;"> 0.04 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> fox </td>
   <td style="text-align:right;"> 0.90 </td>
   <td style="text-align:right;"> 0.53 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> wild boar </td>
   <td style="text-align:right;"> 0.75 </td>
   <td style="text-align:right;"> 0.94 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> cow </td>
   <td style="text-align:right;"> 0.01 </td>
   <td style="text-align:right;"> 0.25 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;"> vehicule </td>
   <td style="text-align:right;"> 0.94 </td>
   <td style="text-align:right;"> 0.51 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p>To better understand this pattern, we display the results under the form of a confusion matrix that compares model classifications to manual classifications (<a href="#fig-confusion">Figure&nbsp;2</a>). There were a lot of false negatives for chamois, meaning that when a chamois was present in an image, it was often classified as another species by our model.</p>
<div class="column-page">
<div class="cell">
<div class="cell-output-display">
<div id="fig-confusion" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="published-202204-gimenez-lynx_files/figure-html/fig-confusion-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2: Confusion matrix comparing automatic to manual species classifications. Species that were predicted by our model are in columns, and species that are actually in the images are in rows. Column and row percentages are also provided at the bottom and right side of each cells respectively. An example of column percentage is as follows: of all pictures for which we predict a wild boar, 75.1% actually contained a wild boar. An example of row percentage is as follows: of all pictures in which we have a wild boar, we predict 94% of them to be badgers.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
<p>Overall, our model trained on images from the Jura study site did poorly at correctly predicting species on images from the Ain study site. This result does not come as a surprise, as generalizing classification algorithms to new environments is known to be difficult <span class="citation" data-cites="beery_recognition_2018">(<a href="#ref-beery_recognition_2018" role="doc-biblioref">Beery, Horn, and Perona 2018</a>)</span>. While a computer scientist might be disappointed in these results, an ecologist would probably wonder whether ecological inference about the co-occurrence between lynx and its prey is biased by these average performances, a question we address in the next section.</p>
</section>
</section>
<section id="spatial-co-occurrence" class="level1 page-columns page-full" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Spatial co-occurrence</h1>
<p>Here, we analysed the data we acquired from the previous section. For the sake of comparison, we considered two datasets, one made of the images manually labelled for both the Jura and Ain study sites pooled together (<em>ground truth dataset</em>), and the other in which we pooled the images that were manually labelled for the Jura study site and the images that were automatically labelled for the Ain study site using our trained model (<em>classified dataset</em>).</p>
<p>We formatted the data by generating monthly detection histories, that is a sequence of detections (<span class="math inline">Y_{sit} = 1</span>) and non-detections (<span class="math inline">Y_{sit} = 0</span>), for species <span class="math inline">s</span> at site <span class="math inline">i</span> and sampling occasion <span class="math inline">t</span> (see <a href="#fig-occupancy">Figure&nbsp;3</a>).</p>
<div class="column-page">
<div class="cell">
<div class="cell-output-display">
<div id="fig-occupancy" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="published-202204-gimenez-lynx_files/figure-html/fig-occupancy-1.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;3: Detections (black) and non-detections (light grey) for each of the 3 species lynx, chamois and roe deer between March and November for all years pooled together. Sites are on the Y axis, while sampling occasions are on the X axis. Only data from the ground truth dataset are displayed.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
<p>To quantify spatial co-occurrence betwen lynx and its preys, we used a multispecies occupancy modeling approach <span class="citation" data-cites="Rota2016">(<a href="#ref-Rota2016" role="doc-biblioref">Rota et al. 2016</a>)</span> implemented in the <code>R</code> package <code>unmarked</code> <span class="citation" data-cites="unmarkedFiske">(<a href="#ref-unmarkedFiske" role="doc-biblioref">Fiske and Chandler 2011</a>)</span> within the maximum likelihood framework. The multispecies occupancy model assumes that observations <span class="math inline">y_{sit}</span>, conditional on <span class="math inline">Z_{si}</span> the latent occupancy state of species <span class="math inline">s</span> at site <span class="math inline">i</span> are drawn from Bernoulli random variables <span class="math inline">Y_{sit} | Z_{si} \sim \text{Bernoulli}(Z_{si}p_{sit})</span> where <span class="math inline">p_{sit}</span> is the detection probability of species <span class="math inline">s</span> at site <span class="math inline">i</span> and sampling occasion <span class="math inline">t</span>. Detection probabilities can be modeled as a function of site and/or sampling covariates, or the presence/absence of other species, but for the sake of illustration, we will make them only species-specific here.</p>
<p>The latent occupancy states are assumed to be distributed as multivariate Bernoulli random variables <span class="citation" data-cites="dai_multivariate_2013">(<a href="#ref-dai_multivariate_2013" role="doc-biblioref">Dai, Ding, and Wahba 2013</a>)</span>. Let us consider 2 species, species 1 and 2, then <span class="math inline">Z_i = (Z_{i1}, Z_{i2}) \sim \text{multivariate Bernoulli}(\psi_{11}, \psi_{10}, \psi_{01}, \psi_{00})</span> where <span class="math inline">\psi_{11}</span> is the probability that a site is occupied by both species 1 and 2, <span class="math inline">\psi_{10}</span> the probability that a site is occupied by species 1 but not 2, <span class="math inline">\psi_{01}</span> the probability that a site is occupied by species 2 but not 1, and <span class="math inline">\psi_{00}</span> the probability a site is occupied by none of them. Note that we considered species-specific only occupancy probabilities but these could be modeled as site-specific covariates. Marginal occupancy probabilities are obtained as <span class="math inline">\Pr(Z_{i1}=1) = \psi_{11} + \psi_{10}</span> and <span class="math inline">\Pr(Z_{i2}=1) = \psi_{11} + \psi_{01}</span>. With this model, we may also infer co-occurrence by calculating conditional probabilities such as for example the probability of a site being occupied by species 2 conditional of species 1 with <span class="math inline">\Pr(Z_{i2} = 1| Z_{i1} = 1) = \displaystyle{\frac{\psi_{11}}{\psi_{11}+\psi_{10}}}</span>.</p>
<p>Despite its appeal and increasing use in ecology, multispecies occupancy models can be difficult to fit to real-world data in practice. First, these models are data-hungry and regularization methods <span class="citation" data-cites="clipp2021">(<a href="#ref-clipp2021" role="doc-biblioref">Clipp et al. 2021</a>)</span> are needed to avoid occupancy probabilities to be estimated at the boundary of the parameter space or with large uncertainty. Second, and this is true for any joint species distribution models, these models quickly become very complex with many parameters to be estimated when the number of species increases and co-occurrence is allowed between all species. Here, ecological expertise should be used to consider only meaningful species interactions and apply parsimony when parameterizing models.</p>
<p>We now turn to the results obtained from a model with five species namely lynx, chamois, roe deer, fox and cat and co-occurrence allowed between lynx and chamois and roe deer only.</p>
<p>Detection probabilities were indistinguishable (at the third decimal) whether we used the ground truth or the classified dataset, with <span class="math inline">p_{\text{lynx}} = 0.51 (0.45, 0.58)</span>, <span class="math inline">p_{\text{roe deer}} = 0.63 (0.57, 0.68)</span> and <span class="math inline">p_{\text{chamois}} = 0.61 (0.55, 0.67)</span>.</p>
<p>We also found that occupancy probability estimates were similar whether we used the ground truth or the classified dataset (<a href="#fig-four">Figure&nbsp;4</a>). Roe deer was the most prevalent species, but lynx and chamois were also occurring with high probability (<a href="#fig-four">Figure&nbsp;4</a>). Note that, despite chamois being often misclassified (<a href="#fig-confusion">Figure&nbsp;2</a>), its marginal occupancy tends to be higher when estimated with the classified dataset. Ecologically speaking, this might well be the case if the correctly classified detections are spread over all camera traps. The difference in marginal occupancy seems however non-significant judging by the overlap between the two confidence intervals.</p>
<div class="column-page">
<div class="cell">
<div class="cell-output-display">
<div id="fig-four" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="img/marginal_occ_penalized.png" class="img-fluid figure-img" width="9000"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;4: Marginal occupancy probabilities for all three species (lynx, roe deer and chamois). Parameter estimates are from a multispecies occupancy model using either the ground truth dataset (in red) or the classified dataset (in blue-grey). Note that marginal occupancy probabilities are estimated with high precision for roe deer, which explain why the associated confidence intervals do not show.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
<p>Because marginal occupancy probabilities were high, probabilities of co-occurrence were also estimated high (<a href="#fig-five">Figure&nbsp;5</a>). Our results should be interpreted bearing in mind that co-occurrence is a necessary but not sufficient condition for actual interaction. When both preys were present, lynx was more present than when they were both absent (<a href="#fig-five">Figure&nbsp;5</a>). Lynx was more sensitive to the presence of roe deer than that of chamois (<a href="#fig-five">Figure&nbsp;5</a>).</p>
<div class="column-page">
<div class="cell">
<div class="cell-output-display">
<div id="fig-five" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="img/conditional_occ_penalized.png" class="img-fluid figure-img" width="9000"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;5: Lynx occupancy probability conditional on the presence or absence of its preys (roe deer and chamois). Parameter estimates are from a multispecies occupancy model using either the ground truth dataset (in red) or the classified dataset (in blue-grey).</figcaption><p></p>
</figure>
</div>
</div>
</div>
</div>
<p>Overall, we found similar or higher uncertainty in estimates obtained from the classified dataset (<a href="#fig-four">Figure&nbsp;4</a> and <a href="#fig-five">Figure&nbsp;5</a>). Sample size being similar for both datasets, we do not have a solid explanation for this pattern.</p>
</section>
<section id="discussion" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Discussion</h1>
<p>In this paper, we aimed at illustrating a reproducible workflow for studying the structure of an animal community and species spatial co-occurrence (<em>why</em>) using images acquired from camera traps and automatically labelled with deep learning (<em>what</em>) which we analysed with statistical occupancy models accounting for imperfect species detection (<em>how</em>). Overall, we found that, even though model transferability could be improved, inference about the co-occurrence of lynx and its preys was similar whether we analysed the ground truth data or classified data.</p>
<p>This result calls for further work on the trade-offs between time and resources allocated to train models with deep learning and our ability to correctly answer key ecological questions with camera-trap surveys. In other words, while a computer scientist might be keen on spending time training models to achieve top performances, an ecologist would rather rely on a model showing average performances and use this time to proceed with statistical analyses if, of course, errors in computer-annotated images do not make ecological inference flawed. The right balance may be found with collaborative projects in which scientists from artificial intelligence, statistics and ecology agree on a common objective, and identify research questions that can pick the interest of all parties.</p>
<p>Our demonstration remains however empirical, and ecological inference might no longer be robust to misclassification if detection and non-detections were pooled weekly or daily, or if more complex models, e.g.&nbsp;including time-varying detection probabilities and/or habitat-specific occupancy probabilities, were fitted to the data. Therefore, we encourage others to try and replicate our results. In that spirit, we praise previous work on plants which used deep learning to produce occurrence data and tested the sensitivity of species distribution models to image classification errors <span class="citation" data-cites="botella2018">(<a href="#ref-botella2018" role="doc-biblioref">Botella et al. 2018</a>)</span>. We also see two avenues of research that could benefit the integration of deep learning and ecological statistics. First, a simulation study could be conducted to evaluate bias and precision in ecological parameter estimators with regard to errors in image annotation by computers. The outcome of this exercise could be, for example, guidelines informing on the confidence an investigator may place in ecological inference as a function of the amount of false negatives and false positives. Second, annotation errors could be accomodated directly in statistical models. For example, single-species occupancy models account for false negatives when a species is not detected by the camera at a site where it is present, as well as false positives when a species is detected at a site where it is not present due to species misidentification by the observer <span class="citation" data-cites="miller2011">(<a href="#ref-miller2011" role="doc-biblioref">Miller et al. 2011</a>)</span>. Pending a careful distinction between ecological vs.&nbsp;computer-generated false negatives and false positives, error rates could be added to multispecies occupancy models <span class="citation" data-cites="chambert2018">(<a href="#ref-chambert2018" role="doc-biblioref">Chambert et al. 2018</a>)</span> and informed by recall and precision metrics obtained during model training <span class="citation" data-cites="tabak_improving_2020">(<a href="#ref-tabak_improving_2020" role="doc-biblioref">Tabak et al. 2020</a>)</span>. An alternative quick and dirty approach would consist in adopting a Monte Carlo approach by sampling the species detected or non-detected in each picture according to its predicted probability of belonging to a given class, then building the corresponding dataset and fitting occupancy models to it for each sample.</p>
<p>When it comes to the case study, our results should be discussed with regard to the sensitivity of co-occurrence estimates to errors in automatic species classification. In particular, we expected that confusions between the two prey species might artificially increase the estimated probability of co-occurrence with lynx. This was illustrated by <span class="math inline">\Pr(\text{lynx present} | \text{roe deer present and chamois absent})</span> (resp. <span class="math inline">\Pr(\text{lynx present} | \text{roe deer absent and chamois present})</span>) being estimated higher (resp. lower) with the classified than the ground truth dataset (<a href="#fig-five">Figure&nbsp;5</a>). This pattern could be explained by chamois being often classified as (and confused with) roe deer (<a href="#fig-confusion">Figure&nbsp;2</a>).</p>
<p>Our results are only preliminary and we see several perspectives to our work. First, we focused our analysis on lynx and its main prey, while other species should be included to get a better understanding of the community structure. For example, both lynx and fox prey on small rodents and birds and a model including co-occurrence between these two predators showed better support by the data (AIC was 1544 when co-occurrence was included vs.&nbsp;1557 when it was not). Second, we aim at quantifying the relative contribution of biotic (lynx predation on chamois and roe deer) and abiotic (habitat quality) processes to the composition and dynamic of this ecological community. Third, to benefit future camera trap studies of lynx in the Jura mountains, we plan to train a model again using more manually annotated images from both the Jura and the Ain study sites. These perspectives are the object of ongoing work.</p>
<p>With the rapid advances in technologies for biodiversity monitoring <span class="citation" data-cites="lahoz-monfort_comprehensive_2021">(<a href="#ref-lahoz-monfort_comprehensive_2021" role="doc-biblioref">Lahoz-Monfort and Magrath 2021</a>)</span>, the possibility of analysing large amounts of images makes deep learning appealing to ecologists. We hope that our proposal of a reproducible <code>R</code> workflow for deep learning and statistical ecology will encourage further studies in the integration of these disciplines, and contribute to the adoption of computer vision by ecologists.</p>
</section>
<section id="appendix-reproducible-example-of-species-identification-on-camera-trap-images-with-cpu" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Appendix: Reproducible example of species identification on camera trap images with CPU</h1>
<p>In this section, we go through a reproducible example of the entire deep learning workflow, including data preparation, model training, and automatic labeling of new images. We used a subsample of 467 images from the original dataset in the Jura county to allow the training of our model with CPU on a personal computer. We also used 14 images from the original dataset in the Ain county to illustrate prediction.</p>
<section id="training-and-validation-datasets" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="training-and-validation-datasets"><span class="header-section-number">6.1</span> Training and validation datasets</h2>
<p>We first split the dataset of Jura images in two datasets, a dataset for training, and the other one for validation. We use the <code>exifr</code> package to extract metadata from images, get a list of images names and extract the species from these.</p>
<div>
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(exifr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>pix_folder <span class="ot">&lt;-</span> <span class="st">'pix/pixJura/'</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>file_list <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> pix_folder,</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">recursive =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pattern =</span> <span class="st">"*.jpg"</span>,</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>labels <span class="ot">&lt;-</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_exif</span>(file_list) <span class="sc">%&gt;%</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(Keywords, <span class="at">keep_empty =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="co"># keep_empty = TRUE keeps pix with no labels (empty pix)</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SourceFile) <span class="sc">%&gt;%</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>() <span class="sc">%&gt;%</span> <span class="co"># when several labels in a pix, keep first only</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> <span class="fu">as_factor</span>(Keywords)) <span class="sc">%&gt;%</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> <span class="fu">fct_explicit_na</span>(Keywords, <span class="st">"wo_tag"</span>)) <span class="sc">%&gt;%</span> <span class="co"># when pix has no tag</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SourceFile, FileName, Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> <span class="fu">fct_recode</span>(Keywords,</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"chat"</span> <span class="ot">=</span> <span class="st">"chat forestier"</span>,</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"lievre"</span> <span class="ot">=</span> <span class="st">"lièvre"</span>,</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"vehicule"</span> <span class="ot">=</span> <span class="st">"véhicule"</span>,</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"ni"</span> <span class="ot">=</span> <span class="st">"Non identifié"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(Keywords <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ni"</span>, <span class="st">"wo_tag"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="tbl-datasets" class="cell tbl-parent quarto-layout-panel anchored" data-tbl-cap="Species considered, and number of images with these species in them.">
<div class="panel-caption table-caption">
<p><strong>?(caption)</strong></p>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">

<table class="table" style="margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> Keywords </th>
   <th style="text-align:right;"> n </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> humain </td>
   <td style="text-align:right;"> 143 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> vehicule </td>
   <td style="text-align:right;"> 135 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> renard </td>
   <td style="text-align:right;"> 58 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> sangliers </td>
   <td style="text-align:right;"> 33 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> chasseur </td>
   <td style="text-align:right;"> 17 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> chien </td>
   <td style="text-align:right;"> 14 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> lynx </td>
   <td style="text-align:right;"> 13 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> chevreuil </td>
   <td style="text-align:right;"> 13 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> chamois </td>
   <td style="text-align:right;"> 12 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> blaireaux </td>
   <td style="text-align:right;"> 10 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> chat </td>
   <td style="text-align:right;"> 8 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> lievre </td>
   <td style="text-align:right;"> 4 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> fouine </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> cavalier </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<p>Then we pick 80<span class="math inline">\%</span> of the images for training in each category, the rest being used for validation.</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># training dataset</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>pix_train <span class="ot">&lt;-</span> labels <span class="sc">%&gt;%</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SourceFile, FileName, Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(<span class="fu">row_number</span>(), <span class="dv">1</span>, <span class="fu">floor</span>(<span class="fu">n</span>()<span class="sc">*</span><span class="dv">80</span><span class="sc">/</span><span class="dv">100</span>))) <span class="co"># 80% per category</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># validation dataset</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>pix_valid <span class="ot">&lt;-</span> labels <span class="sc">%&gt;%</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(<span class="fu">row_number</span>(), <span class="fu">floor</span>(<span class="fu">n</span>()<span class="sc">*</span><span class="dv">80</span><span class="sc">/</span><span class="dv">100</span>) <span class="sc">+</span> <span class="dv">1</span>, <span class="fu">n</span>()))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Eventually, we store these images in two distinct directories named <code>train</code> and <code>valid</code>.</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create dir train/ and copy pix there, organised by categories</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">'pix/train'</span>) <span class="co"># create training directory</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">levels</span>(<span class="fu">fct_drop</span>(pix_train<span class="sc">$</span>Keywords))) <span class="fu">dir.create</span>(<span class="fu">paste0</span>(<span class="st">'pix/train/'</span>,i)) <span class="co"># create dir for labels</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pix_train)){</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">file.copy</span>(<span class="fu">as.character</span>(pix_train<span class="sc">$</span>SourceFile[i]),</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>              <span class="fu">paste0</span>(<span class="st">'pix/train/'</span>, <span class="fu">as.character</span>(pix_train<span class="sc">$</span>Keywords[i]))) <span class="co"># copy pix in corresp dir</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># create dir valid/ and copy pix there, organised by categories.</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">'pix/valid'</span>) <span class="co"># create validation dir</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">levels</span>(<span class="fu">fct_drop</span>(pix_train<span class="sc">$</span>Keywords))) <span class="fu">dir.create</span>(<span class="fu">paste0</span>(<span class="st">'pix/valid/'</span>,i)) <span class="co"># create dir for labels</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pix_valid)){</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">file.copy</span>(<span class="fu">as.character</span>(pix_valid<span class="sc">$</span>SourceFile[i]),</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>              <span class="fu">paste0</span>(<span class="st">'pix/valid/'</span>, <span class="fu">as.character</span>(pix_valid<span class="sc">$</span>Keywords[i]))) <span class="co"># copy pix in corresp dir</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="co"># delete pictures in valid/ directory for which we did not train the model</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>to_be_deleted <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">levels</span>(<span class="fu">fct_drop</span>(pix_valid<span class="sc">$</span>Keywords)), <span class="fu">levels</span>(<span class="fu">fct_drop</span>(pix_train<span class="sc">$</span>Keywords)))</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is_empty</span>(to_be_deleted)) {</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(to_be_deleted)){</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unlink</span>(<span class="fu">paste0</span>(<span class="st">'pix/valid/'</span>, to_be_deleted[i]))</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>What is the sample size of these two datasets?</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(<span class="st">"training"</span> <span class="ot">=</span> pix_train, <span class="st">"validation"</span> <span class="ot">=</span> pix_valid, <span class="at">.id =</span> <span class="st">"dataset"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">category =</span> Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-samplesize" class="anchored">

<table class="table" style="margin-left: auto; margin-right: auto;"><caption>Table&nbsp;4:  Sample size (n) for the training and validation datasets. </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> dataset </th>
   <th style="text-align:left;"> category </th>
   <th style="text-align:right;"> n </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> humain </td>
   <td style="text-align:right;"> 114 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> vehicule </td>
   <td style="text-align:right;"> 108 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> chamois </td>
   <td style="text-align:right;"> 9 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> blaireaux </td>
   <td style="text-align:right;"> 8 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> sangliers </td>
   <td style="text-align:right;"> 26 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> renard </td>
   <td style="text-align:right;"> 46 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> chasseur </td>
   <td style="text-align:right;"> 13 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> lynx </td>
   <td style="text-align:right;"> 10 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> chien </td>
   <td style="text-align:right;"> 11 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> chat </td>
   <td style="text-align:right;"> 6 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> chevreuil </td>
   <td style="text-align:right;"> 10 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> training </td>
   <td style="text-align:left;"> lievre </td>
   <td style="text-align:right;"> 3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> humain </td>
   <td style="text-align:right;"> 29 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> vehicule </td>
   <td style="text-align:right;"> 27 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> chamois </td>
   <td style="text-align:right;"> 3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> blaireaux </td>
   <td style="text-align:right;"> 2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> sangliers </td>
   <td style="text-align:right;"> 7 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> renard </td>
   <td style="text-align:right;"> 12 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> chasseur </td>
   <td style="text-align:right;"> 4 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> lynx </td>
   <td style="text-align:right;"> 3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> chien </td>
   <td style="text-align:right;"> 3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> fouine </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> chat </td>
   <td style="text-align:right;"> 2 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> chevreuil </td>
   <td style="text-align:right;"> 3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> lievre </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> validation </td>
   <td style="text-align:left;"> cavalier </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="transfer-learning" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="transfer-learning"><span class="header-section-number">6.2</span> Transfer learning</h2>
<p>We proceed with transfer learning using images from the Jura county (or a subsample more exactly). We first load images and apply standard transformations to improve training (flip, rotate, zoom, rotate, ligth transform).</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#reticulate::use_condaenv("computo")</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastai)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>dls <span class="ot">&lt;-</span> <span class="fu">ImageDataLoaders_from_folder</span>(</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">path =</span> <span class="st">"pix/"</span>,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">train =</span> <span class="st">"train"</span>,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">valid =</span> <span class="st">"valid"</span>,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">item_tfms =</span> <span class="fu">Resize</span>(<span class="at">size =</span> <span class="dv">460</span>),</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">bs =</span> <span class="dv">10</span>,</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">batch_tfms =</span> <span class="fu">list</span>(<span class="fu">aug_transforms</span>(<span class="at">size =</span> <span class="dv">224</span>,</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">min_scale =</span> <span class="fl">0.75</span>), <span class="co"># transformation</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">Normalize_from_stats</span>( <span class="fu">imagenet_stats</span>() )),</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">num_workers =</span> <span class="dv">0</span>,</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">ImageFile.LOAD_TRUNCATED_IMAGES =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Then we get the model architecture. For the sake of illustration, we use a resnet18 here, but we used a resnet50 to get the full results presented in the main text.</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>learn <span class="ot">&lt;-</span> <span class="fu">cnn_learner</span>(<span class="at">dls =</span> dls,</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">arch =</span> <span class="fu">resnet18</span>(),</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">metrics =</span> <span class="fu">list</span>(accuracy, error_rate))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we are ready to train our model. Again, for the sake of illustration, we use only 2 epochs here, but used 20 epochs to get the full results presented in the main text. With all pictures and a resnet50, it took 75 minutes per epoch approximatively on a Mac with a 2.4Ghz processor and 64Go memory, and less than half an hour on a machine with GPU. On this reduced dataset, it took a bit more than a minute per epoch on the same Mac. Note that we save the model after each epoch for later use.</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>one_cycle <span class="ot">&lt;-</span> learn <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fit_one_cycle</span>(<span class="dv">2</span>, <span class="at">cbs =</span> <span class="fu">SaveModelCallback</span>(<span class="at">every_epoch =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">fname =</span> <span class="st">'model'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>epoch   train_loss   valid_loss   accuracy   error_rate   time 
------  -----------  -----------  ---------  -----------  -----
Epoch   1/2          :            |----------------------------------------|   0.00%        [0/36 
Epoch   1/2          :            |---------------------------------------|   2.78%        [1/36 
Epoch   1/2          :            |--------------------------------------|   5.56%        [2/36 
Epoch   1/2          :            |-------------------------------------|   8.33%        [3/36 
Epoch   1/2          :            |------------------------------------|   11.11%       [4/36 
Epoch   1/2          :            |-----------------------------------|   13.89%       [5/36 
Epoch   1/2          :            |----------------------------------|   16.67%       [6/36 
Epoch   1/2          :            |---------------------------------|   19.44%       [7/36 
Epoch   1/2          :            |--------------------------------|   22.22%       [8/36 
Epoch   1/2          :            |------------------------------|   25.00%       [9/36 
Epoch   1/2          :            |-----------------------------|   27.78%       [10/36 
Epoch   1/2          :            |----------------------------|   30.56%       [11/36 
Epoch   1/2          :            |---------------------------|   33.33%       [12/36 
Epoch   1/2          :            |--------------------------|   36.11%       [13/36 
Epoch   1/2          :            |-------------------------|   38.89%       [14/36 
Epoch   1/2          :            |------------------------|   41.67%       [15/36 
Epoch   1/2          :            |-----------------------|   44.44%       [16/36 
Epoch   1/2          :            |----------------------|   47.22%       [17/36 
Epoch   1/2          :            |--------------------|   50.00%       [18/36 
Epoch   1/2          :            |-------------------|   52.78%       [19/36 
Epoch   1/2          :            |------------------|   55.56%       [20/36 
Epoch   1/2          :            |-----------------|   58.33%       [21/36 
Epoch   1/2          :            |----------------|   61.11%       [22/36 
Epoch   1/2          :            |---------------|   63.89%       [23/36 
Epoch   1/2          :            |--------------|   66.67%       [24/36 
Epoch   1/2          :            |-------------|   69.44%       [25/36 
Epoch   1/2          :            |------------|   72.22%       [26/36 
Epoch   1/2          :            |----------|   75.00%       [27/36 
Epoch   1/2          :            |---------|   77.78%       [28/36 
Epoch   1/2          :            |--------|   80.56%       [29/36 
Epoch   1/2          :            |-------|   83.33%       [30/36 
Epoch   1/2          :            |------|   86.11%       [31/36 
Epoch   1/2          :            |-----|    88.89%       [32/36 
Epoch   1/2          :            |----|     91.67%       [33/36 
Epoch   1/2          :            |---|      94.44%       [34/36 
Epoch   1/2          :            |--|       97.22%       [35/36 
Epoch   1/2          :            ||         100.00%      [36/36 
Epoch   1/2          :            |----------------------------------------|   0.00%        [0/10 
Epoch   1/2          :            |------------------------------------|   10.00%       [1/10 
Epoch   1/2          :            |--------------------------------|   20.00%       [2/10 
Epoch   1/2          :            |----------------------------|   30.00%       [3/10 
Epoch   1/2          :            |------------------------|   40.00%       [4/10 
Epoch   1/2          :            |--------------------|   50.00%       [5/10 
Epoch   1/2          :            |----------------|   60.00%       [6/10 
Epoch   1/2          :            |------------|   70.00%       [7/10 
Epoch   1/2          :            |--------|   80.00%       [8/10 
Epoch   1/2          :            |----|     90.00%       [9/10 
Epoch   1/2          :            ||         100.00%      [10/10 
0       2.622631     0.955220     0.750000   0.250000     00:47 
Epoch   2/2          :            |----------------------------------------|   0.00%        [0/36 
Epoch   2/2          :            |---------------------------------------|   2.78%        [1/36 
Epoch   2/2          :            |--------------------------------------|   5.56%        [2/36 
Epoch   2/2          :            |-------------------------------------|   8.33%        [3/36 
Epoch   2/2          :            |------------------------------------|   11.11%       [4/36 
Epoch   2/2          :            |-----------------------------------|   13.89%       [5/36 
Epoch   2/2          :            |----------------------------------|   16.67%       [6/36 
Epoch   2/2          :            |---------------------------------|   19.44%       [7/36 
Epoch   2/2          :            |--------------------------------|   22.22%       [8/36 
Epoch   2/2          :            |------------------------------|   25.00%       [9/36 
Epoch   2/2          :            |-----------------------------|   27.78%       [10/36 
Epoch   2/2          :            |----------------------------|   30.56%       [11/36 
Epoch   2/2          :            |---------------------------|   33.33%       [12/36 
Epoch   2/2          :            |--------------------------|   36.11%       [13/36 
Epoch   2/2          :            |-------------------------|   38.89%       [14/36 
Epoch   2/2          :            |------------------------|   41.67%       [15/36 
Epoch   2/2          :            |-----------------------|   44.44%       [16/36 
Epoch   2/2          :            |----------------------|   47.22%       [17/36 
Epoch   2/2          :            |--------------------|   50.00%       [18/36 
Epoch   2/2          :            |-------------------|   52.78%       [19/36 
Epoch   2/2          :            |------------------|   55.56%       [20/36 
Epoch   2/2          :            |-----------------|   58.33%       [21/36 
Epoch   2/2          :            |----------------|   61.11%       [22/36 
Epoch   2/2          :            |---------------|   63.89%       [23/36 
Epoch   2/2          :            |--------------|   66.67%       [24/36 
Epoch   2/2          :            |-------------|   69.44%       [25/36 
Epoch   2/2          :            |------------|   72.22%       [26/36 
Epoch   2/2          :            |----------|   75.00%       [27/36 
Epoch   2/2          :            |---------|   77.78%       [28/36 
Epoch   2/2          :            |--------|   80.56%       [29/36 
Epoch   2/2          :            |-------|   83.33%       [30/36 
Epoch   2/2          :            |------|   86.11%       [31/36 
Epoch   2/2          :            |-----|    88.89%       [32/36 
Epoch   2/2          :            |----|     91.67%       [33/36 
Epoch   2/2          :            |---|      94.44%       [34/36 
Epoch   2/2          :            |--|       97.22%       [35/36 
Epoch   2/2          :            ||         100.00%      [36/36 
Epoch   2/2          :            |----------------------------------------|   0.00%        [0/10 
Epoch   2/2          :            |------------------------------------|   10.00%       [1/10 
Epoch   2/2          :            |--------------------------------|   20.00%       [2/10 
Epoch   2/2          :            |----------------------------|   30.00%       [3/10 
Epoch   2/2          :            |------------------------|   40.00%       [4/10 
Epoch   2/2          :            |--------------------|   50.00%       [5/10 
Epoch   2/2          :            |----------------|   60.00%       [6/10 
Epoch   2/2          :            |------------|   70.00%       [7/10 
Epoch   2/2          :            |--------|   80.00%       [8/10 
Epoch   2/2          :            |----|     90.00%       [9/10 
Epoch   2/2          :            ||         100.00%      [10/10 
1       1.792528     0.849985     0.770833   0.229167     00:46 </code></pre>
</div>
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>one_cycle</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  epoch train_loss valid_loss  accuracy error_rate
1     0   2.622631  0.9552196 0.7500000  0.2500000
2     1   1.792528  0.8499851 0.7708333  0.2291667</code></pre>
</div>
</div>
<p>We may dig a bit deeper in training performances by loading the best model, here <code>model_1.pth</code>, and display some metrics for each species.</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>learn<span class="sc">$</span><span class="fu">load</span>(<span class="st">"model_1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Sequential(
  (0): Sequential(
    (0): Conv2d(3, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False)
    (1): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): ReLU(inplace=True)
    (3): MaxPool2d(kernel_size=3, stride=2, padding=1, dilation=1, ceil_mode=False)
    (4): Sequential(
      (0): BasicBlock(
        (conv1): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn1): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
      (1): BasicBlock(
        (conv1): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn1): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
    )
    (5): Sequential(
      (0): BasicBlock(
        (conv1): Conv2d(64, 128, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
        (bn1): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (conv2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (downsample): Sequential(
          (0): Conv2d(64, 128, kernel_size=(1, 1), stride=(2, 2), bias=False)
          (1): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
      (1): BasicBlock(
        (conv1): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn1): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (conv2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(128, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
    )
    (6): Sequential(
      (0): BasicBlock(
        (conv1): Conv2d(128, 256, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
        (bn1): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (conv2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (downsample): Sequential(
          (0): Conv2d(128, 256, kernel_size=(1, 1), stride=(2, 2), bias=False)
          (1): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
      (1): BasicBlock(
        (conv1): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn1): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (conv2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
    )
    (7): Sequential(
      (0): BasicBlock(
        (conv1): Conv2d(256, 512, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
        (bn1): BatchNorm2d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (conv2): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (downsample): Sequential(
          (0): Conv2d(256, 512, kernel_size=(1, 1), stride=(2, 2), bias=False)
          (1): BatchNorm2d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        )
      )
      (1): BasicBlock(
        (conv1): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn1): BatchNorm2d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (relu): ReLU(inplace=True)
        (conv2): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn2): BatchNorm2d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
    )
  )
  (1): Sequential(
    (0): AdaptiveConcatPool2d(
      (ap): AdaptiveAvgPool2d(output_size=1)
      (mp): AdaptiveMaxPool2d(output_size=1)
    )
    (1): fastai.layers.Flatten(full=False)
    (2): BatchNorm1d(1024, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (3): Dropout(p=0.25, inplace=False)
    (4): Linear(in_features=1024, out_features=512, bias=False)
    (5): ReLU(inplace=True)
    (6): BatchNorm1d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (7): Dropout(p=0.5, inplace=False)
    (8): Linear(in_features=512, out_features=12, bias=False)
  )
)</code></pre>
</div>
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>interp <span class="ot">&lt;-</span> <span class="fu">ClassificationInterpretation_from_learner</span>(learn)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>interp<span class="sc">$</span><span class="fu">print_classification_report</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We may extract the categories that get the most confused.</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>interp <span class="sc">%&gt;%</span> <span class="fu">most_confused</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>          V1        V2 V3
1   chasseur    humain  3
2     humain  vehicule  3
3       lynx    renard  3
4    chamois chevreuil  2
5       chat    renard  2
6  chevreuil    renard  2
7      chien sangliers  2
8  blaireaux sangliers  1
9  chevreuil      chat  1
10     chien    renard  1
11    lievre    renard  1
12    renard sangliers  1</code></pre>
</div>
</div>
</section>
<section id="transferability" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="transferability"><span class="header-section-number">6.3</span> Transferability</h2>
<p>In this section, we show how to use our freshly trained model to label images that were taken in another study site in the Ain county, and not used to train our model. First, we get the path to the images.</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>fls <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">"pix/pixAin"</span>,</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">full.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Then we carry out prediction, and compare to the truth.</p>
<div class="cell">
<details open="">
<summary>Hide/Show the code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>predicted <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="dv">3</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>categories <span class="ot">&lt;-</span> interp<span class="sc">$</span>vocab <span class="sc">%&gt;%</span> <span class="fu">as.character</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace_all</span>(<span class="st">"[[:punct:]]"</span>, <span class="st">" "</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_trim</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_split</span>(<span class="st">"   "</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>()</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fls)){</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> learn <span class="sc">%&gt;%</span> <span class="fu">predict</span>(fls[i]) <span class="co"># make prediction</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  result[[<span class="dv">3</span>]] <span class="sc">%&gt;%</span> <span class="fu">as.character</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_extract</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.integer</span>() <span class="ot">-&gt;</span> index <span class="co"># extract relevant info</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  predicted[i] <span class="ot">&lt;-</span> categories[index <span class="sc">+</span> <span class="dv">1</span>] <span class="co"># match it with categories</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">truth =</span> <span class="fu">c</span>(<span class="st">"lynx"</span>, <span class="st">"roe deer"</span>, <span class="st">"wild boar"</span>),</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>           <span class="at">prediction =</span> predicted) <span class="sc">%&gt;%</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-prediction" class="anchored">

<table class="table" style="margin-left: auto; margin-right: auto;"><caption>Table&nbsp;5:  Comparison of the predictions vs.&nbsp;ground truth. </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> truth </th>
   <th style="text-align:left;"> prediction </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> lynx </td>
   <td style="text-align:left;"> renard </td>
  </tr>
  <tr>
   <td style="text-align:left;"> roe deer </td>
   <td style="text-align:left;"> humain </td>
  </tr>
  <tr>
   <td style="text-align:left;"> wild boar </td>
   <td style="text-align:left;"> sangliers </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
</section>


<section id="bibliography" class="level1 unnumbered">


</section>

<div id="quarto-appendix" class="default"><section id="session-information" class="level1 appendix unnumbered"><h2 class="quarto-appendix-heading">Session information</h2><div class="quarto-appendix-contents">

<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] reticulate_1.28   exifr_0.3.2       unmarked_1.2.5    cvms_1.3.9       
 [5] janitor_2.1.0     highcharter_0.9.4 fastai_2.2.0      ggtext_0.1.2     
 [9] wesanderson_0.3.6 kableExtra_1.3.4  stringi_1.7.12    lubridate_1.9.0  
[13] timechange_0.2.0  cowplot_1.1.1     sf_1.0-9          forcats_0.5.2    
[17] stringr_1.5.0     dplyr_1.1.0       purrr_1.0.1       readr_2.1.3      
[21] tidyr_1.3.0       tibble_3.1.8      ggplot2_3.4.0     tidyverse_1.3.2  

loaded via a namespace (and not attached):
  [1] minqa_1.2.5         googledrive_2.0.0   colorspace_2.1-0   
  [4] ggsignif_0.6.4      ellipsis_0.3.2      class_7.3-21       
  [7] snakecase_0.11.0    markdown_1.4        fs_1.6.0           
 [10] gridtext_0.1.5      rstudioapi_0.14     proxy_0.4-27       
 [13] ggpubr_0.5.0        farver_2.1.1        bit64_4.0.5        
 [16] fansi_1.0.4         xml2_1.3.3          splines_4.2.2      
 [19] knitr_1.41          rlist_0.4.6.2       jsonlite_1.8.4     
 [22] nloptr_2.0.3        broom_1.0.3         dbplyr_2.3.0       
 [25] png_0.1-8           compiler_4.2.2      httr_1.4.4         
 [28] backports_1.4.1     assertthat_0.2.1    Matrix_1.5-3       
 [31] fastmap_1.1.0       gargle_1.2.1        cli_3.6.0          
 [34] htmltools_0.5.4     tools_4.2.2         igraph_1.3.5       
 [37] gtable_0.3.1        glue_1.6.2          rappdirs_0.3.3     
 [40] Rcpp_1.0.10         carData_3.0-5       cellranger_1.1.0   
 [43] vctrs_0.5.2         nlme_3.1-161        svglite_2.1.1      
 [46] xfun_0.36           lme4_1.1-31         rvest_1.0.3        
 [49] lifecycle_1.0.3     renv_0.16.0         rstatix_0.7.1      
 [52] googlesheets4_1.0.1 MASS_7.3-58.2       zoo_1.8-11         
 [55] scales_1.2.1        vroom_1.6.0         hms_1.1.2          
 [58] parallel_4.2.2      RColorBrewer_1.1-3  yaml_2.3.6         
 [61] quantmod_0.4.20     curl_5.0.0          pbapply_1.7-0      
 [64] highr_0.10          e1071_1.7-12        checkmate_2.1.0    
 [67] TTR_0.24.3          boot_1.3-28.1       commonmark_1.8.1   
 [70] rlang_1.0.6         pkgconfig_2.0.3     systemfonts_1.0.4  
 [73] evaluate_0.20       lattice_0.20-45     labeling_0.4.2     
 [76] htmlwidgets_1.6.1   bit_4.0.5           tidyselect_1.2.0   
 [79] plyr_1.8.8          magrittr_2.0.3      R6_2.5.1           
 [82] generics_0.1.3      DBI_1.1.3           pillar_1.8.1       
 [85] haven_2.5.1         withr_2.5.0         units_0.8-1        
 [88] xts_0.12.2          abind_1.4-5         car_3.1-1          
 [91] modelr_0.1.10       crayon_1.5.2        KernSmooth_2.23-20 
 [94] utf8_1.2.2          tzdb_0.3.0          rmarkdown_2.19     
 [97] jpeg_0.1-10         grid_4.2.2          readxl_1.4.1       
[100] data.table_1.14.6   reprex_2.0.2        digest_0.6.31      
[103] classInt_0.4-8      webshot_0.5.4       munsell_0.5.0      
[106] viridisLite_0.4.1  </code></pre>
</div>
</div>
</div></section><section id="acknowledgments" class="level1 appendix unnumbered"><h2 class="quarto-appendix-heading">Acknowledgments</h2><div class="quarto-appendix-contents">

<p>We warmly thank Mathieu Massaviol, Remy Dernat and Khalid Belkhir for their help in using GPU machines on the Montpellier Bioinformatics Biodiversity platform, Julien Renoult for helpful discussions, Delphine Dinouart and Chloé Quillard for their precious help in manually tagging the images, and Vincent Miele for having inspired this work, and his help and support along the way. We also thank the staff of the Federations of Hunters from the Jura and Ain counties, hunters who helped to find locations for camera traps and volunteers who contributed in collecting data. Our thanks also go to Hannah Clipp, Chris Rota and Ken Kellner for sharing a development version of unmarked, and an unpublished version of their paper. The Lynx Predator Prey Program was funded by Auvergne-Rhône-Alpes Region, Ain and Jura departmental Councils, The French National Federation of Hunters, French Environmental Ministry based in Auvergne-Rhone-Alpes and Bourgogne Franche-Comté Region and the French Office for Biodiversity. Our work was also partly funded by the French National Research Agency (grant ANR-16-CE02-0007).</p>
<!-- -->

</div></section><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">Bibliography</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-reticulate_ref" class="csl-entry" role="doc-biblioentry">
Allaire, JJ, Kevin Ushey, Yuan Tang, and Dirk Eddelbuettel. 2017. <em>Reticulate: R Interface to Python</em>. <a href="https://github.com/rstudio/reticulate">https://github.com/rstudio/reticulate</a>.
</div>
<div id="ref-baraniuk_science_2020" class="csl-entry" role="doc-biblioentry">
Baraniuk, Richard, David Donoho, and Matan Gavish. 2020. <span>“The Science of Deep Learning.”</span> <em>Proceedings of the National Academy of Sciences</em> 117 (48): 30029–32. <a href="https://doi.org/10.1073/pnas.2020596117">https://doi.org/10.1073/pnas.2020596117</a>.
</div>
<div id="ref-beery_recognition_2018" class="csl-entry" role="doc-biblioentry">
Beery, Sara, Grant van Horn, and Pietro Perona. 2018. <span>“Recognition in <span>Terra</span> <span>Incognita</span>.”</span> <em>arXiv:1807.04975</em>. <a href="http://arxiv.org/abs/1807.04975">http://arxiv.org/abs/1807.04975</a>.
</div>
<div id="ref-botella2018" class="csl-entry" role="doc-biblioentry">
Botella, Christophe, Alexis Joly, Pierre Bonnet, Pascal Monestiez, and François Munoz. 2018. <span>“Species Distribution Modeling Based on the Automated Identification of Citizen Observations.”</span> <em>Applications in Plant Sciences</em> 6 (2): e1029.
</div>
<div id="ref-breitenmoser_large_1998" class="csl-entry" role="doc-biblioentry">
Breitenmoser, Urs. 1998. <span>“Large Predators in the <span>Alps</span>: <span>The</span> Fall and Rise of Man’s Competitors.”</span> <em>Biological Conservation</em>, Conservation <span>Biology</span> and <span>Biodiversity</span> <span>Strategies</span>, 83 (3): 279–89. <a href="https://doi.org/10.1016/S0006-3207(97)00084-0">https://doi.org/10.1016/S0006-3207(97)00084-0</a>.
</div>
<div id="ref-chambert2018" class="csl-entry" role="doc-biblioentry">
Chambert, Thierry, Evan H. Campbell Grant, David A. W. Miller, James D. Nichols, Kevin P. Mulder, and Adrianne B. Brand. 2018. <span>“Two-Species Occupancy Modelling Accounting for Species Misidentification and Non-Detection.”</span> <em>Methods in Ecology and Evolution</em> 9 (6): 1468–77. https://doi.org/<a href="https://doi.org/10.1111/2041-210X.12985">https://doi.org/10.1111/2041-210X.12985</a>.
</div>
<div id="ref-christin_applications_2019" class="csl-entry" role="doc-biblioentry">
Christin, Sylvain, Éric Hervet, and Nicolas Lecomte. 2019. <span>“Applications for Deep Learning in Ecology.”</span> Edited by Hao Ye. <em>Methods in Ecology and Evolution</em> 10 (10): 1632–44. <a href="https://doi.org/10.1111/2041-210X.13256">https://doi.org/10.1111/2041-210X.13256</a>.
</div>
<div id="ref-clipp2021" class="csl-entry" role="doc-biblioentry">
Clipp, Hannah L., Amber L. Evans, Brin E. Kessinger, K. Kellner, and Christopher T. Rota. 2021. <span>“A Penalized Likelihood for Multi-Species Occupancy Models Improves Predictions of Species Interactions.”</span> <em>Ecology</em>.
</div>
<div id="ref-dai_multivariate_2013" class="csl-entry" role="doc-biblioentry">
Dai, Bin, Shilin Ding, and Grace Wahba. 2013. <span>“Multivariate <span>Bernoulli</span> Distribution.”</span> <em>Bernoulli</em> 19 (4). <a href="https://doi.org/10.3150/12-BEJSP10">https://doi.org/10.3150/12-BEJSP10</a>.
</div>
<div id="ref-Duggan2021" class="csl-entry" role="doc-biblioentry">
Duggan, Matthew T., Melissa F. Groleau, Ethan P. Shealy, Lillian S. Self, Taylor E. Utter, Matthew M. Waller, Bryan C. Hall, Chris G. Stone, Layne L. Anderson, and Timothy A. Mousseau. 2021. <span>“An Approach to Rapid Processing of Camera Trap Images with Minimal Human Input.”</span> <em>Ecology and Evolution</em>. https://doi.org/<a href="https://doi.org/10.1002/ece3.7970">https://doi.org/10.1002/ece3.7970</a>.
</div>
<div id="ref-unmarkedFiske" class="csl-entry" role="doc-biblioentry">
Fiske, Ian, and Richard Chandler. 2011. <span>“<span class="nocase">unmarked</span>: An <span>R</span> Package for Fitting Hierarchical Models of Wildlife Occurrence and Abundance.”</span> <em>Journal of Statistical Software</em> 43 (10): 1–23. <a href="https://www.jstatsoft.org/v43/i10/">https://www.jstatsoft.org/v43/i10/</a>.
</div>
<div id="ref-gimenez_statistical_2014" class="csl-entry" role="doc-biblioentry">
Gimenez, Olivier, Stephen T. Buckland, Byron J. T. Morgan, Nicolas Bez, Sophie Bertrand, Rémi Choquet, Stéphane Dray, et al. 2014. <span>“Statistical Ecology Comes of Age.”</span> <em>Biology Letters</em> 10 (12): 20140698. <a href="https://doi.org/10.1098/rsbl.2014.0698">https://doi.org/10.1098/rsbl.2014.0698</a>.
</div>
<div id="ref-he_deep_2016" class="csl-entry" role="doc-biblioentry">
He, Kaiming, Xiangyu Zhang, Shaoqing Ren, and Jian Sun. 2016. <span>“Deep <span>Residual</span> <span>Learning</span> for <span>Image</span> <span>Recognition</span>.”</span> In <em>2016 <span>IEEE</span> <span>Conference</span> on <span>Computer</span> <span>Vision</span> and <span>Pattern</span> <span>Recognition</span> (<span>CVPR</span>)</em>, 770–78. <a href="https://doi.org/10.1109/CVPR.2016.90">https://doi.org/10.1109/CVPR.2016.90</a>.
</div>
<div id="ref-NIPS2012_4824" class="csl-entry" role="doc-biblioentry">
Krizhevsky, Alex, Ilya Sutskever, and Geoffrey E. Hinton. 2012. <span>“ImageNet Classification with Deep Convolutional Neural Networks.”</span> In <em>Advances in Neural Information Processing Systems 25</em>, edited by F. Pereira, C. J. C. Burges, L. Bottou, and K. Q. Weinberger, 1097–1105. Curran Associates, Inc.
</div>
<div id="ref-lahoz-monfort_comprehensive_2021" class="csl-entry" role="doc-biblioentry">
Lahoz-Monfort, José J, and Michael J L Magrath. 2021. <span>“A <span>Comprehensive</span> <span>Overview</span> of <span>Technologies</span> for <span>Species</span> and <span>Habitat</span> <span>Monitoring</span> and <span>Conservation</span>.”</span> <em>BioScience</em>. <a href="https://doi.org/10.1093/biosci/biab073">https://doi.org/10.1093/biosci/biab073</a>.
</div>
<div id="ref-lai_evaluating_2019" class="csl-entry" role="doc-biblioentry">
Lai, Jiangshan, Christopher J. Lortie, Robert A. Muenchen, Jian Yang, and Keping Ma. 2019. <span>“Evaluating the Popularity of <span>R</span> in Ecology.”</span> <em>Ecosphere</em> 10 (1). <a href="https://doi.org/10.1002/ecs2.2567">https://doi.org/10.1002/ecs2.2567</a>.
</div>
<div id="ref-lamba_deep_2019" class="csl-entry" role="doc-biblioentry">
Lamba, Aakash, Phillip Cassey, Ramesh Raja Segaran, and Lian Pin Koh. 2019. <span>“Deep Learning for Environmental Conservation.”</span> <em>Current Biology</em> 29 (19): R977–82. <a href="https://doi.org/10.1016/j.cub.2019.08.016">https://doi.org/10.1016/j.cub.2019.08.016</a>.
</div>
<div id="ref-lecun_deep_2015" class="csl-entry" role="doc-biblioentry">
LeCun, Yann, Yoshua Bengio, and Geoffrey Hinton. 2015. <span>“Deep Learning.”</span> <em>Nature</em> 521 (7553): 436–44. <a href="https://doi.org/10.1038/nature14539">https://doi.org/10.1038/nature14539</a>.
</div>
<div id="ref-miller2011" class="csl-entry" role="doc-biblioentry">
Miller, David A., James D. Nichols, Brett T. McClintock, Evan H. Campbell Grant, Larissa L. Bailey, and Linda A. Weir. 2011. <span>“Improving Occupancy Estimation When Two Types of Observational Error Occur: Non-Detection and Species Misidentification.”</span> <em>Ecology</em> 92 (7): 1422–28. https://doi.org/<a href="https://doi.org/10.1890/10-1396.1">https://doi.org/10.1890/10-1396.1</a>.
</div>
<div id="ref-molinari-jobin_variation_2007" class="csl-entry" role="doc-biblioentry">
Molinari-Jobin, Anja, Fridolin Zimmermann, Andreas Ryser, Christine Breitenmoser-Würsten, Simon Capt, Urs Breitenmoser, Paolo Molinari, Heinrich Haller, and Roman Eyholzer. 2007. <span>“Variation in Diet, Prey Selectivity and Home-Range Size of <span>Eurasian</span> Lynx <span>Lynx</span> Lynx in <span>Switzerland</span>.”</span> <em>Wildlife Biology</em> 13 (4): 393–405. <a href="https://doi.org/10.2981/0909-6396(2007)13[393:VIDPSA]2.0.CO;2">https://doi.org/10.2981/0909-6396(2007)13[393:VIDPSA]2.0.CO;2</a>.
</div>
<div id="ref-norouzzadeh_deep_2021" class="csl-entry" role="doc-biblioentry">
Norouzzadeh, Mohammad Sadegh, Dan Morris, Sara Beery, Neel Joshi, Nebojsa Jojic, and Jeff Clune. 2021. <span>“A Deep Active Learning System for Species Identification and Counting in Camera Trap Images.”</span> Edited by Matthew Schofield. <em>Methods in Ecology and Evolution</em> 12 (1): 150–61. <a href="https://doi.org/10.1111/2041-210X.13504">https://doi.org/10.1111/2041-210X.13504</a>.
</div>
<div id="ref-NEURIPS2019_9015" class="csl-entry" role="doc-biblioentry">
Paszke, Adam, Sam Gross, Francisco Massa, Adam Lerer, James Bradbury, Gregory Chanan, Trevor Killeen, et al. 2019. <span>“PyTorch: An Imperative Style, High-Performance Deep Learning Library.”</span> In <em>Advances in Neural Information Processing Systems 32</em>, edited by H. Wallach, H. Larochelle, A. Beygelzimer, F. dAlché-Buc, E. Fox, and R. Garnett, 8024–35. Curran Associates, Inc. <a href="http://papers.neurips.cc/paper/9015-pytorch-an-imperative-style-high-performance-deep-learning-library.pdf">http://papers.neurips.cc/paper/9015-pytorch-an-imperative-style-high-performance-deep-learning-library.pdf</a>.
</div>
<div id="ref-Rota2016" class="csl-entry" role="doc-biblioentry">
Rota, Christopher T., Marco A. R. Ferreira, Roland W. Kays, Tavis D. Forrester, Elizabeth L. Kalies, William J. McShea, Arielle W. Parsons, and Joshua J. Millspaugh. 2016. <span>“A Multispecies Occupancy Model for Two or More Interacting Species.”</span> <em>Methods in Ecology and Evolution</em> 7 (10): 1164–73. https://doi.org/<a href="https://doi.org/10.1111/2041-210X.12587">https://doi.org/10.1111/2041-210X.12587</a>.
</div>
<div id="ref-shao_transfer_2015" class="csl-entry" role="doc-biblioentry">
Shao, Ling, Fan Zhu, and Xuelong Li. 2015. <span>“Transfer <span>Learning</span> for <span>Visual</span> <span>Categorization</span>: <span>A</span> <span>Survey</span>.”</span> <em>IEEE Transactions on Neural Networks and Learning Systems</em> 26 (5): 1019–34. <a href="https://doi.org/10.1109/TNNLS.2014.2330900">https://doi.org/10.1109/TNNLS.2014.2330900</a>.
</div>
<div id="ref-shorten_survey_2019" class="csl-entry" role="doc-biblioentry">
Shorten, Connor, and Taghi M. Khoshgoftaar. 2019. <span>“A Survey on <span>Image</span> <span>Data</span> <span>Augmentation</span> for <span>Deep</span> <span>Learning</span>.”</span> <em>Journal of Big Data</em> 6 (1): 60. <a href="https://doi.org/10.1186/s40537-019-0197-0">https://doi.org/10.1186/s40537-019-0197-0</a>.
</div>
<div id="ref-strubell2019energy" class="csl-entry" role="doc-biblioentry">
Strubell, Emma, Ananya Ganesh, and Andrew McCallum. 2019. <span>“Energy and Policy Considerations for Deep Learning in <span>NLP</span>.”</span> <em>arXiv:1906.02243</em>. <a href="http://arxiv.org/abs/1906.02243">http://arxiv.org/abs/1906.02243</a>.
</div>
<div id="ref-sutherland_identification_2013" class="csl-entry" role="doc-biblioentry">
Sutherland, William J., Robert P. Freckleton, H. Charles J. Godfray, Steven R. Beissinger, Tim Benton, Duncan D. Cameron, Yohay Carmel, et al. 2013. <span>“Identification of 100 Fundamental Ecological Questions.”</span> Edited by David Gibson. <em>Journal of Ecology</em> 101 (1): 58–67. <a href="https://doi.org/10.1111/1365-2745.12025">https://doi.org/10.1111/1365-2745.12025</a>.
</div>
<div id="ref-tabak_improving_2020" class="csl-entry" role="doc-biblioentry">
Tabak, Michael A., Mohammad S. Norouzzadeh, David W. Wolfson, Erica J. Newton, Raoul K. Boughton, Jacob S. Ivan, Eric A. Odell, et al. 2020. <span>“Improving the Accessibility and Transferability of Machine Learning Algorithms for Identification of Animals in Camera Trap Images: <span>MLWIC2</span>.”</span> <em>Ecology and Evolution</em> 10 (19): 10374–83. <a href="https://doi.org/10.1002/ece3.6692">https://doi.org/10.1002/ece3.6692</a>.
</div>
<div id="ref-tabak_machine_2019" class="csl-entry" role="doc-biblioentry">
Tabak, Michael A., Mohammad S. Norouzzadeh, David W. Wolfson, Steven J. Sweeney, Kurt C. Vercauteren, Nathan P. Snow, Joseph M. Halseth, et al. 2019. <span>“Machine Learning to Classify Animal Species in Camera Trap Images: <span>Applications</span> in Ecology.”</span> Edited by Theoni Photopoulou. <em>Methods in Ecology and Evolution</em> 10 (4): 585–90. <a href="https://doi.org/10.1111/2041-210X.13120">https://doi.org/10.1111/2041-210X.13120</a>.
</div>
<div id="ref-vandel_distribution_2005" class="csl-entry" role="doc-biblioentry">
Vandel, Jean-Michel, and Philippe Stahl. 2005. <span>“Distribution Trend of the <span>Eurasian</span> Lynx <span>Lynx</span> Lynx Populations in <span>France</span>.”</span> <em>Mammalia</em> 69 (2). <a href="https://doi.org/10.1515/mamm.2005.013">https://doi.org/10.1515/mamm.2005.013</a>.
</div>
<div id="ref-andina_deep_2018" class="csl-entry" role="doc-biblioentry">
Voulodimos, Athanasios, Nikolaos Doulamis, Anastasios Doulamis, and Eftychios Protopapadakis. 2018. <span>“Deep <span>Learning</span> for <span>Computer</span> <span>Vision</span>: <span>A</span> <span>Brief</span> <span>Review</span>.”</span> Edited by Diego Andina. <em>Computational Intelligence and Neuroscience</em> 2018 (February): 7068349. <a href="https://doi.org/10.1155/2018/7068349">https://doi.org/10.1155/2018/7068349</a>.
</div>
<div id="ref-wearn_responsible_2019" class="csl-entry" role="doc-biblioentry">
Wearn, Oliver R., Robin Freeman, and David M. P. Jacoby. 2019. <span>“Responsible <span>AI</span> for Conservation.”</span> <em>Nature Machine Intelligence</em> 1 (2): 72–73. <a href="https://doi.org/10.1038/s42256-019-0022-7">https://doi.org/10.1038/s42256-019-0022-7</a>.
</div>
<div id="ref-weinstein_computer_2018" class="csl-entry" role="doc-biblioentry">
Weinstein, Ben G. 2018. <span>“A Computer Vision for Animal Ecology.”</span> Edited by Laura Prugh. <em>Journal of Animal Ecology</em> 87 (3): 533–45. <a href="https://doi.org/10.1111/1365-2656.12780">https://doi.org/10.1111/1365-2656.12780</a>.
</div>
<div id="ref-willi_identifying_2019" class="csl-entry" role="doc-biblioentry">
Willi, Marco, Ross T. Pitman, Anabelle W. Cardoso, Christina Locke, Alexandra Swanson, Amy Boyer, Marten Veldthuis, and Lucy Fortson. 2019. <span>“Identifying Animal Species in Camera Trap Images Using Deep Learning and Citizen Science.”</span> Edited by Oscar Gaggiotti. <em>Methods in Ecology and Evolution</em> 10 (1): 80–91. <a href="https://doi.org/10.1111/2041-210X.13099">https://doi.org/10.1111/2041-210X.13099</a>.
</div>
<div id="ref-Yosinski2014" class="csl-entry" role="doc-biblioentry">
Yosinski, Jason, Jeff Clune, Yoshua Bengio, and Hod Lipson. 2014. <span>“How Transferable Are Features in Deep Neural Networks?”</span> In <em>Proceedings of the 27th International Conference on Neural Information Processing Systems - Volume 2</em>, 3320–28. NIPS’14. Cambridge, MA, USA: MIT Press.
</div>
<div id="ref-zimmermann_optimizing_2013" class="csl-entry" role="doc-biblioentry">
Zimmermann, Fridolin, Christine Breitenmoser-Würsten, Anja Molinari-Jobin, and Urs Breitenmoser. 2013. <span>“Optimizing the Size of the Area Surveyed for Monitoring a <span>Eurasian</span> Lynx (<span>Lynx</span> Lynx) Population in the <span>Swiss</span> <span>Alps</span> by Means of Photographic Capture-Recapture.”</span> <em>Integrative Zoology</em> 8 (3): 232–43. <a href="https://doi.org/10.1111/1749-4877.12017">https://doi.org/10.1111/1749-4877.12017</a>.
</div>
</div></section><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div quarto-reuse="quarto-reuse" class="quarto-appendix-contents"><a rel="license" href="https://creativecommons.org/licenses/by/4.0/">https://creativecommons.org/licenses/by/4.0/</a></div></section><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@article{gimenez2022,
  author = {Olivier Gimenez and Maëlis Kervellec and Jean-Baptiste
    Fanjul and Anna Chaine and Lucile Marescot and Yoann Bollet and
    Christophe Duchamp},
  publisher = {Société Française de Statistique},
  title = {Trade-Off Between Deep Learning for Species Identification
    and Inference about Predator-Prey Co-Occurrence},
  journal = {Computo},
  date = {2022-04-22},
  url = {https://computo.sfds.asso.fr/published-202204-deeplearning-occupancy-lynx},
  doi = {10.57750/yfm2-5f45},
  issn = {2824-7795},
  langid = {en},
  abstract = {Deep learning is used in computer vision problems with
    important applications in several scientific fields. In ecology for
    example, there is a growing interest in deep learning for
    automatizing repetitive analyses on large amounts of images, such as
    animal species identification. However, there are challenging issues
    toward the wide adoption of deep learning by the community of
    ecologists. First, there is a programming barrier as most algorithms
    are written in `Python` while most ecologists are versed in `R`.
    Second, recent applications of deep learning in ecology have focused
    on computational aspects and simple tasks without addressing the
    underlying ecological questions or carrying out the statistical data
    analysis to answer these questions. Here, we showcase a reproducible
    `R` workflow integrating both deep learning and statistical models
    using predator-prey relationships as a case study. We illustrate
    deep learning for the identification of animal species on images
    collected with camera traps, and quantify spatial co-occurrence
    using multispecies occupancy models. Despite average model
    classification performances, ecological inference was similar
    whether we analysed the ground truth dataset or the classified
    dataset. This result calls for further work on the trade-offs
    between time and resources allocated to train models with deep
    learning and our ability to properly address key ecological
    questions with biodiversity monitoring. We hope that our
    reproducible workflow will be useful to ecologists and applied
    statisticians.}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-gimenez2022" class="csl-entry quarto-appendix-citeas" role="doc-biblioentry">
Olivier Gimenez, Maëlis Kervellec, Jean-Baptiste Fanjul, Anna Chaine,
Lucile Marescot, Yoann Bollet, and Christophe Duchamp. 2022.
<span>“Trade-Off Between Deep Learning for Species Identification and
Inference about Predator-Prey Co-Occurrence.”</span> <em>Computo</em>,
April. <a href="https://doi.org/10.57750/yfm2-5f45">https://doi.org/10.57750/yfm2-5f45</a>.
</div></div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
    var links = window.document.querySelectorAll('a:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
      }
    }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb19" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Trade-off between deep learning for species identification and inference about predator-prey co-occurrence"</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Reproducible R workflow integrating models in computer vision and ecological statistics"</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="an">date-modified:</span><span class="co"> last-modified</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2022-04-22</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Olivier Gimenez</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co">    url: https://oliviergimenez.github.io/</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: CEFE, Univ Montpellier, CNRS, EPHE, IRD, Montpellier, France</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://www.cefe.cnrs.fr/</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Maëlis Kervellec</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: CEFE, Univ Montpellier, CNRS, EPHE, IRD, Montpellier, France</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://www.cefe.cnrs.fr/</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Jean-Baptiste Fanjul</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: Fédérations Départementales des Chasseurs du Jura, Arlay, France</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://www.chasseurdujura.com/</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Anna Chaine</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: CEFE, Univ Montpellier, CNRS, EPHE, IRD, Montpellier, France</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://www.cefe.cnrs.fr/</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Lucile Marescot</span></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: CBGP, Univ Montpellier, CIRAD, INRAE, Institut Agro, IRD, Montpellier, France</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://www6.montpellier.inrae.fr/cbgp</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Yoann Bollet</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: Fédération Départementale des Chasseurs de l'Ain, Bourg-en-Bresse, France</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: http://www.fdcain.com/</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Christophe Duchamp</span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation: Office Français de la Biodiversité, Unité Espèces à enjeux, Gap, France</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-url: https://www.ofb.gouv.fr/</span></span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a><span class="an">abstract:</span><span class="co"> &gt;+</span></span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a><span class="co">  Deep learning is used in computer vision problems with important applications in several scientific fields. In ecology for example, there is a growing interest in deep learning for automatizing repetitive analyses on large amounts of images, such as animal species identification. However, there are challenging issues toward the wide adoption of deep learning by the community of ecologists. First, there is a programming barrier as most algorithms are written in `Python` while most ecologists are versed in `R`. Second, recent applications of deep learning in ecology have focused on computational aspects and simple tasks without addressing the underlying ecological questions or carrying out the statistical data analysis to answer these questions. Here, we showcase a reproducible `R` workflow integrating both deep learning and statistical models using predator-prey relationships as a case study. We illustrate deep learning for the identification of animal species on images collected with camera traps, and quantify spatial co-occurrence using multispecies occupancy models. Despite average model classification performances, ecological inference was similar whether we analysed the ground truth dataset or the classified dataset. This result calls for further work on the trade-offs between time and resources allocated to train models with deep learning and our ability to properly address key ecological questions with biodiversity monitoring. We hope that our reproducible workflow will be useful to ecologists and applied statisticians.</span></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a><span class="an">keywords:</span><span class="co"> [computer-vision, deep-learning, species-distribution-modeling, ecological-statistics]</span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a><span class="an">citation:</span></span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a><span class="co">  type: article-journal</span></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a><span class="co">  container-title: "Computo"</span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a><span class="co">  publisher: "Société Française de Statistique"</span></span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a><span class="co">  doi: "10.57750/yfm2-5f45"</span></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a><span class="co">  url: https://computo.sfds.asso.fr/published-202204-deeplearning-occupancy-lynx</span></span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a><span class="co">  issn: "2824-7795"</span></span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a><span class="an">google-scholar:</span><span class="co"> true</span></span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a><span class="an">github-user:</span><span class="co"> computorg</span></span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a><span class="an">repo:</span><span class="co"> "published-202204-deeplearning-occupancy-lynx"</span></span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a><span class="an">draft:</span><span class="co"> false </span></span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a><span class="an">published:</span><span class="co"> true</span></span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a><span class="co">  computo-html: default</span></span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a><span class="co">  computo-pdf: default</span></span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">tidy =</span> <span class="cn">FALSE</span>, </span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a>                      <span class="at">fig.width =</span> <span class="dv">8</span>, </span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a>                      <span class="at">fig.height =</span> <span class="dv">8</span>, </span>
<span id="cb19-54"><a href="#cb19-54" aria-hidden="true" tabindex="-1"></a>                      <span class="at">echo =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-55"><a href="#cb19-55" aria-hidden="true" tabindex="-1"></a>                      <span class="at">warning =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-56"><a href="#cb19-56" aria-hidden="true" tabindex="-1"></a>                      <span class="at">message =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-57"><a href="#cb19-57" aria-hidden="true" tabindex="-1"></a>                      <span class="at">cache =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-58"><a href="#cb19-58" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">htmltools.dir.version =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-59"><a href="#cb19-59" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2022</span>) <span class="co"># because 2019, 2020 and 2021 were not great</span></span>
<span id="cb19-60"><a href="#cb19-60" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb19-61"><a href="#cb19-61" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_light</span>(<span class="at">base_size =</span> <span class="dv">14</span>))</span>
<span id="cb19-62"><a href="#cb19-62" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb19-63"><a href="#cb19-63" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb19-64"><a href="#cb19-64" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lubridate)</span>
<span id="cb19-65"><a href="#cb19-65" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb19-66"><a href="#cb19-66" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb19-67"><a href="#cb19-67" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(wesanderson)</span>
<span id="cb19-68"><a href="#cb19-68" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtext)</span>
<span id="cb19-69"><a href="#cb19-69" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastai)</span>
<span id="cb19-70"><a href="#cb19-70" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(highcharter)</span>
<span id="cb19-71"><a href="#cb19-71" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb19-72"><a href="#cb19-72" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cvms)</span>
<span id="cb19-73"><a href="#cb19-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-74"><a href="#cb19-74" aria-hidden="true" tabindex="-1"></a><span class="co"># visualise occupancy data</span></span>
<span id="cb19-75"><a href="#cb19-75" aria-hidden="true" tabindex="-1"></a><span class="co"># hack of the unmarked package plot function</span></span>
<span id="cb19-76"><a href="#cb19-76" aria-hidden="true" tabindex="-1"></a>plot_unmarked <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb19-77"><a href="#cb19-77" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> data</span>
<span id="cb19-78"><a href="#cb19-78" aria-hidden="true" tabindex="-1"></a>  y1 <span class="ot">&lt;-</span> <span class="fu">getY</span>(x)</span>
<span id="cb19-79"><a href="#cb19-79" aria-hidden="true" tabindex="-1"></a>  ym <span class="ot">&lt;-</span> <span class="fu">max</span>(y1, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb19-80"><a href="#cb19-80" aria-hidden="true" tabindex="-1"></a>  M <span class="ot">&lt;-</span> <span class="fu">nrow</span>(y1)</span>
<span id="cb19-81"><a href="#cb19-81" aria-hidden="true" tabindex="-1"></a>  J <span class="ot">&lt;-</span> <span class="fu">ncol</span>(y1)</span>
<span id="cb19-82"><a href="#cb19-82" aria-hidden="true" tabindex="-1"></a>  S <span class="ot">&lt;-</span> <span class="fu">length</span>(x<span class="sc">@</span>ylist)</span>
<span id="cb19-83"><a href="#cb19-83" aria-hidden="true" tabindex="-1"></a>  y1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">do.call</span>(rbind,x<span class="sc">@</span>ylist))</span>
<span id="cb19-84"><a href="#cb19-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(y1) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"obs"</span>,<span class="dv">1</span><span class="sc">:</span>J)</span>
<span id="cb19-85"><a href="#cb19-85" aria-hidden="true" tabindex="-1"></a>  y1<span class="sc">$</span>site <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>M,S)</span>
<span id="cb19-86"><a href="#cb19-86" aria-hidden="true" tabindex="-1"></a>  y1<span class="sc">$</span>species <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">rep</span>(<span class="fu">names</span>(x<span class="sc">@</span>ylist),<span class="at">each=</span>M))</span>
<span id="cb19-87"><a href="#cb19-87" aria-hidden="true" tabindex="-1"></a>  y2 <span class="ot">&lt;-</span> <span class="fu">reshape</span>(y1, <span class="at">idvar=</span><span class="fu">c</span>(<span class="st">"site"</span>, <span class="st">"species"</span>), <span class="at">varying=</span><span class="fu">list</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">obsNum</span>(x)),</span>
<span id="cb19-88"><a href="#cb19-88" aria-hidden="true" tabindex="-1"></a>                <span class="at">v.names=</span><span class="st">"value"</span>, <span class="at">direction=</span><span class="st">"long"</span>)</span>
<span id="cb19-89"><a href="#cb19-89" aria-hidden="true" tabindex="-1"></a>  y2<span class="sc">$</span>variable <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">paste</span>(<span class="st">"obs"</span>, y2<span class="sc">$</span>time))</span>
<span id="cb19-90"><a href="#cb19-90" aria-hidden="true" tabindex="-1"></a>  y2 <span class="sc">%&gt;%</span></span>
<span id="cb19-91"><a href="#cb19-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">as_factor</span>(value),</span>
<span id="cb19-92"><a href="#cb19-92" aria-hidden="true" tabindex="-1"></a>           <span class="at">site =</span> <span class="fu">as_factor</span>(site),</span>
<span id="cb19-93"><a href="#cb19-93" aria-hidden="true" tabindex="-1"></a>           <span class="at">variable =</span> <span class="fu">fct_reorder</span>(variable, time)) <span class="sc">%&gt;%</span></span>
<span id="cb19-94"><a href="#cb19-94" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb19-95"><a href="#cb19-95" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> variable, <span class="at">y =</span> site, <span class="at">fill =</span> value) <span class="sc">+</span> </span>
<span id="cb19-96"><a href="#cb19-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb19-97"><a href="#cb19-97" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">name =</span> <span class="st">'site'</span>,</span>
<span id="cb19-98"><a href="#cb19-98" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">rownames</span>(x<span class="sc">@</span>ylist<span class="sc">$</span>lynx)) <span class="sc">+</span> </span>
<span id="cb19-99"><a href="#cb19-99" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(species), <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-100"><a href="#cb19-100" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>( <span class="st">"#99CCFF"</span>,<span class="st">"#FF9900"</span>))<span class="sc">+</span></span>
<span id="cb19-101"><a href="#cb19-101" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">'time'</span>) <span class="sc">+</span></span>
<span id="cb19-102"><a href="#cb19-102" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb19-103"><a href="#cb19-103" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-104"><a href="#cb19-104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-105"><a href="#cb19-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-106"><a href="#cb19-106" aria-hidden="true" tabindex="-1"></a><span class="in">```{r speciesdata, include = FALSE}</span></span>
<span id="cb19-107"><a href="#cb19-107" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Jura pix</span></span>
<span id="cb19-108"><a href="#cb19-108" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">load</span>(<span class="st">"dat/metadata_Jura.RData"</span>)</span>
<span id="cb19-109"><a href="#cb19-109" aria-hidden="true" tabindex="-1"></a><span class="co"># nrow(allfiles)</span></span>
<span id="cb19-110"><a href="#cb19-110" aria-hidden="true" tabindex="-1"></a>allfiles <span class="sc">%&gt;%</span></span>
<span id="cb19-111"><a href="#cb19-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(DateTimeOriginal, <span class="fu">c</span>(<span class="st">"years"</span>, <span class="st">"months"</span>, <span class="st">"days"</span>, <span class="st">"hours"</span>, <span class="st">"mins"</span>, <span class="st">"secs"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-112"><a href="#cb19-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">years =</span> <span class="fu">replace</span>(years, <span class="fu">str_detect</span>(SourceFile, <span class="st">"!! ATTENTION 2017 AU LIEU DE 2016 !!"</span>), <span class="st">"2017"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-113"><a href="#cb19-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">years =</span> <span class="fu">replace</span>(years, <span class="fu">str_detect</span>(years, <span class="st">"2006"</span>), <span class="st">"2016"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-114"><a href="#cb19-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">years =</span> <span class="fu">replace</span>(years, <span class="fu">str_detect</span>(years, <span class="st">"2011"</span>), <span class="st">"2016"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-115"><a href="#cb19-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">years =</span> <span class="fu">replace</span>(years, <span class="fu">str_detect</span>(years, <span class="st">"2012"</span>), <span class="st">"2016"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-116"><a href="#cb19-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">years =</span> <span class="fu">as.numeric</span>(years), </span>
<span id="cb19-117"><a href="#cb19-117" aria-hidden="true" tabindex="-1"></a>         <span class="at">months =</span> <span class="fu">as.numeric</span>(months),  </span>
<span id="cb19-118"><a href="#cb19-118" aria-hidden="true" tabindex="-1"></a>         <span class="at">days =</span> <span class="fu">as.numeric</span>(days), </span>
<span id="cb19-119"><a href="#cb19-119" aria-hidden="true" tabindex="-1"></a>         <span class="at">hours =</span> <span class="fu">as.numeric</span>(hours), </span>
<span id="cb19-120"><a href="#cb19-120" aria-hidden="true" tabindex="-1"></a>         <span class="at">mins =</span> <span class="fu">as.numeric</span>(mins), </span>
<span id="cb19-121"><a href="#cb19-121" aria-hidden="true" tabindex="-1"></a>         <span class="at">secs =</span> <span class="fu">as.numeric</span>(secs)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-122"><a href="#cb19-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y =</span> years, </span>
<span id="cb19-123"><a href="#cb19-123" aria-hidden="true" tabindex="-1"></a>         <span class="at">m =</span> months,</span>
<span id="cb19-124"><a href="#cb19-124" aria-hidden="true" tabindex="-1"></a>         <span class="at">d =</span> days) <span class="sc">%&gt;%</span> </span>
<span id="cb19-125"><a href="#cb19-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(Date, years, months, days, <span class="at">sep=</span><span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-126"><a href="#cb19-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">h =</span> hours, <span class="at">mi =</span> mins, <span class="at">s =</span> secs) <span class="sc">%&gt;%</span> </span>
<span id="cb19-127"><a href="#cb19-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(Time, hours, mins, secs, <span class="at">sep=</span><span class="st">":"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-128"><a href="#cb19-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(DateTime, Date, Time, <span class="at">sep=</span><span class="st">" "</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-129"><a href="#cb19-129" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DateTime =</span> <span class="fu">ymd_hms</span>(DateTime)) <span class="sc">%&gt;%</span></span>
<span id="cb19-130"><a href="#cb19-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FileName, Directory, Keywords, DateTime, y, m, d, h , mi, s) <span class="ot">-&gt;</span> allpic_Jura</span>
<span id="cb19-131"><a href="#cb19-131" aria-hidden="true" tabindex="-1"></a><span class="co"># min(allpic_Jura$DateTime)</span></span>
<span id="cb19-132"><a href="#cb19-132" aria-hidden="true" tabindex="-1"></a><span class="co"># max(allpic_Jura$DateTime)</span></span>
<span id="cb19-133"><a href="#cb19-133" aria-hidden="true" tabindex="-1"></a><span class="co"># Get trap id.</span></span>
<span id="cb19-134"><a href="#cb19-134" aria-hidden="true" tabindex="-1"></a>trapic_final_Jura <span class="ot">&lt;-</span> allpic_Jura <span class="sc">%&gt;%</span> </span>
<span id="cb19-135"><a href="#cb19-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(FileName, <span class="fu">c</span>(<span class="st">"value"</span>, <span class="st">"value2"</span>, <span class="st">"key"</span>), <span class="at">extra =</span> <span class="st">"merge"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-136"><a href="#cb19-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">"n_point"</span>,value, value2, <span class="at">sep=</span><span class="st">"."</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-137"><a href="#cb19-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">as.numeric</span>(n_point))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-138"><a href="#cb19-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(<span class="at">sep=</span><span class="st">","</span>, Keywords, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-139"><a href="#cb19-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb19-140"><a href="#cb19-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">stri_replace_all_fixed</span>(especes, <span class="st">"vehicule"</span>, <span class="st">"humain"</span>),</span>
<span id="cb19-141"><a href="#cb19-141" aria-hidden="true" tabindex="-1"></a>         <span class="at">especes =</span> <span class="fu">stri_replace_all_fixed</span>(especes, <span class="st">"chien"</span>, <span class="st">"humain"</span>),</span>
<span id="cb19-142"><a href="#cb19-142" aria-hidden="true" tabindex="-1"></a>         <span class="at">especes =</span> <span class="fu">stri_replace_all_fixed</span>(especes, <span class="st">"cavalier"</span>, <span class="st">"humain"</span>),</span>
<span id="cb19-143"><a href="#cb19-143" aria-hidden="true" tabindex="-1"></a>         <span class="at">especes =</span> <span class="fu">stri_replace_all_fixed</span>(especes, <span class="st">"chasseur"</span>, <span class="st">"humain"</span>),</span>
<span id="cb19-144"><a href="#cb19-144" aria-hidden="true" tabindex="-1"></a>         <span class="at">especes =</span> <span class="fu">stri_replace_all_fixed</span>(especes, <span class="st">"frequentationhumaine"</span>, <span class="st">"humain"</span>),</span>
<span id="cb19-145"><a href="#cb19-145" aria-hidden="true" tabindex="-1"></a>         <span class="at">especes =</span> <span class="fu">stri_replace_all_fixed</span>(especes, <span class="st">"vaches"</span>, <span class="st">"humain"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-146"><a href="#cb19-146" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'lynx'</span>,<span class="st">'chevreuil'</span>,<span class="st">'chamois'</span>,<span class="st">'chat'</span>,<span class="st">'renard'</span>))</span>
<span id="cb19-147"><a href="#cb19-147" aria-hidden="true" tabindex="-1"></a><span class="co"># Format occupancy data</span></span>
<span id="cb19-148"><a href="#cb19-148" aria-hidden="true" tabindex="-1"></a>tocc <span class="ot">&lt;-</span> trapic_final_Jura <span class="sc">%&gt;%</span></span>
<span id="cb19-149"><a href="#cb19-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">as_factor</span>(especes),</span>
<span id="cb19-150"><a href="#cb19-150" aria-hidden="true" tabindex="-1"></a>         <span class="at">n_point =</span> <span class="fu">as_factor</span>(n_point)) <span class="sc">%&gt;%</span></span>
<span id="cb19-151"><a href="#cb19-151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mois =</span> <span class="fu">month</span>(DateTime)) <span class="sc">%&gt;%</span></span>
<span id="cb19-152"><a href="#cb19-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(mois,n_point,especes,<span class="at">.drop =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-153"><a href="#cb19-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-154"><a href="#cb19-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(mois, <span class="dv">3</span>, <span class="dv">11</span>)) </span>
<span id="cb19-155"><a href="#cb19-155" aria-hidden="true" tabindex="-1"></a><span class="co"># Table of detections and non-detections for lynx. </span></span>
<span id="cb19-156"><a href="#cb19-156" aria-hidden="true" tabindex="-1"></a>dat_lynx <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-157"><a href="#cb19-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'lynx'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-158"><a href="#cb19-158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-159"><a href="#cb19-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-160"><a href="#cb19-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-161"><a href="#cb19-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-162"><a href="#cb19-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-163"><a href="#cb19-163" aria-hidden="true" tabindex="-1"></a><span class="co"># Roe deer</span></span>
<span id="cb19-164"><a href="#cb19-164" aria-hidden="true" tabindex="-1"></a>dat_chevreuil <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-165"><a href="#cb19-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'chevreuil'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-166"><a href="#cb19-166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-167"><a href="#cb19-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-168"><a href="#cb19-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-169"><a href="#cb19-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-170"><a href="#cb19-170" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-171"><a href="#cb19-171" aria-hidden="true" tabindex="-1"></a><span class="co"># Chamois</span></span>
<span id="cb19-172"><a href="#cb19-172" aria-hidden="true" tabindex="-1"></a>dat_chamois <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-173"><a href="#cb19-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'chamois'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-174"><a href="#cb19-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-175"><a href="#cb19-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-176"><a href="#cb19-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-177"><a href="#cb19-177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-178"><a href="#cb19-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-179"><a href="#cb19-179" aria-hidden="true" tabindex="-1"></a><span class="co"># Fox</span></span>
<span id="cb19-180"><a href="#cb19-180" aria-hidden="true" tabindex="-1"></a>dat_renard <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-181"><a href="#cb19-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'renard'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-182"><a href="#cb19-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-183"><a href="#cb19-183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-184"><a href="#cb19-184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-185"><a href="#cb19-185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-186"><a href="#cb19-186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-187"><a href="#cb19-187" aria-hidden="true" tabindex="-1"></a><span class="co"># Cat</span></span>
<span id="cb19-188"><a href="#cb19-188" aria-hidden="true" tabindex="-1"></a>dat_chat <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-189"><a href="#cb19-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'chat'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-190"><a href="#cb19-190" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-191"><a href="#cb19-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-192"><a href="#cb19-192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-193"><a href="#cb19-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-194"><a href="#cb19-194" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-195"><a href="#cb19-195" aria-hidden="true" tabindex="-1"></a><span class="co"># Gather everything for later use</span></span>
<span id="cb19-196"><a href="#cb19-196" aria-hidden="true" tabindex="-1"></a>dat_jura <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">deer =</span> dat_chevreuil,</span>
<span id="cb19-197"><a href="#cb19-197" aria-hidden="true" tabindex="-1"></a>                 <span class="at">lynx =</span> dat_lynx,</span>
<span id="cb19-198"><a href="#cb19-198" aria-hidden="true" tabindex="-1"></a>                 <span class="at">chamois =</span> dat_chamois,</span>
<span id="cb19-199"><a href="#cb19-199" aria-hidden="true" tabindex="-1"></a>                 <span class="at">cat =</span> dat_chat,</span>
<span id="cb19-200"><a href="#cb19-200" aria-hidden="true" tabindex="-1"></a>                 <span class="at">fox =</span> dat_renard)</span>
<span id="cb19-201"><a href="#cb19-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-202"><a href="#cb19-202" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Ain data, pictures manually labelled</span></span>
<span id="cb19-203"><a href="#cb19-203" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">load</span>(<span class="st">'dat/metadata_Ain.RData'</span>)</span>
<span id="cb19-204"><a href="#cb19-204" aria-hidden="true" tabindex="-1"></a><span class="co"># nrow(allfiles)</span></span>
<span id="cb19-205"><a href="#cb19-205" aria-hidden="true" tabindex="-1"></a>allfiles <span class="sc">%&gt;%</span></span>
<span id="cb19-206"><a href="#cb19-206" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> observed) <span class="sc">%&gt;%</span> <span class="co"># pick manual tags</span></span>
<span id="cb19-207"><a href="#cb19-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(DateTimeOriginal, <span class="fu">c</span>(<span class="st">"years"</span>, <span class="st">"months"</span>, <span class="st">"days"</span>, <span class="st">"hours"</span>, <span class="st">"mins"</span>, <span class="st">"secs"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-208"><a href="#cb19-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">years =</span> <span class="fu">as.numeric</span>(years), </span>
<span id="cb19-209"><a href="#cb19-209" aria-hidden="true" tabindex="-1"></a>         <span class="at">months =</span> <span class="fu">as.numeric</span>(months),  </span>
<span id="cb19-210"><a href="#cb19-210" aria-hidden="true" tabindex="-1"></a>         <span class="at">days =</span> <span class="fu">as.numeric</span>(days), </span>
<span id="cb19-211"><a href="#cb19-211" aria-hidden="true" tabindex="-1"></a>         <span class="at">hours =</span> <span class="fu">as.numeric</span>(hours), </span>
<span id="cb19-212"><a href="#cb19-212" aria-hidden="true" tabindex="-1"></a>         <span class="at">mins =</span> <span class="fu">as.numeric</span>(mins), </span>
<span id="cb19-213"><a href="#cb19-213" aria-hidden="true" tabindex="-1"></a>         <span class="at">secs =</span> <span class="fu">as.numeric</span>(secs)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-214"><a href="#cb19-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y =</span> years, </span>
<span id="cb19-215"><a href="#cb19-215" aria-hidden="true" tabindex="-1"></a>         <span class="at">m =</span> months,</span>
<span id="cb19-216"><a href="#cb19-216" aria-hidden="true" tabindex="-1"></a>         <span class="at">d =</span> days) <span class="sc">%&gt;%</span> </span>
<span id="cb19-217"><a href="#cb19-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(Date, years, months, days, <span class="at">sep=</span><span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-218"><a href="#cb19-218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">h =</span> hours, <span class="at">mi =</span> mins, <span class="at">s =</span> secs) <span class="sc">%&gt;%</span> </span>
<span id="cb19-219"><a href="#cb19-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(Time, hours, mins, secs, <span class="at">sep=</span><span class="st">":"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-220"><a href="#cb19-220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(DateTime, Date, Time, <span class="at">sep=</span><span class="st">" "</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-221"><a href="#cb19-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DateTime =</span> <span class="fu">ymd_hms</span>(DateTime)) <span class="sc">%&gt;%</span></span>
<span id="cb19-222"><a href="#cb19-222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FileName =</span> pix) <span class="sc">%&gt;%</span></span>
<span id="cb19-223"><a href="#cb19-223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FileName, Directory, Keywords, DateTime, y, m, d, h , mi, s) <span class="ot">-&gt;</span> allpic_Ain</span>
<span id="cb19-224"><a href="#cb19-224" aria-hidden="true" tabindex="-1"></a><span class="co"># min(allpic$DateTime, na.rm = TRUE)</span></span>
<span id="cb19-225"><a href="#cb19-225" aria-hidden="true" tabindex="-1"></a><span class="co"># max(allpic$DateTime, na.rm = TRUE)</span></span>
<span id="cb19-226"><a href="#cb19-226" aria-hidden="true" tabindex="-1"></a><span class="co"># Get trap id</span></span>
<span id="cb19-227"><a href="#cb19-227" aria-hidden="true" tabindex="-1"></a>trapic_final_Ain <span class="ot">&lt;-</span> allpic_Ain <span class="sc">%&gt;%</span> </span>
<span id="cb19-228"><a href="#cb19-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(FileName, <span class="fu">c</span>(<span class="st">"value"</span>, <span class="st">"value2"</span>, <span class="st">"key"</span>), <span class="at">extra =</span> <span class="st">"merge"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-229"><a href="#cb19-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">"n_point"</span>,value, value2, <span class="at">sep=</span><span class="st">"."</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-230"><a href="#cb19-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">as.numeric</span>(n_point))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-231"><a href="#cb19-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Keywords <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'lynx'</span>,<span class="st">'chevreuil'</span>,<span class="st">'chamois'</span>,<span class="st">'chat'</span>,<span class="st">'renard'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-232"><a href="#cb19-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> <span class="fu">fct_drop</span>(Keywords))</span>
<span id="cb19-233"><a href="#cb19-233" aria-hidden="true" tabindex="-1"></a><span class="co"># Format occupancy data</span></span>
<span id="cb19-234"><a href="#cb19-234" aria-hidden="true" tabindex="-1"></a>tocc <span class="ot">&lt;-</span> trapic_final_Ain <span class="sc">%&gt;%</span></span>
<span id="cb19-235"><a href="#cb19-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">as_factor</span>(Keywords),</span>
<span id="cb19-236"><a href="#cb19-236" aria-hidden="true" tabindex="-1"></a>         <span class="at">n_point =</span> <span class="fu">as_factor</span>(n_point)) <span class="sc">%&gt;%</span></span>
<span id="cb19-237"><a href="#cb19-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mois =</span> <span class="fu">month</span>(DateTime)) <span class="sc">%&gt;%</span></span>
<span id="cb19-238"><a href="#cb19-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(mois,n_point,especes,<span class="at">.drop =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-239"><a href="#cb19-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-240"><a href="#cb19-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(mois, <span class="dv">3</span>, <span class="dv">11</span>)) </span>
<span id="cb19-241"><a href="#cb19-241" aria-hidden="true" tabindex="-1"></a><span class="co"># Table of detections and non-detections for roe deer. </span></span>
<span id="cb19-242"><a href="#cb19-242" aria-hidden="true" tabindex="-1"></a>dat_chevreuil <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-243"><a href="#cb19-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'chevreuil'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-244"><a href="#cb19-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-245"><a href="#cb19-245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-246"><a href="#cb19-246" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-247"><a href="#cb19-247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-248"><a href="#cb19-248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-249"><a href="#cb19-249" aria-hidden="true" tabindex="-1"></a><span class="co"># lynx</span></span>
<span id="cb19-250"><a href="#cb19-250" aria-hidden="true" tabindex="-1"></a>dat_lynx <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-251"><a href="#cb19-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'lynx'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-252"><a href="#cb19-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-253"><a href="#cb19-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-254"><a href="#cb19-254" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-255"><a href="#cb19-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-256"><a href="#cb19-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-257"><a href="#cb19-257" aria-hidden="true" tabindex="-1"></a><span class="co"># Chamois. </span></span>
<span id="cb19-258"><a href="#cb19-258" aria-hidden="true" tabindex="-1"></a>dat_chamois <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-259"><a href="#cb19-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'chamois'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-260"><a href="#cb19-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-261"><a href="#cb19-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-262"><a href="#cb19-262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-263"><a href="#cb19-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-264"><a href="#cb19-264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-265"><a href="#cb19-265" aria-hidden="true" tabindex="-1"></a><span class="co"># Fox</span></span>
<span id="cb19-266"><a href="#cb19-266" aria-hidden="true" tabindex="-1"></a>dat_renard <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-267"><a href="#cb19-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'renard'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-268"><a href="#cb19-268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-269"><a href="#cb19-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-270"><a href="#cb19-270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-271"><a href="#cb19-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-272"><a href="#cb19-272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-273"><a href="#cb19-273" aria-hidden="true" tabindex="-1"></a><span class="co"># Cat</span></span>
<span id="cb19-274"><a href="#cb19-274" aria-hidden="true" tabindex="-1"></a>dat_chat <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-275"><a href="#cb19-275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'chat'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-276"><a href="#cb19-276" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-277"><a href="#cb19-277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-278"><a href="#cb19-278" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-279"><a href="#cb19-279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-280"><a href="#cb19-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-281"><a href="#cb19-281" aria-hidden="true" tabindex="-1"></a><span class="co"># Gather data together</span></span>
<span id="cb19-282"><a href="#cb19-282" aria-hidden="true" tabindex="-1"></a>dat_ain <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">deer =</span> dat_chevreuil,</span>
<span id="cb19-283"><a href="#cb19-283" aria-hidden="true" tabindex="-1"></a>                <span class="at">lynx =</span> dat_lynx,</span>
<span id="cb19-284"><a href="#cb19-284" aria-hidden="true" tabindex="-1"></a>                <span class="at">chamois =</span> dat_chamois,</span>
<span id="cb19-285"><a href="#cb19-285" aria-hidden="true" tabindex="-1"></a>                <span class="at">cat =</span> dat_chat,</span>
<span id="cb19-286"><a href="#cb19-286" aria-hidden="true" tabindex="-1"></a>                <span class="at">fox =</span> dat_renard)</span>
<span id="cb19-287"><a href="#cb19-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-288"><a href="#cb19-288" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Ain data, pictures automatically labelled</span></span>
<span id="cb19-289"><a href="#cb19-289" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">load</span>(<span class="st">'dat/metadata_Ain.RData'</span>)</span>
<span id="cb19-290"><a href="#cb19-290" aria-hidden="true" tabindex="-1"></a><span class="co"># nrow(allfiles)</span></span>
<span id="cb19-291"><a href="#cb19-291" aria-hidden="true" tabindex="-1"></a>allfiles <span class="sc">%&gt;%</span></span>
<span id="cb19-292"><a href="#cb19-292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> predicted) <span class="sc">%&gt;%</span> <span class="co"># pick automatic tags</span></span>
<span id="cb19-293"><a href="#cb19-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(DateTimeOriginal, <span class="fu">c</span>(<span class="st">"years"</span>, <span class="st">"months"</span>, <span class="st">"days"</span>, <span class="st">"hours"</span>, <span class="st">"mins"</span>, <span class="st">"secs"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-294"><a href="#cb19-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">years =</span> <span class="fu">as.numeric</span>(years), </span>
<span id="cb19-295"><a href="#cb19-295" aria-hidden="true" tabindex="-1"></a>         <span class="at">months =</span> <span class="fu">as.numeric</span>(months),  </span>
<span id="cb19-296"><a href="#cb19-296" aria-hidden="true" tabindex="-1"></a>         <span class="at">days =</span> <span class="fu">as.numeric</span>(days), </span>
<span id="cb19-297"><a href="#cb19-297" aria-hidden="true" tabindex="-1"></a>         <span class="at">hours =</span> <span class="fu">as.numeric</span>(hours), </span>
<span id="cb19-298"><a href="#cb19-298" aria-hidden="true" tabindex="-1"></a>         <span class="at">mins =</span> <span class="fu">as.numeric</span>(mins), </span>
<span id="cb19-299"><a href="#cb19-299" aria-hidden="true" tabindex="-1"></a>         <span class="at">secs =</span> <span class="fu">as.numeric</span>(secs)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-300"><a href="#cb19-300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y =</span> years, </span>
<span id="cb19-301"><a href="#cb19-301" aria-hidden="true" tabindex="-1"></a>         <span class="at">m =</span> months,</span>
<span id="cb19-302"><a href="#cb19-302" aria-hidden="true" tabindex="-1"></a>         <span class="at">d =</span> days) <span class="sc">%&gt;%</span> </span>
<span id="cb19-303"><a href="#cb19-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(Date, years, months, days, <span class="at">sep=</span><span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-304"><a href="#cb19-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">h =</span> hours, <span class="at">mi =</span> mins, <span class="at">s =</span> secs) <span class="sc">%&gt;%</span> </span>
<span id="cb19-305"><a href="#cb19-305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(Time, hours, mins, secs, <span class="at">sep=</span><span class="st">":"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-306"><a href="#cb19-306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(DateTime, Date, Time, <span class="at">sep=</span><span class="st">" "</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-307"><a href="#cb19-307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DateTime =</span> <span class="fu">ymd_hms</span>(DateTime)) <span class="sc">%&gt;%</span></span>
<span id="cb19-308"><a href="#cb19-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FileName =</span> pix) <span class="sc">%&gt;%</span></span>
<span id="cb19-309"><a href="#cb19-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(FileName, Directory, Keywords, DateTime, y, m, d, h , mi, s) <span class="ot">-&gt;</span> allpic_Ain</span>
<span id="cb19-310"><a href="#cb19-310" aria-hidden="true" tabindex="-1"></a><span class="co"># min(allpic$DateTime, na.rm = TRUE)</span></span>
<span id="cb19-311"><a href="#cb19-311" aria-hidden="true" tabindex="-1"></a><span class="co"># max(allpic$DateTime, na.rm = TRUE)</span></span>
<span id="cb19-312"><a href="#cb19-312" aria-hidden="true" tabindex="-1"></a><span class="co"># Get trap id</span></span>
<span id="cb19-313"><a href="#cb19-313" aria-hidden="true" tabindex="-1"></a>trapic_final_Ain <span class="ot">&lt;-</span> allpic_Ain <span class="sc">%&gt;%</span> </span>
<span id="cb19-314"><a href="#cb19-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(FileName, <span class="fu">c</span>(<span class="st">"value"</span>, <span class="st">"value2"</span>, <span class="st">"key"</span>), <span class="at">extra =</span> <span class="st">"merge"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-315"><a href="#cb19-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">"n_point"</span>,value, value2, <span class="at">sep=</span><span class="st">"."</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-316"><a href="#cb19-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">as.numeric</span>(n_point))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-317"><a href="#cb19-317" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Keywords <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">'lynx'</span>,<span class="st">'chevreuil'</span>,<span class="st">'chamois'</span>,<span class="st">'chat'</span>,<span class="st">'renard'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-318"><a href="#cb19-318" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> <span class="fu">fct_drop</span>(Keywords))</span>
<span id="cb19-319"><a href="#cb19-319" aria-hidden="true" tabindex="-1"></a><span class="co"># Format occupancy data</span></span>
<span id="cb19-320"><a href="#cb19-320" aria-hidden="true" tabindex="-1"></a>tocc <span class="ot">&lt;-</span> trapic_final_Ain <span class="sc">%&gt;%</span></span>
<span id="cb19-321"><a href="#cb19-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">as_factor</span>(Keywords),</span>
<span id="cb19-322"><a href="#cb19-322" aria-hidden="true" tabindex="-1"></a>         <span class="at">n_point =</span> <span class="fu">as_factor</span>(n_point)) <span class="sc">%&gt;%</span></span>
<span id="cb19-323"><a href="#cb19-323" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mois =</span> <span class="fu">month</span>(DateTime)) <span class="sc">%&gt;%</span></span>
<span id="cb19-324"><a href="#cb19-324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(mois,n_point,especes,<span class="at">.drop =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-325"><a href="#cb19-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-326"><a href="#cb19-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(mois, <span class="dv">3</span>, <span class="dv">11</span>)) </span>
<span id="cb19-327"><a href="#cb19-327" aria-hidden="true" tabindex="-1"></a><span class="co"># Table of detections and non-detections for roe deer. </span></span>
<span id="cb19-328"><a href="#cb19-328" aria-hidden="true" tabindex="-1"></a>dat_chevreuil <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-329"><a href="#cb19-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'chevreuil'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-330"><a href="#cb19-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-331"><a href="#cb19-331" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-332"><a href="#cb19-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-333"><a href="#cb19-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-334"><a href="#cb19-334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-335"><a href="#cb19-335" aria-hidden="true" tabindex="-1"></a><span class="co"># lynx</span></span>
<span id="cb19-336"><a href="#cb19-336" aria-hidden="true" tabindex="-1"></a>dat_lynx <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-337"><a href="#cb19-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'lynx'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-338"><a href="#cb19-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-339"><a href="#cb19-339" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-340"><a href="#cb19-340" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-341"><a href="#cb19-341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-342"><a href="#cb19-342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-343"><a href="#cb19-343" aria-hidden="true" tabindex="-1"></a><span class="co"># Chamois. </span></span>
<span id="cb19-344"><a href="#cb19-344" aria-hidden="true" tabindex="-1"></a>dat_chamois <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-345"><a href="#cb19-345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'chamois'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-346"><a href="#cb19-346" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-347"><a href="#cb19-347" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-348"><a href="#cb19-348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-349"><a href="#cb19-349" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-350"><a href="#cb19-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-351"><a href="#cb19-351" aria-hidden="true" tabindex="-1"></a><span class="co"># Fox</span></span>
<span id="cb19-352"><a href="#cb19-352" aria-hidden="true" tabindex="-1"></a>dat_renard <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-353"><a href="#cb19-353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'renard'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-354"><a href="#cb19-354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-355"><a href="#cb19-355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-356"><a href="#cb19-356" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-357"><a href="#cb19-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-358"><a href="#cb19-358" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-359"><a href="#cb19-359" aria-hidden="true" tabindex="-1"></a><span class="co"># Cat</span></span>
<span id="cb19-360"><a href="#cb19-360" aria-hidden="true" tabindex="-1"></a>dat_chat <span class="ot">&lt;-</span> tocc <span class="sc">%&gt;%</span></span>
<span id="cb19-361"><a href="#cb19-361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(especes <span class="sc">==</span> <span class="st">'chat'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-362"><a href="#cb19-362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">counts =</span> <span class="fu">if_else</span>(n <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-363"><a href="#cb19-363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-364"><a href="#cb19-364" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(mois,n_point,counts) <span class="sc">%&gt;%</span></span>
<span id="cb19-365"><a href="#cb19-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> mois, <span class="at">values_from =</span> counts) <span class="sc">%&gt;%</span> </span>
<span id="cb19-366"><a href="#cb19-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() </span>
<span id="cb19-367"><a href="#cb19-367" aria-hidden="true" tabindex="-1"></a><span class="co"># Gather data together (obtained via deeplearning)</span></span>
<span id="cb19-368"><a href="#cb19-368" aria-hidden="true" tabindex="-1"></a>dat_ain_dl <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">deer =</span> dat_chevreuil,</span>
<span id="cb19-369"><a href="#cb19-369" aria-hidden="true" tabindex="-1"></a>                <span class="at">lynx =</span> dat_lynx,</span>
<span id="cb19-370"><a href="#cb19-370" aria-hidden="true" tabindex="-1"></a>                <span class="at">chamois =</span> dat_chamois,</span>
<span id="cb19-371"><a href="#cb19-371" aria-hidden="true" tabindex="-1"></a>                <span class="at">cat =</span> dat_chat,</span>
<span id="cb19-372"><a href="#cb19-372" aria-hidden="true" tabindex="-1"></a>                <span class="at">fox =</span> dat_renard)</span>
<span id="cb19-373"><a href="#cb19-373" aria-hidden="true" tabindex="-1"></a><span class="co"># Map('-', dat_ain_dl, dat_ain)</span></span>
<span id="cb19-374"><a href="#cb19-374" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-375"><a href="#cb19-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-376"><a href="#cb19-376" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb19-377"><a href="#cb19-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-378"><a href="#cb19-378" aria-hidden="true" tabindex="-1"></a>Computer vision is a field of artificial intelligence in which a machine is taught how to extract and interpret the content of an image <span class="co">[</span><span class="ot">@NIPS2012_4824</span><span class="co">]</span>. Computer vision relies on deep learning that allows computational models to learn from training data -- a set of manually labelled images -- and make predictions on new data -- a set of unlabelled images <span class="co">[</span><span class="ot">@baraniuk_science_2020; @lecun_deep_2015</span><span class="co">]</span>. With the growing availability of massive data, computer vision with deep learning is being increasingly used to perform tasks such as object detection, face recognition, action and activity recognition or human pose estimation in fields as diverse as medicine, robotics, transportation, genomics, sports and agriculture <span class="co">[</span><span class="ot">@andina_deep_2018</span><span class="co">]</span>. </span>
<span id="cb19-379"><a href="#cb19-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-380"><a href="#cb19-380" aria-hidden="true" tabindex="-1"></a>In ecology in particular, there is a growing interest in deep learning for automatizing repetitive analyses on large amounts of images, such as identifying plant and animal species, distinguishing individuals of the same or different species, counting individuals or detecting relevant features <span class="co">[</span><span class="ot">@christin_applications_2019; @lamba_deep_2019; @weinstein_computer_2018</span><span class="co">]</span>. By saving hours of manual data analyses and tapping into massive amounts of data that keep accumulating with technological advances, deep learning has the potential to become an essential tool for ecologists and applied statisticians.</span>
<span id="cb19-381"><a href="#cb19-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-382"><a href="#cb19-382" aria-hidden="true" tabindex="-1"></a>Despite the promising future of computer vision and deep learning, there are challenging issues toward their wide adoption by the community of ecologists <span class="co">[</span><span class="ot">e.g. @wearn_responsible_2019</span><span class="co">]</span>. First, there is a programming barrier as most algorithms are written in the <span class="in">`Python`</span> language (but see <span class="co">[</span><span class="ot">MXNet in R</span><span class="co">](https://mxnet.apache.org/versions/1.6/api/r/docs/index.html)</span> and the <span class="co">[</span><span class="ot">R interface to Keras</span><span class="co">](https://keras.rstudio.com/index.html)</span>) while most ecologists are versed in <span class="in">`R`</span>  <span class="co">[</span><span class="ot">@lai_evaluating_2019</span><span class="co">]</span>. If ecologists are to use computer vision in routine, there is a need for bridges between these two languages (through, e.g., the <span class="in">`reticulate`</span> package @reticulate_ref or the <span class="in">`shiny`</span> package @tabak_improving_2020). Second, ecologists may be reluctant to develop deep learning algorithms that require large amounts of computation time and consequently come with an environmental cost due to carbon emissions <span class="co">[</span><span class="ot">@strubell2019energy</span><span class="co">]</span>. Third, recent applications of computer vision via deep learning in ecology have focused on computational aspects and simple tasks without addressing the underlying ecological questions <span class="co">[</span><span class="ot">@sutherland_identification_2013</span><span class="co">]</span>, or carrying out statistical data analysis to answer these questions <span class="co">[</span><span class="ot">@gimenez_statistical_2014</span><span class="co">]</span>. Although perfectly understandable given the challenges at hand, we argue that a better integration of the *why* (ecological questions), the *what* (automatically labelled images) and the *how* (statistics) would be beneficial to computer vision for ecology <span class="co">[</span><span class="ot">see also @weinstein_computer_2018</span><span class="co">]</span>.</span>
<span id="cb19-383"><a href="#cb19-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-384"><a href="#cb19-384" aria-hidden="true" tabindex="-1"></a>Here, we showcase a full why-what-how workflow in <span class="in">`R`</span> using a case study on the structure of an ecological community (a set of co-occurring species) composed of the Eurasian lynx (*Lynx lynx*) and its two main preys. First, we introduce the case study and motivate the need for deep learning. Second we illustrate deep learning for the identification of animal species in large amounts of images, including model training and validation with a dataset of labelled images, and prediction with a new dataset of unlabelled images. Last, we proceed with the quantification of spatial co-occurrence using statistical models. </span>
<span id="cb19-385"><a href="#cb19-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-386"><a href="#cb19-386" aria-hidden="true" tabindex="-1"></a><span class="fu"># Collecting images with camera traps</span></span>
<span id="cb19-387"><a href="#cb19-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-388"><a href="#cb19-388" aria-hidden="true" tabindex="-1"></a>Lynx (*Lynx lynx*) went extinct in France at the end of the 19th century due to habitat degradation, human persecution and decrease in prey availability <span class="co">[</span><span class="ot">@vandel_distribution_2005</span><span class="co">]</span>. The species was reintroduced in Switzerland in the 1970s <span class="co">[</span><span class="ot">@breitenmoser_large_1998</span><span class="co">]</span>, then re-colonised France through the Jura mountains in the 1980s <span class="co">[</span><span class="ot">@vandel_distribution_2005</span><span class="co">]</span>. The species is listed as endangered under the 2017 IUCN Red list and is of conservation concern in France due to habitat fragmentation, poaching and collisions with vehicles. The Jura holds the bulk of the French lynx population. </span>
<span id="cb19-389"><a href="#cb19-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-390"><a href="#cb19-390" aria-hidden="true" tabindex="-1"></a>To better understand its distribution, we need to quantify its interactions with its main preys, roe deer (*Capreolus capreolus*) and chamois (*Rupicapra rupicapra*) <span class="co">[</span><span class="ot">@molinari-jobin_variation_2007</span><span class="co">]</span>, two ungulate species that are also hunted. To assess the relative contribution of predation and hunting to the community structure and dynamics, a predator-prey program was set up jointly by the French Office for Biodiversity, the Federations of Hunters from the Jura, Ain and Haute-Savoie counties and the French National Centre for Scientific Research. </span>
<span id="cb19-391"><a href="#cb19-391" aria-hidden="true" tabindex="-1"></a>Animal detections were made using a set of camera traps in the Jura mountains that were deployed in the Jura and Ain counties (see @fig-area). Altitude in the Jura site ranges from 520m to 1150m, and from 400m to 950m for the Ain site. Woodland areas cover 69% of the Ain site, with deciduous forests (63%) followed by coniferous (19.5%) and mixed forest (12.5%). In the Jura site, woodland areas cover 62% of the area, with mixed forests (46.6%), deciduous forests (37.3%) and coniferous (14%). In both sites, the remaining habitat is meadows used by cattle.</span>
<span id="cb19-392"><a href="#cb19-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-393"><a href="#cb19-393" aria-hidden="true" tabindex="-1"></a>We divided the two study areas into grids of 2.7 $\times$ 2.7 km cells or sites hereafter <span class="co">[</span><span class="ot">@zimmermann_optimizing_2013</span><span class="co">]</span> in which we set two camera traps per site (Xenon white flash with passive infrared trigger mechanisms, model Capture, Ambush and Attack; Cuddeback), with <span class="in">`r lapply(dat_jura, nrow)$lynx`</span> sites in the Jura study area, and <span class="in">`r lapply(dat_ain, nrow)$lynx`</span> in the Ain study area that were active over the study period (from February 2016 to October 2017 for the Jura county, and from February 2017 to May 2019 for the Ain county). The location of camera traps was chosen to maximise lynx detection. More precisely, camera traps were set up along large paths in the forest, on each side of the path at 50cm high. Camera traps were checked weekly to change memory cards, batteries and to remove fresh snow after heavy snowfall. </span>
<span id="cb19-394"><a href="#cb19-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-395"><a href="#cb19-395" aria-hidden="true" tabindex="-1"></a>:::{.column-page}</span>
<span id="cb19-396"><a href="#cb19-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-397"><a href="#cb19-397" aria-hidden="true" tabindex="-1"></a><span class="in">```{r studysite, fig.width = 10}</span></span>
<span id="cb19-398"><a href="#cb19-398" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-area </span></span>
<span id="cb19-399"><a href="#cb19-399" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Study area, grid and camera trap locations."</span></span>
<span id="cb19-400"><a href="#cb19-400" aria-hidden="true" tabindex="-1"></a>site_jura <span class="ot">&lt;-</span> <span class="fu">st_read</span>(<span class="st">"shp/SIG_39/communes_site_etude_jura.shp"</span>, <span class="at">quiet =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-401"><a href="#cb19-401" aria-hidden="true" tabindex="-1"></a>area_jura <span class="ot">&lt;-</span> <span class="fu">st_read</span>(<span class="st">"shp/SIG_39/grille_jura_2.7x2.7.shp"</span>, <span class="at">quiet =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-402"><a href="#cb19-402" aria-hidden="true" tabindex="-1"></a>trap_jura <span class="ot">&lt;-</span> <span class="fu">st_read</span>(<span class="st">"shp/SIG_39/Pieges_photos_JURA_l93.shp"</span>, <span class="at">quiet =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-403"><a href="#cb19-403" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(site_jura) <span class="ot">&lt;-</span>  <span class="dv">2154</span></span>
<span id="cb19-404"><a href="#cb19-404" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(area_jura) <span class="ot">&lt;-</span>  <span class="dv">2154</span></span>
<span id="cb19-405"><a href="#cb19-405" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(trap_jura) <span class="ot">&lt;-</span>  <span class="dv">2154</span></span>
<span id="cb19-406"><a href="#cb19-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-407"><a href="#cb19-407" aria-hidden="true" tabindex="-1"></a>fig1a <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb19-408"><a href="#cb19-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> site_jura) <span class="sc">+</span></span>
<span id="cb19-409"><a href="#cb19-409" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> area_jura,  <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-410"><a href="#cb19-410" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> trap_jura, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">shape =</span> <span class="dv">4</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb19-411"><a href="#cb19-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"A. Jura county"</span>) </span>
<span id="cb19-412"><a href="#cb19-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-413"><a href="#cb19-413" aria-hidden="true" tabindex="-1"></a>site_ain <span class="ot">&lt;-</span> <span class="fu">st_read</span>(<span class="st">"shp/SIG_01/Ain_ZE.shp"</span>, <span class="at">quiet =</span> <span class="cn">TRUE</span>) <span class="co"># l'aire d'etude</span></span>
<span id="cb19-414"><a href="#cb19-414" aria-hidden="true" tabindex="-1"></a>area_ain <span class="ot">&lt;-</span> <span class="fu">st_read</span>(<span class="st">"shp/SIG_01/Maille_2.7_Ain.shp"</span>, <span class="at">quiet =</span> <span class="cn">TRUE</span>) <span class="co"># la grille pour les pieges</span></span>
<span id="cb19-415"><a href="#cb19-415" aria-hidden="true" tabindex="-1"></a>trap_ain <span class="ot">&lt;-</span> <span class="fu">st_read</span>(<span class="st">"shp/SIG_01/PP01_restreint.shp"</span>, <span class="at">quiet =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-416"><a href="#cb19-416" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(site_ain) <span class="ot">&lt;-</span>  <span class="dv">2154</span></span>
<span id="cb19-417"><a href="#cb19-417" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(area_ain) <span class="ot">&lt;-</span>  <span class="dv">2154</span></span>
<span id="cb19-418"><a href="#cb19-418" aria-hidden="true" tabindex="-1"></a><span class="fu">st_crs</span>(trap_ain) <span class="ot">&lt;-</span>  <span class="dv">2154</span></span>
<span id="cb19-419"><a href="#cb19-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-420"><a href="#cb19-420" aria-hidden="true" tabindex="-1"></a>fig1b <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb19-421"><a href="#cb19-421" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> site_ain) <span class="sc">+</span></span>
<span id="cb19-422"><a href="#cb19-422" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> area_ain,  <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-423"><a href="#cb19-423" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> trap_ain, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">shape =</span> <span class="dv">4</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb19-424"><a href="#cb19-424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"B. Ain county"</span>)</span>
<span id="cb19-425"><a href="#cb19-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-426"><a href="#cb19-426" aria-hidden="true" tabindex="-1"></a>plot_row <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(fig1a, fig1b)</span>
<span id="cb19-427"><a href="#cb19-427" aria-hidden="true" tabindex="-1"></a>plot_row</span>
<span id="cb19-428"><a href="#cb19-428" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-429"><a href="#cb19-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-430"><a href="#cb19-430" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb19-431"><a href="#cb19-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-434"><a href="#cb19-434" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-435"><a href="#cb19-435" aria-hidden="true" tabindex="-1"></a>species_Jura <span class="ot">&lt;-</span> allpic_Jura <span class="sc">%&gt;%</span> </span>
<span id="cb19-436"><a href="#cb19-436" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(FileName, <span class="fu">c</span>(<span class="st">"value"</span>, <span class="st">"value2"</span>, <span class="st">"key"</span>), <span class="at">extra =</span> <span class="st">"merge"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-437"><a href="#cb19-437" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">"n_point"</span>,value, value2, <span class="at">sep=</span><span class="st">"."</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-438"><a href="#cb19-438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">as.numeric</span>(n_point))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-439"><a href="#cb19-439" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(<span class="at">sep=</span><span class="st">","</span>, Keywords, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-440"><a href="#cb19-440" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">as_factor</span>(Keywords)) <span class="sc">%&gt;%</span></span>
<span id="cb19-441"><a href="#cb19-441" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(especes), </span>
<span id="cb19-442"><a href="#cb19-442" aria-hidden="true" tabindex="-1"></a>         <span class="sc">!</span>especes<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">"Non identifié"</span>, <span class="st">"Non id"</span>, <span class="st">"Non iden"</span>, <span class="st">""</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-443"><a href="#cb19-443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">fct_recode</span>(especes, </span>
<span id="cb19-444"><a href="#cb19-444" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"sanglier"</span> <span class="ot">=</span> <span class="st">"sangli"</span>,</span>
<span id="cb19-445"><a href="#cb19-445" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"humain"</span> <span class="ot">=</span> <span class="st">"humai"</span>,</span>
<span id="cb19-446"><a href="#cb19-446" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"chat"</span> <span class="ot">=</span> <span class="st">"chat forestier"</span>,</span>
<span id="cb19-447"><a href="#cb19-447" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"chat"</span> <span class="ot">=</span> <span class="st">"chat domestique"</span>,</span>
<span id="cb19-448"><a href="#cb19-448" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"chasseur"</span> <span class="ot">=</span> <span class="st">"chasse"</span>,</span>
<span id="cb19-449"><a href="#cb19-449" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"sanglier"</span> <span class="ot">=</span> <span class="st">"sangliers"</span>,</span>
<span id="cb19-450"><a href="#cb19-450" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"blaireau"</span> <span class="ot">=</span> <span class="st">"blaireaux"</span>)) </span>
<span id="cb19-451"><a href="#cb19-451" aria-hidden="true" tabindex="-1"></a>species_Ain <span class="ot">&lt;-</span> allpic_Ain <span class="sc">%&gt;%</span> </span>
<span id="cb19-452"><a href="#cb19-452" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(FileName, <span class="fu">c</span>(<span class="st">"value"</span>, <span class="st">"value2"</span>, <span class="st">"key"</span>), <span class="at">extra =</span> <span class="st">"merge"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-453"><a href="#cb19-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">"n_point"</span>,value, value2, <span class="at">sep=</span><span class="st">"."</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-454"><a href="#cb19-454" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">as.numeric</span>(n_point))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-455"><a href="#cb19-455" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(<span class="at">sep=</span><span class="st">","</span>, Keywords, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-456"><a href="#cb19-456" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">as_factor</span>(Keywords)) <span class="sc">%&gt;%</span></span>
<span id="cb19-457"><a href="#cb19-457" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(especes), </span>
<span id="cb19-458"><a href="#cb19-458" aria-hidden="true" tabindex="-1"></a>         <span class="sc">!</span>especes<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">"Non identifié"</span>, <span class="st">"Non id"</span>, <span class="st">"Non iden"</span>, <span class="st">""</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-459"><a href="#cb19-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">fct_recode</span>(especes, </span>
<span id="cb19-460"><a href="#cb19-460" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"sanglier"</span> <span class="ot">=</span> <span class="st">"sangliers"</span>,</span>
<span id="cb19-461"><a href="#cb19-461" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"blaireau"</span> <span class="ot">=</span> <span class="st">"blaireaux"</span>)) </span>
<span id="cb19-462"><a href="#cb19-462" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-463"><a href="#cb19-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-464"><a href="#cb19-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-465"><a href="#cb19-465" aria-hidden="true" tabindex="-1"></a>In total, <span class="in">`r nrow(species_Jura)`</span> and <span class="in">`r nrow(species_Ain)`</span> pictures were considered in the Jura and Ain sites respectively after manually droping empty pictures and pictures with unidentified species. Note that classifying empty images could be automatised with deep learning <span class="co">[</span><span class="ot">@norouzzadeh_deep_2021; @tabak_improving_2020</span><span class="co">]</span>. We identified the species present on all images by hand (see @tbl-jura) using <span class="in">`digiKam`</span> a free open-source digital photo management application (<span class="ot">&lt;https://www.digikam.org/&gt;</span>). This operation took several weeks of labor full time, which is often identified as a limitation of camera trap studies. To expedite this tedious task, computer vision with deep learning has been identified as a promising approach <span class="co">[</span><span class="ot">@norouzzadeh_deep_2021; @tabak_machine_2019; @willi_identifying_2019</span><span class="co">]</span>.</span>
<span id="cb19-466"><a href="#cb19-466" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Labelled images for the Ain county are available from &lt;https://mycore.core-cloud.net/index.php/s/1DHbKdQzPKuDbxl&gt; and that for the Jura county are available from &lt;https://mycore.core-cloud.net/index.php/s/8GsYfAoyeDGzuNY&gt;.  --&gt;</span>  </span>
<span id="cb19-467"><a href="#cb19-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-468"><a href="#cb19-468" aria-hidden="true" tabindex="-1"></a><span class="in">```{r categories, cache = FALSE}</span></span>
<span id="cb19-469"><a href="#cb19-469" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-jura</span></span>
<span id="cb19-470"><a href="#cb19-470" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Species identified in the Jura and Ain study sites with samples size (n). Only first 10 species with most images are shown."</span></span>
<span id="cb19-471"><a href="#cb19-471" aria-hidden="true" tabindex="-1"></a>n_Jura <span class="ot">&lt;-</span> species_Jura <span class="sc">%&gt;%</span> <span class="fu">count</span>(especes, <span class="at">sort =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-472"><a href="#cb19-472" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">fct_recode</span>(especes,</span>
<span id="cb19-473"><a href="#cb19-473" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"human"</span> <span class="ot">=</span> <span class="st">"humain"</span>,</span>
<span id="cb19-474"><a href="#cb19-474" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"vehicule"</span> <span class="ot">=</span> <span class="st">"véhicule"</span>,</span>
<span id="cb19-475"><a href="#cb19-475" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"dog"</span> <span class="ot">=</span> <span class="st">"chien"</span>,</span>
<span id="cb19-476"><a href="#cb19-476" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"fox"</span> <span class="ot">=</span> <span class="st">"renard"</span>,</span>
<span id="cb19-477"><a href="#cb19-477" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"wild boar"</span> <span class="ot">=</span> <span class="st">"sanglier"</span>,</span>
<span id="cb19-478"><a href="#cb19-478" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"roe deer"</span> <span class="ot">=</span> <span class="st">"chevreuil"</span>,</span>
<span id="cb19-479"><a href="#cb19-479" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"cat"</span> <span class="ot">=</span> <span class="st">"chat"</span>,</span>
<span id="cb19-480"><a href="#cb19-480" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"badger"</span> <span class="ot">=</span> <span class="st">"blaireau"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-481"><a href="#cb19-481" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Species in Jura study site"</span> <span class="ot">=</span> especes)</span>
<span id="cb19-482"><a href="#cb19-482" aria-hidden="true" tabindex="-1"></a>n_Ain <span class="ot">&lt;-</span> species_Ain <span class="sc">%&gt;%</span> <span class="fu">count</span>(especes, <span class="at">sort =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-483"><a href="#cb19-483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">especes =</span> <span class="fu">fct_recode</span>(especes,</span>
<span id="cb19-484"><a href="#cb19-484" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"human"</span> <span class="ot">=</span> <span class="st">"humain"</span>,</span>
<span id="cb19-485"><a href="#cb19-485" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"dog"</span> <span class="ot">=</span> <span class="st">"chien"</span>,</span>
<span id="cb19-486"><a href="#cb19-486" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"fox"</span> <span class="ot">=</span> <span class="st">"renard"</span>,</span>
<span id="cb19-487"><a href="#cb19-487" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"wild boar"</span> <span class="ot">=</span> <span class="st">"sanglier"</span>,</span>
<span id="cb19-488"><a href="#cb19-488" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"roe deer"</span> <span class="ot">=</span> <span class="st">"chevreuil"</span>,</span>
<span id="cb19-489"><a href="#cb19-489" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"cat"</span> <span class="ot">=</span> <span class="st">"chat"</span>,</span>
<span id="cb19-490"><a href="#cb19-490" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"badger"</span> <span class="ot">=</span> <span class="st">"blaireau"</span>,</span>
<span id="cb19-491"><a href="#cb19-491" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"hunter"</span> <span class="ot">=</span> <span class="st">"chasseur"</span>,</span>
<span id="cb19-492"><a href="#cb19-492" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"rider"</span> <span class="ot">=</span> <span class="st">"cavalier"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-493"><a href="#cb19-493" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Species in Ain study site"</span> <span class="ot">=</span> especes)</span>
<span id="cb19-494"><a href="#cb19-494" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(n_Jura, n_Ain) <span class="sc">%&gt;%</span> <span class="fu">kable</span>() <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>()</span>
<span id="cb19-495"><a href="#cb19-495" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-496"><a href="#cb19-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-497"><a href="#cb19-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-498"><a href="#cb19-498" aria-hidden="true" tabindex="-1"></a><span class="fu"># Deep learning for species identification</span></span>
<span id="cb19-499"><a href="#cb19-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-500"><a href="#cb19-500" aria-hidden="true" tabindex="-1"></a>Using the images we obtained with camera traps (Table 1), we trained a model for identifying species using the Jura study site as a calibration dataset. We then assessed this model's ability to automatically identify species on a new dataset, also known as transferability, using the Ain study site as an evaluation dataset. Even though in the present work we quantified co-occurrence between lynx and its prey, we included other species in the training to investigate the structure and dynamics of the entire community in future work. Also, the use of specific species categories instead of just a "other" category besides the focal species should help the algorithm to determine with better confidence when a picture does not contain a focal species in situations where there is no doubt that this is another species (think of a vehicle for example), or where a species is detected with which a focal species can be confused, e.g. lynx with fox. </span>
<span id="cb19-501"><a href="#cb19-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-502"><a href="#cb19-502" aria-hidden="true" tabindex="-1"></a><span class="fu">## Training - Jura study site</span></span>
<span id="cb19-503"><a href="#cb19-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-504"><a href="#cb19-504" aria-hidden="true" tabindex="-1"></a>We selected at random 80\% of the annotated images for each species in the Jura study site for training, and 20\% for testing. We applied various transformations (flipping, brightness and contrast modifications following @shorten_survey_2019) to improve training (see Appendix). To reduce model training time and overcome the small number of images, we used transfer learning <span class="co">[</span><span class="ot">@Yosinski2014; @shao_transfer_2015</span><span class="co">]</span> and considered a pre-trained model as a starting point. Specifically, we trained a deep convolutional neural network (ResNet-50) architecture <span class="co">[</span><span class="ot">@he_deep_2016</span><span class="co">]</span> using the <span class="in">`fastai`</span> library (<span class="ot">&lt;https://docs.fast.ai/&gt;</span>) that implements the <span class="in">`PyTorch`</span> library <span class="co">[</span><span class="ot">@NEURIPS2019_9015</span><span class="co">]</span>. Interestingly, the <span class="in">`fastai`</span> library comes with an <span class="in">`R`</span> interface (<span class="ot">&lt;https://eagerai.github.io/fastai/&gt;</span>) that uses the <span class="in">`reticulate`</span> package to communicate with <span class="in">`Python`</span>, therefore allowing <span class="in">`R`</span> users to access up-to-date deep learning tools. We trained models on the Montpellier Bioinformatics Biodiversity platform using a GPU machine (Titan Xp nvidia) with 16Go of RAM. We used 20 epochs which took approximately 10 hours. The computational burden prevented us from providing a full reproducible analysis, but we do so with a subsample of the dataset in the Appendix. All trained models are available from <span class="ot">&lt;https://doi.org/10.5281/zenodo.5164796&gt;</span>.</span>
<span id="cb19-505"><a href="#cb19-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-506"><a href="#cb19-506" aria-hidden="true" tabindex="-1"></a>Using the testing dataset, we calculated three metrics to evaluate our model performance at correctly identifying species <span class="co">[</span><span class="ot">e.g. @Duggan2021</span><span class="co">]</span>. Specifically, we relied on *accuracy* the ratio of correct predictions to the total number of predictions, *recall* a measure of false negatives (FN; e.g. an image with a lynx for which our model predicts another species) with recall = TP / (TP + FN) where TP is for true positives, and *precision* a measure of false positives (FP; e.g. an image with any species but a lynx for which our model predicts a lynx) with precision = TP / (TP + FP). In camera trap studies, a strategy <span class="co">[</span><span class="ot">@Duggan2021</span><span class="co">]</span> consists in optimizing precision if the focus is on rare species (lynx), while recall should be optimized if the focus is on commom species (chamois and roe deer). </span>
<span id="cb19-507"><a href="#cb19-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-508"><a href="#cb19-508" aria-hidden="true" tabindex="-1"></a>We achieved 85\% accuracy during training. Our model had good performances for the three classes we were interested in, with 87\% precision for lynx and 81\% recall for both roe deer and chamois (@tbl-deer). </span>
<span id="cb19-509"><a href="#cb19-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-510"><a href="#cb19-510" aria-hidden="true" tabindex="-1"></a><span class="in">```{r perf, cache = FALSE}</span></span>
<span id="cb19-511"><a href="#cb19-511" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-deer</span></span>
<span id="cb19-512"><a href="#cb19-512" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Model performance metrics. Images from the Jura study site were used for training."</span></span>
<span id="cb19-513"><a href="#cb19-513" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">knitr.kable.NA =</span> <span class="st">''</span>)</span>
<span id="cb19-514"><a href="#cb19-514" aria-hidden="true" tabindex="-1"></a><span class="fu">read_csv2</span>(<span class="st">"dat/perf_fastai.csv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-515"><a href="#cb19-515" aria-hidden="true" tabindex="-1"></a>  <span class="fu">clean_names</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-516"><a href="#cb19-516" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-517"><a href="#cb19-517" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-518"><a href="#cb19-518" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-519"><a href="#cb19-519" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-520"><a href="#cb19-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-521"><a href="#cb19-521" aria-hidden="true" tabindex="-1"></a><span class="fu">## Transferability - Ain study site</span></span>
<span id="cb19-522"><a href="#cb19-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-523"><a href="#cb19-523" aria-hidden="true" tabindex="-1"></a>We evaluated transferability for our trained model by predicting species on images from the Ain study site which were not used for training. Precision was 77\% for lynx, and while we achieved 86\% recall for roe deer, our model performed poorly for chamois with 8\% recall (@tbl-chamois). </span>
<span id="cb19-524"><a href="#cb19-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-525"><a href="#cb19-525" aria-hidden="true" tabindex="-1"></a><span class="in">```{r transf,  cache = FALSE}</span></span>
<span id="cb19-526"><a href="#cb19-526" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-chamois</span></span>
<span id="cb19-527"><a href="#cb19-527" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Model transferability performance. Images from the Ain study site were used for assessing transferability."</span></span>
<span id="cb19-528"><a href="#cb19-528" aria-hidden="true" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">load</span>(<span class="st">"dat/pixAinannot.RData"</span>)</span>
<span id="cb19-529"><a href="#cb19-529" aria-hidden="true" tabindex="-1"></a><span class="do">## compute accuracy</span></span>
<span id="cb19-530"><a href="#cb19-530" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp &lt;- unclass(table(tt_final$predicted,tt_final$observed))</span></span>
<span id="cb19-531"><a href="#cb19-531" aria-hidden="true" tabindex="-1"></a><span class="co">#mask &lt;- colnames(tmp)[!colnames(tmp) %in% rownames(tmp)]</span></span>
<span id="cb19-532"><a href="#cb19-532" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp &lt;- tmp[,!colnames(tmp)%in%mask]</span></span>
<span id="cb19-533"><a href="#cb19-533" aria-hidden="true" tabindex="-1"></a><span class="co">#caret::confusionMatrix(tmp)</span></span>
<span id="cb19-534"><a href="#cb19-534" aria-hidden="true" tabindex="-1"></a>tt_final <span class="sc">%&gt;%</span></span>
<span id="cb19-535"><a href="#cb19-535" aria-hidden="true" tabindex="-1"></a><span class="co">#  add_count(observed, sort = TRUE) %&gt;%</span></span>
<span id="cb19-536"><a href="#cb19-536" aria-hidden="true" tabindex="-1"></a><span class="co">#  filter(n &gt; 200) %&gt;%</span></span>
<span id="cb19-537"><a href="#cb19-537" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">observed =</span> <span class="fu">fct_recode</span>(observed,</span>
<span id="cb19-538"><a href="#cb19-538" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"human"</span> <span class="ot">=</span> <span class="st">"humain"</span>,</span>
<span id="cb19-539"><a href="#cb19-539" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"dog"</span> <span class="ot">=</span> <span class="st">"chien"</span>,</span>
<span id="cb19-540"><a href="#cb19-540" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"red deer"</span> <span class="ot">=</span> <span class="st">"cerf"</span>,</span>
<span id="cb19-541"><a href="#cb19-541" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"fox"</span> <span class="ot">=</span> <span class="st">"renard"</span>,</span>
<span id="cb19-542"><a href="#cb19-542" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"wild boar"</span> <span class="ot">=</span> <span class="st">"sangliers"</span>,</span>
<span id="cb19-543"><a href="#cb19-543" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"roe deer"</span> <span class="ot">=</span> <span class="st">"chevreuil"</span>,</span>
<span id="cb19-544"><a href="#cb19-544" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"cat"</span> <span class="ot">=</span> <span class="st">"chat"</span>,</span>
<span id="cb19-545"><a href="#cb19-545" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"badger"</span> <span class="ot">=</span> <span class="st">"blaireaux"</span>,</span>
<span id="cb19-546"><a href="#cb19-546" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"hunter"</span> <span class="ot">=</span> <span class="st">"chasseur"</span>,</span>
<span id="cb19-547"><a href="#cb19-547" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"rider"</span> <span class="ot">=</span> <span class="st">"cavalier"</span>,</span>
<span id="cb19-548"><a href="#cb19-548" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"marten"</span> <span class="ot">=</span> <span class="st">"martre"</span>,</span>
<span id="cb19-549"><a href="#cb19-549" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"cow"</span> <span class="ot">=</span> <span class="st">"vaches"</span>,</span>
<span id="cb19-550"><a href="#cb19-550" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"hare"</span> <span class="ot">=</span> <span class="st">"lievre"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-551"><a href="#cb19-551" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">observed =</span> <span class="fu">fct_drop</span>(observed)) <span class="sc">%&gt;%</span></span>
<span id="cb19-552"><a href="#cb19-552" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">predicted =</span> <span class="fu">fct_recode</span>(predicted,</span>
<span id="cb19-553"><a href="#cb19-553" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"human"</span> <span class="ot">=</span> <span class="st">"humain"</span>,</span>
<span id="cb19-554"><a href="#cb19-554" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"dog"</span> <span class="ot">=</span> <span class="st">"chien"</span>,</span>
<span id="cb19-555"><a href="#cb19-555" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"fox"</span> <span class="ot">=</span> <span class="st">"renard"</span>,</span>
<span id="cb19-556"><a href="#cb19-556" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"wild boar"</span> <span class="ot">=</span> <span class="st">"sangliers"</span>,</span>
<span id="cb19-557"><a href="#cb19-557" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"roe deer"</span> <span class="ot">=</span> <span class="st">"chevreuil"</span>,</span>
<span id="cb19-558"><a href="#cb19-558" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"cat"</span> <span class="ot">=</span> <span class="st">"chat"</span>,</span>
<span id="cb19-559"><a href="#cb19-559" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"badger"</span> <span class="ot">=</span> <span class="st">"blaireaux"</span>,</span>
<span id="cb19-560"><a href="#cb19-560" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"hunter"</span> <span class="ot">=</span> <span class="st">"chasseur"</span>,</span>
<span id="cb19-561"><a href="#cb19-561" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"rider"</span> <span class="ot">=</span> <span class="st">"cavalier"</span>,</span>
<span id="cb19-562"><a href="#cb19-562" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"red deer"</span> <span class="ot">=</span> <span class="st">"cerf"</span>,</span>
<span id="cb19-563"><a href="#cb19-563" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"marten"</span> <span class="ot">=</span> <span class="st">"martre"</span>,</span>
<span id="cb19-564"><a href="#cb19-564" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"cow"</span> <span class="ot">=</span> <span class="st">"vaches"</span>,</span>
<span id="cb19-565"><a href="#cb19-565" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"hare"</span> <span class="ot">=</span> <span class="st">"lievre"</span>)) <span class="ot">-&gt;</span> tt_final_clean</span>
<span id="cb19-566"><a href="#cb19-566" aria-hidden="true" tabindex="-1"></a>dout <span class="ot">&lt;-</span> <span class="fu">unclass</span>(<span class="fu">table</span>(tt_final_clean<span class="sc">$</span>predicted, tt_final_clean<span class="sc">$</span>observed))</span>
<span id="cb19-567"><a href="#cb19-567" aria-hidden="true" tabindex="-1"></a>mask <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dout)[<span class="sc">!</span><span class="fu">colnames</span>(dout) <span class="sc">%in%</span> <span class="fu">rownames</span>(dout)]</span>
<span id="cb19-568"><a href="#cb19-568" aria-hidden="true" tabindex="-1"></a>dout <span class="ot">&lt;-</span> dout[,<span class="sc">!</span><span class="fu">colnames</span>(dout)<span class="sc">%in%</span>mask]</span>
<span id="cb19-569"><a href="#cb19-569" aria-hidden="true" tabindex="-1"></a>true_positives  <span class="ot">&lt;-</span> <span class="fu">diag</span>(dout)</span>
<span id="cb19-570"><a href="#cb19-570" aria-hidden="true" tabindex="-1"></a>false_positives <span class="ot">&lt;-</span> <span class="fu">colSums</span>(dout) <span class="sc">-</span> true_positives</span>
<span id="cb19-571"><a href="#cb19-571" aria-hidden="true" tabindex="-1"></a>false_negatives <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(dout) <span class="sc">-</span> true_positives</span>
<span id="cb19-572"><a href="#cb19-572" aria-hidden="true" tabindex="-1"></a>true_negatives  <span class="ot">&lt;-</span> <span class="fu">sum</span>(dout) <span class="sc">-</span> true_positives <span class="sc">-</span> false_positives <span class="sc">-</span> false_negatives</span>
<span id="cb19-573"><a href="#cb19-573" aria-hidden="true" tabindex="-1"></a>precision <span class="ot">&lt;-</span> true_positives<span class="sc">/</span>(true_positives<span class="sc">+</span>false_positives)</span>
<span id="cb19-574"><a href="#cb19-574" aria-hidden="true" tabindex="-1"></a>recall <span class="ot">&lt;-</span> true_positives<span class="sc">/</span>(true_positives<span class="sc">+</span>false_negatives)</span>
<span id="cb19-575"><a href="#cb19-575" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">precision =</span> <span class="fu">round</span>(precision,<span class="dv">2</span>),</span>
<span id="cb19-576"><a href="#cb19-576" aria-hidden="true" tabindex="-1"></a>           <span class="at">recall =</span> <span class="fu">round</span>(recall,<span class="dv">2</span>)) <span class="sc">%&gt;%</span> <span class="co">#, n=rowSums(dout)) %&gt;%</span></span>
<span id="cb19-577"><a href="#cb19-577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-578"><a href="#cb19-578" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-579"><a href="#cb19-579" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-580"><a href="#cb19-580" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-581"><a href="#cb19-581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-582"><a href="#cb19-582" aria-hidden="true" tabindex="-1"></a>To better understand this pattern, we display the results under the form of a confusion matrix that compares model classifications to manual classifications (@fig-confusion). There were a lot of false negatives for chamois, meaning that when a chamois was present in an image, it was often classified as another species by our model. </span>
<span id="cb19-583"><a href="#cb19-583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-584"><a href="#cb19-584" aria-hidden="true" tabindex="-1"></a>:::{.column-page}</span>
<span id="cb19-585"><a href="#cb19-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-586"><a href="#cb19-586" aria-hidden="true" tabindex="-1"></a><span class="in">```{r figure2}</span></span>
<span id="cb19-587"><a href="#cb19-587" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-confusion </span></span>
<span id="cb19-588"><a href="#cb19-588" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Confusion matrix comparing automatic to manual species classifications. Species that were predicted by our model are in columns, and species that are actually in the images are in rows. Column and row percentages are also provided at the bottom and right side of each cells respectively. An example of column percentage is as follows: of all pictures for which we predict a wild boar, 75.1% actually contained a wild boar. An example of row percentage is as follows: of all pictures in which we have a wild boar, we predict 94% of them to be badgers."</span></span>
<span id="cb19-589"><a href="#cb19-589" aria-hidden="true" tabindex="-1"></a><span class="co">#perf_prediction &lt;- hchart(dout, label = TRUE) %&gt;%</span></span>
<span id="cb19-590"><a href="#cb19-590" aria-hidden="true" tabindex="-1"></a><span class="co">#    hc_yAxis(title = list(text = 'Actual')) %&gt;%</span></span>
<span id="cb19-591"><a href="#cb19-591" aria-hidden="true" tabindex="-1"></a><span class="co">#    hc_xAxis(title = list(text = 'Predicted'),</span></span>
<span id="cb19-592"><a href="#cb19-592" aria-hidden="true" tabindex="-1"></a><span class="co">#             labels = list(rotation = -90)) </span></span>
<span id="cb19-593"><a href="#cb19-593" aria-hidden="true" tabindex="-1"></a><span class="co">#perf_prediction</span></span>
<span id="cb19-594"><a href="#cb19-594" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cvms)</span>
<span id="cb19-595"><a href="#cb19-595" aria-hidden="true" tabindex="-1"></a>d_binomial <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="st">'Actual'</span> <span class="ot">=</span> tt_final_clean<span class="sc">$</span>observed,</span>
<span id="cb19-596"><a href="#cb19-596" aria-hidden="true" tabindex="-1"></a>                     <span class="st">'Predicted'</span> <span class="ot">=</span> tt_final_clean<span class="sc">$</span>predicted)</span>
<span id="cb19-597"><a href="#cb19-597" aria-hidden="true" tabindex="-1"></a>basic_table <span class="ot">&lt;-</span> <span class="fu">table</span>(d_binomial)</span>
<span id="cb19-598"><a href="#cb19-598" aria-hidden="true" tabindex="-1"></a>cfm <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(basic_table)</span>
<span id="cb19-599"><a href="#cb19-599" aria-hidden="true" tabindex="-1"></a>cvms<span class="sc">::</span><span class="fu">plot_confusion_matrix</span>(cfm, </span>
<span id="cb19-600"><a href="#cb19-600" aria-hidden="true" tabindex="-1"></a>                            <span class="at">target_col =</span> <span class="st">"Actual"</span>, </span>
<span id="cb19-601"><a href="#cb19-601" aria-hidden="true" tabindex="-1"></a>                            <span class="at">prediction_col =</span> <span class="st">"Predicted"</span>,</span>
<span id="cb19-602"><a href="#cb19-602" aria-hidden="true" tabindex="-1"></a>                            <span class="at">counts_col =</span> <span class="st">"n"</span>, </span>
<span id="cb19-603"><a href="#cb19-603" aria-hidden="true" tabindex="-1"></a>                            <span class="at">add_normalized =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-604"><a href="#cb19-604" aria-hidden="true" tabindex="-1"></a>                            <span class="at">add_col_percentages =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-605"><a href="#cb19-605" aria-hidden="true" tabindex="-1"></a>                            <span class="at">add_row_percentages =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-606"><a href="#cb19-606" aria-hidden="true" tabindex="-1"></a>                            <span class="at">rm_zero_text =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-607"><a href="#cb19-607" aria-hidden="true" tabindex="-1"></a>                            <span class="at">rotate_y_text =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-608"><a href="#cb19-608" aria-hidden="true" tabindex="-1"></a>                            <span class="at">place_x_axis_above =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb19-609"><a href="#cb19-609" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Predicted"</span>, <span class="at">y =</span> <span class="st">"Actual"</span>) <span class="sc">+</span> </span>
<span id="cb19-610"><a href="#cb19-610" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb19-611"><a href="#cb19-611" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb19-612"><a href="#cb19-612" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-613"><a href="#cb19-613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-614"><a href="#cb19-614" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb19-615"><a href="#cb19-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-616"><a href="#cb19-616" aria-hidden="true" tabindex="-1"></a>Overall, our model trained on images from the Jura study site did poorly at correctly predicting species on images from the Ain study site. This result does not come as a surprise, as generalizing classification algorithms to new environments is known to be difficult <span class="co">[</span><span class="ot">@beery_recognition_2018</span><span class="co">]</span>. While a computer scientist might be disappointed in these results, an ecologist would probably wonder whether ecological inference about the co-occurrence between lynx and its prey is biased by these average performances, a question we address in the next section.  </span>
<span id="cb19-617"><a href="#cb19-617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-618"><a href="#cb19-618" aria-hidden="true" tabindex="-1"></a><span class="fu"># Spatial co-occurrence</span></span>
<span id="cb19-619"><a href="#cb19-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-620"><a href="#cb19-620" aria-hidden="true" tabindex="-1"></a>Here, we analysed the data we acquired from the previous section. For the sake of comparison, we considered two datasets, one made of the images manually labelled for both the Jura and Ain study sites pooled together (*ground truth dataset*), and the other in which we pooled the images that were manually labelled for the Jura study site and the images that were automatically labelled for the Ain study site using our trained model (*classified dataset*). </span>
<span id="cb19-621"><a href="#cb19-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-622"><a href="#cb19-622" aria-hidden="true" tabindex="-1"></a>We formatted the data by generating monthly detection histories, that is a sequence of detections ($Y_{sit} = 1$) and non-detections ($Y_{sit} = 0$), for species $s$ at site $i$ and sampling occasion $t$ (see @fig-occupancy).</span>
<span id="cb19-623"><a href="#cb19-623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-624"><a href="#cb19-624" aria-hidden="true" tabindex="-1"></a>:::{.column-page}</span>
<span id="cb19-625"><a href="#cb19-625" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-626"><a href="#cb19-626" aria-hidden="true" tabindex="-1"></a><span class="in">```{r figure3}</span></span>
<span id="cb19-627"><a href="#cb19-627" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-occupancy </span></span>
<span id="cb19-628"><a href="#cb19-628" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Detections (black) and non-detections (light grey) for each of the 3 species lynx, chamois and roe deer between March and November for all years pooled together. Sites are on the Y axis, while sampling occasions are on the X axis. Only data from the ground truth dataset are displayed."</span></span>
<span id="cb19-629"><a href="#cb19-629" aria-hidden="true" tabindex="-1"></a><span class="do">## Occupancy analyses</span></span>
<span id="cb19-630"><a href="#cb19-630" aria-hidden="true" tabindex="-1"></a><span class="do">## First with manually tagged data, then with automaticaly tagged data</span></span>
<span id="cb19-631"><a href="#cb19-631" aria-hidden="true" tabindex="-1"></a><span class="co"># my_packages &lt;- c("unmarked")</span></span>
<span id="cb19-632"><a href="#cb19-632" aria-hidden="true" tabindex="-1"></a><span class="co"># not_installed &lt;- my_packages[!(my_packages %in% installed.packages()[,"Package"])]</span></span>
<span id="cb19-633"><a href="#cb19-633" aria-hidden="true" tabindex="-1"></a><span class="co"># if(length(not_installed)) devtools::install_github("rbchan/unmarked")</span></span>
<span id="cb19-634"><a href="#cb19-634" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(unmarked)</span>
<span id="cb19-635"><a href="#cb19-635" aria-hidden="true" tabindex="-1"></a><span class="co"># Format the data as required for package unmarked</span></span>
<span id="cb19-636"><a href="#cb19-636" aria-hidden="true" tabindex="-1"></a>lynx <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain<span class="sc">$</span>lynx[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>lynx[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-637"><a href="#cb19-637" aria-hidden="true" tabindex="-1"></a>deer <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain<span class="sc">$</span>deer[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>deer[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-638"><a href="#cb19-638" aria-hidden="true" tabindex="-1"></a>chamois <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain<span class="sc">$</span>chamois[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>chamois[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-639"><a href="#cb19-639" aria-hidden="true" tabindex="-1"></a>fox <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain<span class="sc">$</span>fox[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>fox[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-640"><a href="#cb19-640" aria-hidden="true" tabindex="-1"></a>cat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain<span class="sc">$</span>cat[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>cat[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-641"><a href="#cb19-641" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">"lynx"</span> <span class="ot">=</span> lynx, <span class="st">"roe deer"</span> <span class="ot">=</span> deer, <span class="st">"chamois"</span> <span class="ot">=</span> chamois)</span>
<span id="cb19-642"><a href="#cb19-642" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">unmarkedFrameOccuMulti</span>(<span class="at">y=</span>y)</span>
<span id="cb19-643"><a href="#cb19-643" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(data)</span></span>
<span id="cb19-644"><a href="#cb19-644" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-645"><a href="#cb19-645" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> data</span>
<span id="cb19-646"><a href="#cb19-646" aria-hidden="true" tabindex="-1"></a>y1 <span class="ot">&lt;-</span> <span class="fu">getY</span>(x)</span>
<span id="cb19-647"><a href="#cb19-647" aria-hidden="true" tabindex="-1"></a>ym <span class="ot">&lt;-</span> <span class="fu">max</span>(y1, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb19-648"><a href="#cb19-648" aria-hidden="true" tabindex="-1"></a>M <span class="ot">&lt;-</span> <span class="fu">nrow</span>(y1)</span>
<span id="cb19-649"><a href="#cb19-649" aria-hidden="true" tabindex="-1"></a>J <span class="ot">&lt;-</span> <span class="fu">ncol</span>(y1)</span>
<span id="cb19-650"><a href="#cb19-650" aria-hidden="true" tabindex="-1"></a>S <span class="ot">&lt;-</span> <span class="fu">length</span>(x<span class="sc">@</span>ylist)</span>
<span id="cb19-651"><a href="#cb19-651" aria-hidden="true" tabindex="-1"></a>y1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">do.call</span>(rbind,x<span class="sc">@</span>ylist))</span>
<span id="cb19-652"><a href="#cb19-652" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(y1) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span>J</span>
<span id="cb19-653"><a href="#cb19-653" aria-hidden="true" tabindex="-1"></a>y1<span class="sc">$</span>site <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>M,S)</span>
<span id="cb19-654"><a href="#cb19-654" aria-hidden="true" tabindex="-1"></a>y1<span class="sc">$</span>species <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">rep</span>(<span class="fu">names</span>(x<span class="sc">@</span>ylist),<span class="at">each=</span>M))</span>
<span id="cb19-655"><a href="#cb19-655" aria-hidden="true" tabindex="-1"></a>y1 <span class="sc">%&gt;%</span></span>
<span id="cb19-656"><a href="#cb19-656" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>,</span>
<span id="cb19-657"><a href="#cb19-657" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"time"</span>,</span>
<span id="cb19-658"><a href="#cb19-658" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-659"><a href="#cb19-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">site =</span> <span class="fu">as_factor</span>(site),</span>
<span id="cb19-660"><a href="#cb19-660" aria-hidden="true" tabindex="-1"></a>         <span class="at">time =</span> <span class="fu">as_factor</span>(time),</span>
<span id="cb19-661"><a href="#cb19-661" aria-hidden="true" tabindex="-1"></a>         <span class="at">value =</span> <span class="fu">as_factor</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb19-662"><a href="#cb19-662" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb19-663"><a href="#cb19-663" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> time, <span class="at">y =</span> site, <span class="at">fill =</span> value) <span class="sc">+</span> </span>
<span id="cb19-664"><a href="#cb19-664" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb19-665"><a href="#cb19-665" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">name =</span> <span class="st">'Sites'</span>,</span>
<span id="cb19-666"><a href="#cb19-666" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">rownames</span>(x<span class="sc">@</span>ylist<span class="sc">$</span>lynx)) <span class="sc">+</span> </span>
<span id="cb19-667"><a href="#cb19-667" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(species), <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-668"><a href="#cb19-668" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>( <span class="st">"gray90"</span>,<span class="st">"black"</span>),</span>
<span id="cb19-669"><a href="#cb19-669" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"non-detection"</span>, <span class="st">"detection"</span>))<span class="sc">+</span></span>
<span id="cb19-670"><a href="#cb19-670" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">'Sampling occasions'</span>, <span class="at">fill =</span> <span class="st">""</span>)</span>
<span id="cb19-671"><a href="#cb19-671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-672"><a href="#cb19-672" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">lynx =</span> lynx, <span class="at">deer =</span> deer, <span class="at">chamois =</span> chamois, <span class="at">fox =</span> fox, <span class="at">cat =</span> cat)</span>
<span id="cb19-673"><a href="#cb19-673" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">unmarkedFrameOccuMulti</span>(<span class="at">y=</span>y)</span>
<span id="cb19-674"><a href="#cb19-674" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-675"><a href="#cb19-675" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-676"><a href="#cb19-676" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb19-677"><a href="#cb19-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-678"><a href="#cb19-678" aria-hidden="true" tabindex="-1"></a>To quantify spatial co-occurrence betwen lynx and its preys, we used a multispecies occupancy modeling approach <span class="co">[</span><span class="ot">@Rota2016</span><span class="co">]</span> implemented in the <span class="in">`R`</span> package <span class="in">`unmarked`</span> <span class="co">[</span><span class="ot">@unmarkedFiske</span><span class="co">]</span> within the maximum likelihood framework. The multispecies occupancy model assumes that observations $y_{sit}$, conditional on $Z_{si}$ the latent occupancy state of species $s$ at site $i$ are drawn from Bernoulli random variables $Y_{sit} | Z_{si} \sim \text{Bernoulli}(Z_{si}p_{sit})$ where $p_{sit}$ is the detection probability of species $s$ at site $i$ and sampling occasion $t$. Detection probabilities can be modeled as a function of site and/or sampling covariates, or the presence/absence of other species, but for the sake of illustration, we will make them only species-specific here. </span>
<span id="cb19-679"><a href="#cb19-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-680"><a href="#cb19-680" aria-hidden="true" tabindex="-1"></a>The latent occupancy states are assumed to be distributed as multivariate Bernoulli random variables <span class="co">[</span><span class="ot">@dai_multivariate_2013</span><span class="co">]</span>. Let us consider 2 species, species 1 and 2, then $Z_i = (Z_{i1}, Z_{i2}) \sim \text{multivariate Bernoulli}(\psi_{11}, \psi_{10}, \psi_{01}, \psi_{00})$ where $\psi_{11}$ is the probability that a site is occupied by both species 1 and 2, $\psi_{10}$ the probability that a site is occupied by species 1 but not 2, $\psi_{01}$ the probability that a site is occupied by species 2 but not 1, and $\psi_{00}$ the probability a site is occupied by none of them. Note that we considered species-specific only occupancy probabilities but these could be modeled as site-specific covariates. Marginal occupancy probabilities are obtained as $\Pr(Z_{i1}=1) = \psi_{11} + \psi_{10}$ and $\Pr(Z_{i2}=1) = \psi_{11} + \psi_{01}$. With this model, we may also infer co-occurrence by calculating conditional probabilities such as for example the probability of a site being occupied by species 2 conditional of species 1 with $\Pr(Z_{i2} = 1| Z_{i1} = 1) = \displaystyle{\frac{\psi_{11}}{\psi_{11}+\psi_{10}}}$. </span>
<span id="cb19-681"><a href="#cb19-681" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-682"><a href="#cb19-682" aria-hidden="true" tabindex="-1"></a>Despite its appeal and increasing use in ecology, multispecies occupancy models can be difficult to fit to real-world data in practice. First, these models are data-hungry and regularization methods <span class="co">[</span><span class="ot">@clipp2021</span><span class="co">]</span> are needed to avoid occupancy probabilities to be estimated at the boundary of the parameter space or with large uncertainty. Second, and this is true for any joint species distribution models, these models quickly become very complex with many parameters to be estimated when the number of species increases and co-occurrence is allowed between all species. Here, ecological expertise should be used to consider only meaningful species interactions and apply parsimony when parameterizing models. </span>
<span id="cb19-683"><a href="#cb19-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-684"><a href="#cb19-684" aria-hidden="true" tabindex="-1"></a>We now turn to the results obtained from a model with five species namely lynx, chamois, roe deer, fox and cat and co-occurrence allowed between lynx and chamois and roe deer only. </span>
<span id="cb19-685"><a href="#cb19-685" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-686"><a href="#cb19-686" aria-hidden="true" tabindex="-1"></a><span class="in">```{r include = FALSE, eval = TRUE}</span></span>
<span id="cb19-687"><a href="#cb19-687" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2022</span>)</span>
<span id="cb19-688"><a href="#cb19-688" aria-hidden="true" tabindex="-1"></a>kk <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb19-689"><a href="#cb19-689" aria-hidden="true" tabindex="-1"></a><span class="co"># y &lt;- list(lynx = lynx, deer = deer, chamois = chamois, fox = fox, cat = cat)</span></span>
<span id="cb19-690"><a href="#cb19-690" aria-hidden="true" tabindex="-1"></a><span class="co"># f1, f2, f3, f4, f5, f12, f13, f14, f15, f23, f24, f25, f34, f35, f45, ...</span></span>
<span id="cb19-691"><a href="#cb19-691" aria-hidden="true" tabindex="-1"></a>occFormulas <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>) </span>
<span id="cb19-692"><a href="#cb19-692" aria-hidden="true" tabindex="-1"></a>detFormulas <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>)</span>
<span id="cb19-693"><a href="#cb19-693" aria-hidden="true" tabindex="-1"></a>seed <span class="ot">&lt;-</span> <span class="dv">24762</span></span>
<span id="cb19-694"><a href="#cb19-694" aria-hidden="true" tabindex="-1"></a>inits <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="dv">12</span>)</span>
<span id="cb19-695"><a href="#cb19-695" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(seed)</span>
<span id="cb19-696"><a href="#cb19-696" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">occuMulti</span>(<span class="at">detformulas =</span> detFormulas,</span>
<span id="cb19-697"><a href="#cb19-697" aria-hidden="true" tabindex="-1"></a>                 <span class="at">stateformulas =</span> occFormulas,</span>
<span id="cb19-698"><a href="#cb19-698" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> data,</span>
<span id="cb19-699"><a href="#cb19-699" aria-hidden="true" tabindex="-1"></a>                 <span class="at">method =</span> <span class="st">"Nelder-Mead"</span>,</span>
<span id="cb19-700"><a href="#cb19-700" aria-hidden="true" tabindex="-1"></a>                 <span class="at">control =</span> <span class="fu">list</span>(<span class="at">maxit =</span> <span class="dv">5000</span>, <span class="at">reltol =</span> <span class="fl">1e-4</span>),</span>
<span id="cb19-701"><a href="#cb19-701" aria-hidden="true" tabindex="-1"></a>                 <span class="at">starts =</span> <span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">12</span>, <span class="at">mean =</span> inits, <span class="at">sd =</span> .<span class="dv">5</span>))</span>
<span id="cb19-702"><a href="#cb19-702" aria-hidden="true" tabindex="-1"></a><span class="co"># look for optimal penalty</span></span>
<span id="cb19-703"><a href="#cb19-703" aria-hidden="true" tabindex="-1"></a>optimal_penalty <span class="ot">&lt;-</span> <span class="fu">optimizePenalty</span>(fit,</span>
<span id="cb19-704"><a href="#cb19-704" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">penalties =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.02</span>, <span class="fl">0.04</span>, <span class="fl">0.08</span>, <span class="fl">0.16</span>, <span class="fl">0.32</span>, <span class="fl">0.64</span>, <span class="fl">1.28</span>, <span class="fl">2.56</span>, <span class="fl">5.12</span>), </span>
<span id="cb19-705"><a href="#cb19-705" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">k =</span> kk, </span>
<span id="cb19-706"><a href="#cb19-706" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">boot =</span> <span class="dv">1</span>)</span>
<span id="cb19-707"><a href="#cb19-707" aria-hidden="true" tabindex="-1"></a><span class="co"># refit model after fixing the best penalty</span></span>
<span id="cb19-708"><a href="#cb19-708" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">occuMulti</span>(<span class="at">detformulas =</span> detFormulas, </span>
<span id="cb19-709"><a href="#cb19-709" aria-hidden="true" tabindex="-1"></a>                 <span class="at">stateformulas =</span> occFormulas, </span>
<span id="cb19-710"><a href="#cb19-710" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> data,</span>
<span id="cb19-711"><a href="#cb19-711" aria-hidden="true" tabindex="-1"></a>                 <span class="at">control =</span> <span class="fu">list</span>(<span class="at">maxit =</span> <span class="dv">5000</span>),</span>
<span id="cb19-712"><a href="#cb19-712" aria-hidden="true" tabindex="-1"></a>                 <span class="at">penalty =</span> optimal_penalty<span class="sc">@</span>call<span class="sc">$</span>penalty) </span>
<span id="cb19-713"><a href="#cb19-713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-714"><a href="#cb19-714" aria-hidden="true" tabindex="-1"></a><span class="co"># Get natural parameter and detection estimates</span></span>
<span id="cb19-715"><a href="#cb19-715" aria-hidden="true" tabindex="-1"></a>mle <span class="ot">&lt;-</span> fit<span class="sc">@</span>opt<span class="sc">$</span>par</span>
<span id="cb19-716"><a href="#cb19-716" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mle) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'f1'</span>,<span class="st">'f2'</span>,<span class="st">'f3'</span>,<span class="st">'f4'</span>,<span class="st">'f5'</span>,<span class="st">'f12'</span>,<span class="st">'f13'</span>,<span class="st">'lp1'</span>,<span class="st">'lp2'</span>,<span class="st">'lp3'</span>,<span class="st">'lp4'</span>,<span class="st">'lp5'</span>)</span>
<span id="cb19-717"><a href="#cb19-717" aria-hidden="true" tabindex="-1"></a><span class="co"># Get occupancy estimates</span></span>
<span id="cb19-718"><a href="#cb19-718" aria-hidden="true" tabindex="-1"></a>psi_lynx <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-719"><a href="#cb19-719" aria-hidden="true" tabindex="-1"></a>psi_deer <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'deer'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-720"><a href="#cb19-720" aria-hidden="true" tabindex="-1"></a>psi_chamois <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'chamois'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-721"><a href="#cb19-721" aria-hidden="true" tabindex="-1"></a>psi_fox <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'fox'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-722"><a href="#cb19-722" aria-hidden="true" tabindex="-1"></a>psi_cat <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'cat'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-723"><a href="#cb19-723" aria-hidden="true" tabindex="-1"></a><span class="co"># With SEs</span></span>
<span id="cb19-724"><a href="#cb19-724" aria-hidden="true" tabindex="-1"></a>psi_lynx <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">"lynx"</span>), <span class="dv">1</span>)</span>
<span id="cb19-725"><a href="#cb19-725" aria-hidden="true" tabindex="-1"></a>psi_deer <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">"deer"</span>), <span class="dv">1</span>)</span>
<span id="cb19-726"><a href="#cb19-726" aria-hidden="true" tabindex="-1"></a>psi_chamois <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">"chamois"</span>), <span class="dv">1</span>)</span>
<span id="cb19-727"><a href="#cb19-727" aria-hidden="true" tabindex="-1"></a><span class="co"># Get detection estimates</span></span>
<span id="cb19-728"><a href="#cb19-728" aria-hidden="true" tabindex="-1"></a>p_lynx <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp1'</span>])</span>
<span id="cb19-729"><a href="#cb19-729" aria-hidden="true" tabindex="-1"></a>p_deer <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp2'</span>])</span>
<span id="cb19-730"><a href="#cb19-730" aria-hidden="true" tabindex="-1"></a>p_chamois <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp3'</span>])</span>
<span id="cb19-731"><a href="#cb19-731" aria-hidden="true" tabindex="-1"></a>p_fox <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp4'</span>])</span>
<span id="cb19-732"><a href="#cb19-732" aria-hidden="true" tabindex="-1"></a>p_cat <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp5'</span>])</span>
<span id="cb19-733"><a href="#cb19-733" aria-hidden="true" tabindex="-1"></a><span class="co"># With SEs</span></span>
<span id="cb19-734"><a href="#cb19-734" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">diag</span>(<span class="fu">solve</span>(fit<span class="sc">@</span>opt<span class="sc">$</span>hessian)))</span>
<span id="cb19-735"><a href="#cb19-735" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(se) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'f1'</span>,<span class="st">'f2'</span>,<span class="st">'f3'</span>,<span class="st">'f4'</span>,<span class="st">'f5'</span>,<span class="st">'f12'</span>,<span class="st">'f13'</span>,<span class="st">'lp1'</span>,<span class="st">'lp2'</span>,<span class="st">'lp3'</span>,<span class="st">'lp4'</span>,<span class="st">'lp5'</span>)</span>
<span id="cb19-736"><a href="#cb19-736" aria-hidden="true" tabindex="-1"></a>p_lynx1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mle =</span> p_lynx, </span>
<span id="cb19-737"><a href="#cb19-737" aria-hidden="true" tabindex="-1"></a>                     <span class="at">lci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp1'</span>] <span class="sc">-</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp1"</span>]), </span>
<span id="cb19-738"><a href="#cb19-738" aria-hidden="true" tabindex="-1"></a>                     <span class="at">uci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp1'</span>] <span class="sc">+</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp1"</span>]))</span>
<span id="cb19-739"><a href="#cb19-739" aria-hidden="true" tabindex="-1"></a>p_deer1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mle =</span> p_deer, </span>
<span id="cb19-740"><a href="#cb19-740" aria-hidden="true" tabindex="-1"></a>                     <span class="at">lci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp2'</span>] <span class="sc">-</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp2"</span>]), </span>
<span id="cb19-741"><a href="#cb19-741" aria-hidden="true" tabindex="-1"></a>                     <span class="at">uci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp2'</span>] <span class="sc">+</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp2"</span>]))</span>
<span id="cb19-742"><a href="#cb19-742" aria-hidden="true" tabindex="-1"></a>p_chamois1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mle =</span> p_chamois, </span>
<span id="cb19-743"><a href="#cb19-743" aria-hidden="true" tabindex="-1"></a>                        <span class="at">lci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp3'</span>] <span class="sc">-</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp3"</span>]), </span>
<span id="cb19-744"><a href="#cb19-744" aria-hidden="true" tabindex="-1"></a>                        <span class="at">uci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp3'</span>] <span class="sc">+</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp3"</span>]))</span>
<span id="cb19-745"><a href="#cb19-745" aria-hidden="true" tabindex="-1"></a><span class="co"># Conditional probabilities of occupancy</span></span>
<span id="cb19-746"><a href="#cb19-746" aria-hidden="true" tabindex="-1"></a><span class="co"># Pr(lynx present | roe deer and chamois present)</span></span>
<span id="cb19-747"><a href="#cb19-747" aria-hidden="true" tabindex="-1"></a>pc1 <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>,<span class="at">cond=</span><span class="fu">c</span>(<span class="st">'deer'</span>,<span class="st">'chamois'</span>))[<span class="dv">1</span>,]</span>
<span id="cb19-748"><a href="#cb19-748" aria-hidden="true" tabindex="-1"></a><span class="co"># Pr(lynx present | roe deer and chamois absent)</span></span>
<span id="cb19-749"><a href="#cb19-749" aria-hidden="true" tabindex="-1"></a>pc2 <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>,<span class="at">cond=</span><span class="fu">c</span>(<span class="st">'-deer'</span>,<span class="st">'-chamois'</span>))[<span class="dv">1</span>,]</span>
<span id="cb19-750"><a href="#cb19-750" aria-hidden="true" tabindex="-1"></a><span class="co"># Pr(lynx present | roe deer present and chamois absent)</span></span>
<span id="cb19-751"><a href="#cb19-751" aria-hidden="true" tabindex="-1"></a>pc3 <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>,<span class="at">cond=</span><span class="fu">c</span>(<span class="st">'deer'</span>,<span class="st">'-chamois'</span>))[<span class="dv">1</span>,]</span>
<span id="cb19-752"><a href="#cb19-752" aria-hidden="true" tabindex="-1"></a><span class="co"># Pr(lynx present | roe deer absent and chamois present)</span></span>
<span id="cb19-753"><a href="#cb19-753" aria-hidden="true" tabindex="-1"></a>pc4 <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>,<span class="at">cond=</span><span class="fu">c</span>(<span class="st">'-deer'</span>,<span class="st">'chamois'</span>))[<span class="dv">1</span>,]</span>
<span id="cb19-754"><a href="#cb19-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-755"><a href="#cb19-755" aria-hidden="true" tabindex="-1"></a>conditional1 <span class="ot">&lt;-</span> </span>
<span id="cb19-756"><a href="#cb19-756" aria-hidden="true" tabindex="-1"></a>  <span class="fu">structure</span>(<span class="fu">list</span>(<span class="at">param =</span> <span class="fu">c</span>(<span class="st">"Pr(lynx present | roe deer and chamois present)"</span>, </span>
<span id="cb19-757"><a href="#cb19-757" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer and chamois absent)"</span>, </span>
<span id="cb19-758"><a href="#cb19-758" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer present and chamois absent)"</span>,</span>
<span id="cb19-759"><a href="#cb19-759" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer absent and chamois present)"</span>), </span>
<span id="cb19-760"><a href="#cb19-760" aria-hidden="true" tabindex="-1"></a>                 <span class="at">low2.5 =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(pc1[<span class="dv">3</span>], pc2[<span class="dv">3</span>], pc3[<span class="dv">3</span>], pc4[<span class="dv">3</span>])), </span>
<span id="cb19-761"><a href="#cb19-761" aria-hidden="true" tabindex="-1"></a>                 <span class="at">est =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(pc1[<span class="dv">1</span>], pc2[<span class="dv">1</span>], pc3[<span class="dv">1</span>], pc4[<span class="dv">1</span>])), </span>
<span id="cb19-762"><a href="#cb19-762" aria-hidden="true" tabindex="-1"></a>                 <span class="at">up2.5 =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(pc1[<span class="dv">4</span>], pc2[<span class="dv">4</span>], pc3[<span class="dv">4</span>], pc4[<span class="dv">4</span>]))), </span>
<span id="cb19-763"><a href="#cb19-763" aria-hidden="true" tabindex="-1"></a>            <span class="at">.Names =</span> <span class="fu">c</span>(<span class="st">"param"</span>, <span class="st">"low2.5"</span>, <span class="st">"est"</span>, <span class="st">"up2.5"</span>), </span>
<span id="cb19-764"><a href="#cb19-764" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="dv">7</span><span class="sc">:</span><span class="dv">10</span>, </span>
<span id="cb19-765"><a href="#cb19-765" aria-hidden="true" tabindex="-1"></a>            <span class="at">class =</span> <span class="fu">c</span>(<span class="st">"data.frame"</span>, <span class="st">"mplus.params"</span>))</span>
<span id="cb19-766"><a href="#cb19-766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-767"><a href="#cb19-767" aria-hidden="true" tabindex="-1"></a>marginal1 <span class="ot">&lt;-</span> </span>
<span id="cb19-768"><a href="#cb19-768" aria-hidden="true" tabindex="-1"></a>  <span class="fu">structure</span>(<span class="fu">list</span>(<span class="at">param =</span> <span class="fu">c</span>(<span class="st">"Pr(lynx present)"</span>, <span class="st">"Pr(roe deer present)"</span>, <span class="st">"Pr(chamois present)"</span>), </span>
<span id="cb19-769"><a href="#cb19-769" aria-hidden="true" tabindex="-1"></a>                 <span class="at">low2.5 =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(psi_lynx[<span class="dv">3</span>], psi_deer[<span class="dv">3</span>], psi_chamois[<span class="dv">3</span>])), </span>
<span id="cb19-770"><a href="#cb19-770" aria-hidden="true" tabindex="-1"></a>                 <span class="at">est =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(psi_lynx[<span class="dv">1</span>], psi_deer[<span class="dv">1</span>], psi_chamois[<span class="dv">1</span>])), </span>
<span id="cb19-771"><a href="#cb19-771" aria-hidden="true" tabindex="-1"></a>                 <span class="at">up2.5 =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(psi_lynx[<span class="dv">4</span>], psi_deer[<span class="dv">4</span>], psi_chamois[<span class="dv">4</span>]))), </span>
<span id="cb19-772"><a href="#cb19-772" aria-hidden="true" tabindex="-1"></a>            <span class="at">.Names =</span> <span class="fu">c</span>(<span class="st">"param"</span>, <span class="st">"low2.5"</span>, <span class="st">"est"</span>, <span class="st">"up2.5"</span>), </span>
<span id="cb19-773"><a href="#cb19-773" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>, </span>
<span id="cb19-774"><a href="#cb19-774" aria-hidden="true" tabindex="-1"></a>            <span class="at">class =</span> <span class="fu">c</span>(<span class="st">"data.frame"</span>, <span class="st">"mplus.params"</span>))</span>
<span id="cb19-775"><a href="#cb19-775" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-776"><a href="#cb19-776" aria-hidden="true" tabindex="-1"></a><span class="co"># Format the data as required for package unmarked</span></span>
<span id="cb19-777"><a href="#cb19-777" aria-hidden="true" tabindex="-1"></a>lynx <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain_dl<span class="sc">$</span>lynx[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>lynx[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-778"><a href="#cb19-778" aria-hidden="true" tabindex="-1"></a>deer <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain_dl<span class="sc">$</span>deer[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>deer[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-779"><a href="#cb19-779" aria-hidden="true" tabindex="-1"></a>chamois <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain_dl<span class="sc">$</span>chamois[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>chamois[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-780"><a href="#cb19-780" aria-hidden="true" tabindex="-1"></a>fox <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain_dl<span class="sc">$</span>fox[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>fox[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-781"><a href="#cb19-781" aria-hidden="true" tabindex="-1"></a>cat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">rbind</span>(dat_ain_dl<span class="sc">$</span>cat[,<span class="sc">-</span><span class="dv">1</span>], dat_jura<span class="sc">$</span>cat[,<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb19-782"><a href="#cb19-782" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">lynx =</span> lynx, <span class="at">deer =</span> deer, <span class="at">chamois =</span> chamois, <span class="at">fox =</span> fox, <span class="at">cat =</span> cat)</span>
<span id="cb19-783"><a href="#cb19-783" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">unmarkedFrameOccuMulti</span>(<span class="at">y=</span>y)</span>
<span id="cb19-784"><a href="#cb19-784" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(data)</span></span>
<span id="cb19-785"><a href="#cb19-785" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(data)</span></span>
<span id="cb19-786"><a href="#cb19-786" aria-hidden="true" tabindex="-1"></a><span class="co"># f1, f2, f3, f4, f5, f12, f13, f14, f15, f23, f24, f25, f34, f35, f45, ...</span></span>
<span id="cb19-787"><a href="#cb19-787" aria-hidden="true" tabindex="-1"></a>occFormulas <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>,<span class="st">'~0'</span>) </span>
<span id="cb19-788"><a href="#cb19-788" aria-hidden="true" tabindex="-1"></a>detFormulas <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>,<span class="st">'~1'</span>)</span>
<span id="cb19-789"><a href="#cb19-789" aria-hidden="true" tabindex="-1"></a>seed <span class="ot">&lt;-</span> <span class="dv">4223</span></span>
<span id="cb19-790"><a href="#cb19-790" aria-hidden="true" tabindex="-1"></a>inits <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="dv">12</span>)</span>
<span id="cb19-791"><a href="#cb19-791" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(seed)</span>
<span id="cb19-792"><a href="#cb19-792" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">occuMulti</span>(<span class="at">detformulas =</span> detFormulas,</span>
<span id="cb19-793"><a href="#cb19-793" aria-hidden="true" tabindex="-1"></a>                 <span class="at">stateformulas =</span> occFormulas,</span>
<span id="cb19-794"><a href="#cb19-794" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> data,</span>
<span id="cb19-795"><a href="#cb19-795" aria-hidden="true" tabindex="-1"></a>                 <span class="at">method =</span> <span class="st">"Nelder-Mead"</span>,</span>
<span id="cb19-796"><a href="#cb19-796" aria-hidden="true" tabindex="-1"></a>                 <span class="at">control =</span> <span class="fu">list</span>(<span class="at">maxit =</span> <span class="dv">5000</span>, <span class="at">reltol =</span> <span class="fl">1e-4</span>),</span>
<span id="cb19-797"><a href="#cb19-797" aria-hidden="true" tabindex="-1"></a>                 <span class="at">starts =</span> <span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">12</span>, <span class="at">mean =</span> inits, <span class="at">sd =</span> .<span class="dv">5</span>))</span>
<span id="cb19-798"><a href="#cb19-798" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-799"><a href="#cb19-799" aria-hidden="true" tabindex="-1"></a><span class="co"># look for optimal penalty</span></span>
<span id="cb19-800"><a href="#cb19-800" aria-hidden="true" tabindex="-1"></a>optimal_penalty <span class="ot">&lt;-</span> <span class="fu">optimizePenalty</span>(<span class="at">object =</span> fit,</span>
<span id="cb19-801"><a href="#cb19-801" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">penalties =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.02</span>, <span class="fl">0.04</span>, <span class="fl">0.08</span>, <span class="fl">0.16</span>, <span class="fl">0.32</span>, <span class="fl">0.64</span>, <span class="fl">1.28</span>, <span class="fl">2.56</span>, <span class="fl">5.12</span>), </span>
<span id="cb19-802"><a href="#cb19-802" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">k =</span> kk, </span>
<span id="cb19-803"><a href="#cb19-803" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">boot =</span> <span class="dv">1</span>)</span>
<span id="cb19-804"><a href="#cb19-804" aria-hidden="true" tabindex="-1"></a><span class="co"># refit model after fixing the best penalty</span></span>
<span id="cb19-805"><a href="#cb19-805" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">occuMulti</span>(<span class="at">detformulas =</span> detFormulas, </span>
<span id="cb19-806"><a href="#cb19-806" aria-hidden="true" tabindex="-1"></a>                 <span class="at">stateformulas =</span> occFormulas, </span>
<span id="cb19-807"><a href="#cb19-807" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> data,</span>
<span id="cb19-808"><a href="#cb19-808" aria-hidden="true" tabindex="-1"></a>                 <span class="at">control =</span> <span class="fu">list</span>(<span class="at">maxit =</span> <span class="dv">5000</span>),</span>
<span id="cb19-809"><a href="#cb19-809" aria-hidden="true" tabindex="-1"></a>                 <span class="at">penalty =</span> optimal_penalty<span class="sc">@</span>call<span class="sc">$</span>penalty) </span>
<span id="cb19-810"><a href="#cb19-810" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-811"><a href="#cb19-811" aria-hidden="true" tabindex="-1"></a><span class="co"># Get natural parameter and detection estimates</span></span>
<span id="cb19-812"><a href="#cb19-812" aria-hidden="true" tabindex="-1"></a>mle <span class="ot">&lt;-</span> fit<span class="sc">@</span>opt<span class="sc">$</span>par</span>
<span id="cb19-813"><a href="#cb19-813" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mle) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'f1'</span>,<span class="st">'f2'</span>,<span class="st">'f3'</span>,<span class="st">'f4'</span>,<span class="st">'f5'</span>,<span class="st">'f12'</span>,<span class="st">'f13'</span>,<span class="st">'lp1'</span>,<span class="st">'lp2'</span>,<span class="st">'lp3'</span>,<span class="st">'lp4'</span>,<span class="st">'lp5'</span>)</span>
<span id="cb19-814"><a href="#cb19-814" aria-hidden="true" tabindex="-1"></a><span class="co"># Get occupancy estimates</span></span>
<span id="cb19-815"><a href="#cb19-815" aria-hidden="true" tabindex="-1"></a>psi_lynx <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-816"><a href="#cb19-816" aria-hidden="true" tabindex="-1"></a>psi_deer <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'deer'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-817"><a href="#cb19-817" aria-hidden="true" tabindex="-1"></a>psi_chamois <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'chamois'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-818"><a href="#cb19-818" aria-hidden="true" tabindex="-1"></a>psi_fox <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'fox'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-819"><a href="#cb19-819" aria-hidden="true" tabindex="-1"></a>psi_cat <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'cat'</span>)[<span class="dv">1</span>,]</span>
<span id="cb19-820"><a href="#cb19-820" aria-hidden="true" tabindex="-1"></a><span class="co"># With SEs</span></span>
<span id="cb19-821"><a href="#cb19-821" aria-hidden="true" tabindex="-1"></a>psi_lynx <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">"lynx"</span>), <span class="dv">1</span>)</span>
<span id="cb19-822"><a href="#cb19-822" aria-hidden="true" tabindex="-1"></a>psi_deer <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">"deer"</span>), <span class="dv">1</span>)</span>
<span id="cb19-823"><a href="#cb19-823" aria-hidden="true" tabindex="-1"></a>psi_chamois <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">"chamois"</span>), <span class="dv">1</span>)</span>
<span id="cb19-824"><a href="#cb19-824" aria-hidden="true" tabindex="-1"></a><span class="co"># Get detection estimates</span></span>
<span id="cb19-825"><a href="#cb19-825" aria-hidden="true" tabindex="-1"></a>p_lynx <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp1'</span>])</span>
<span id="cb19-826"><a href="#cb19-826" aria-hidden="true" tabindex="-1"></a>p_deer <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp2'</span>])</span>
<span id="cb19-827"><a href="#cb19-827" aria-hidden="true" tabindex="-1"></a>p_chamois <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp3'</span>])</span>
<span id="cb19-828"><a href="#cb19-828" aria-hidden="true" tabindex="-1"></a>p_fox <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp4'</span>])</span>
<span id="cb19-829"><a href="#cb19-829" aria-hidden="true" tabindex="-1"></a>p_cat <span class="ot">&lt;-</span> <span class="fu">plogis</span>(mle[<span class="st">'lp5'</span>])</span>
<span id="cb19-830"><a href="#cb19-830" aria-hidden="true" tabindex="-1"></a><span class="co"># With SEs</span></span>
<span id="cb19-831"><a href="#cb19-831" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">diag</span>(<span class="fu">solve</span>(fit<span class="sc">@</span>opt<span class="sc">$</span>hessian)))</span>
<span id="cb19-832"><a href="#cb19-832" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(se) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'f1'</span>,<span class="st">'f2'</span>,<span class="st">'f3'</span>,<span class="st">'f4'</span>,<span class="st">'f5'</span>,<span class="st">'f12'</span>,<span class="st">'f13'</span>,<span class="st">'lp1'</span>,<span class="st">'lp2'</span>,<span class="st">'lp3'</span>,<span class="st">'lp4'</span>,<span class="st">'lp5'</span>)</span>
<span id="cb19-833"><a href="#cb19-833" aria-hidden="true" tabindex="-1"></a>p_lynx2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mle =</span> p_lynx, </span>
<span id="cb19-834"><a href="#cb19-834" aria-hidden="true" tabindex="-1"></a>                     <span class="at">lci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp1'</span>] <span class="sc">-</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp1"</span>]), </span>
<span id="cb19-835"><a href="#cb19-835" aria-hidden="true" tabindex="-1"></a>                     <span class="at">uci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp1'</span>] <span class="sc">+</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp1"</span>]))</span>
<span id="cb19-836"><a href="#cb19-836" aria-hidden="true" tabindex="-1"></a>p_deer2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mle =</span> p_deer, </span>
<span id="cb19-837"><a href="#cb19-837" aria-hidden="true" tabindex="-1"></a>                     <span class="at">lci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp2'</span>] <span class="sc">-</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp2"</span>]), </span>
<span id="cb19-838"><a href="#cb19-838" aria-hidden="true" tabindex="-1"></a>                     <span class="at">uci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp2'</span>] <span class="sc">+</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp2"</span>]))</span>
<span id="cb19-839"><a href="#cb19-839" aria-hidden="true" tabindex="-1"></a>p_chamois2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">mle =</span> p_chamois, </span>
<span id="cb19-840"><a href="#cb19-840" aria-hidden="true" tabindex="-1"></a>                        <span class="at">lci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp3'</span>] <span class="sc">-</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp3"</span>]), </span>
<span id="cb19-841"><a href="#cb19-841" aria-hidden="true" tabindex="-1"></a>                        <span class="at">uci =</span> <span class="fu">plogis</span>(mle[<span class="st">'lp3'</span>] <span class="sc">+</span> <span class="fl">1.96</span> <span class="sc">*</span> se[<span class="st">"lp3"</span>]))</span>
<span id="cb19-842"><a href="#cb19-842" aria-hidden="true" tabindex="-1"></a><span class="co"># Conditional probabilities of occupancy</span></span>
<span id="cb19-843"><a href="#cb19-843" aria-hidden="true" tabindex="-1"></a><span class="co"># Pr(lynx present | roe deer and chamois present)</span></span>
<span id="cb19-844"><a href="#cb19-844" aria-hidden="true" tabindex="-1"></a>pc1 <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>,<span class="at">cond=</span><span class="fu">c</span>(<span class="st">'deer'</span>,<span class="st">'chamois'</span>))[<span class="dv">1</span>,]</span>
<span id="cb19-845"><a href="#cb19-845" aria-hidden="true" tabindex="-1"></a><span class="co"># Pr(lynx present | roe deer and chamois absent)</span></span>
<span id="cb19-846"><a href="#cb19-846" aria-hidden="true" tabindex="-1"></a>pc2 <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>,<span class="at">cond=</span><span class="fu">c</span>(<span class="st">'-deer'</span>,<span class="st">'-chamois'</span>))[<span class="dv">1</span>,]</span>
<span id="cb19-847"><a href="#cb19-847" aria-hidden="true" tabindex="-1"></a><span class="co"># Pr(lynx present | roe deer present and chamois absent)</span></span>
<span id="cb19-848"><a href="#cb19-848" aria-hidden="true" tabindex="-1"></a>pc3 <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>,<span class="at">cond=</span><span class="fu">c</span>(<span class="st">'deer'</span>,<span class="st">'-chamois'</span>))[<span class="dv">1</span>,]</span>
<span id="cb19-849"><a href="#cb19-849" aria-hidden="true" tabindex="-1"></a><span class="co"># Pr(lynx present | roe deer absent and chamois present)</span></span>
<span id="cb19-850"><a href="#cb19-850" aria-hidden="true" tabindex="-1"></a>pc4 <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit,<span class="st">'state'</span>,<span class="at">species=</span><span class="st">'lynx'</span>,<span class="at">cond=</span><span class="fu">c</span>(<span class="st">'-deer'</span>,<span class="st">'chamois'</span>))[<span class="dv">1</span>,]</span>
<span id="cb19-851"><a href="#cb19-851" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-852"><a href="#cb19-852" aria-hidden="true" tabindex="-1"></a>conditional2 <span class="ot">&lt;-</span> </span>
<span id="cb19-853"><a href="#cb19-853" aria-hidden="true" tabindex="-1"></a>  <span class="fu">structure</span>(<span class="fu">list</span>(<span class="at">param =</span> <span class="fu">c</span>(<span class="st">"Pr(lynx present | roe deer and chamois present)"</span>, </span>
<span id="cb19-854"><a href="#cb19-854" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer and chamois absent)"</span>, </span>
<span id="cb19-855"><a href="#cb19-855" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer present and chamois absent)"</span>,</span>
<span id="cb19-856"><a href="#cb19-856" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer absent and chamois present)"</span>), </span>
<span id="cb19-857"><a href="#cb19-857" aria-hidden="true" tabindex="-1"></a>                 <span class="at">low2.5 =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(pc1[<span class="dv">3</span>], pc2[<span class="dv">3</span>], pc3[<span class="dv">3</span>], pc4[<span class="dv">3</span>])), </span>
<span id="cb19-858"><a href="#cb19-858" aria-hidden="true" tabindex="-1"></a>                 <span class="at">est =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(pc1[<span class="dv">1</span>], pc2[<span class="dv">1</span>], pc3[<span class="dv">1</span>], pc4[<span class="dv">1</span>])), </span>
<span id="cb19-859"><a href="#cb19-859" aria-hidden="true" tabindex="-1"></a>                 <span class="at">up2.5 =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(pc1[<span class="dv">4</span>], pc2[<span class="dv">4</span>], pc3[<span class="dv">4</span>], pc4[<span class="dv">4</span>]))), </span>
<span id="cb19-860"><a href="#cb19-860" aria-hidden="true" tabindex="-1"></a>            <span class="at">.Names =</span> <span class="fu">c</span>(<span class="st">"param"</span>, <span class="st">"low2.5"</span>, <span class="st">"est"</span>, <span class="st">"up2.5"</span>), </span>
<span id="cb19-861"><a href="#cb19-861" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="dv">7</span><span class="sc">:</span><span class="dv">10</span>, </span>
<span id="cb19-862"><a href="#cb19-862" aria-hidden="true" tabindex="-1"></a>            <span class="at">class =</span> <span class="fu">c</span>(<span class="st">"data.frame"</span>, <span class="st">"mplus.params"</span>))</span>
<span id="cb19-863"><a href="#cb19-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-864"><a href="#cb19-864" aria-hidden="true" tabindex="-1"></a>marginal2 <span class="ot">&lt;-</span> </span>
<span id="cb19-865"><a href="#cb19-865" aria-hidden="true" tabindex="-1"></a>  <span class="fu">structure</span>(<span class="fu">list</span>(<span class="at">param =</span> <span class="fu">c</span>(<span class="st">"Pr(lynx present)"</span>, <span class="st">"Pr(roe deer present)"</span>, <span class="st">"Pr(chamois present)"</span>), </span>
<span id="cb19-866"><a href="#cb19-866" aria-hidden="true" tabindex="-1"></a>                 <span class="at">low2.5 =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(psi_lynx[<span class="dv">3</span>], psi_deer[<span class="dv">3</span>], psi_chamois[<span class="dv">3</span>])), </span>
<span id="cb19-867"><a href="#cb19-867" aria-hidden="true" tabindex="-1"></a>                 <span class="at">est =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(psi_lynx[<span class="dv">1</span>], psi_deer[<span class="dv">1</span>], psi_chamois[<span class="dv">1</span>])), </span>
<span id="cb19-868"><a href="#cb19-868" aria-hidden="true" tabindex="-1"></a>                 <span class="at">up2.5 =</span> <span class="fu">as_vector</span>(<span class="fu">c</span>(psi_lynx[<span class="dv">4</span>], psi_deer[<span class="dv">4</span>], psi_chamois[<span class="dv">4</span>]))), </span>
<span id="cb19-869"><a href="#cb19-869" aria-hidden="true" tabindex="-1"></a>            <span class="at">.Names =</span> <span class="fu">c</span>(<span class="st">"param"</span>, <span class="st">"low2.5"</span>, <span class="st">"est"</span>, <span class="st">"up2.5"</span>), </span>
<span id="cb19-870"><a href="#cb19-870" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>, </span>
<span id="cb19-871"><a href="#cb19-871" aria-hidden="true" tabindex="-1"></a>            <span class="at">class =</span> <span class="fu">c</span>(<span class="st">"data.frame"</span>, <span class="st">"mplus.params"</span>))</span>
<span id="cb19-872"><a href="#cb19-872" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-873"><a href="#cb19-873" aria-hidden="true" tabindex="-1"></a><span class="co"># Dataviz conditional probabilities</span></span>
<span id="cb19-874"><a href="#cb19-874" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(conditional1, conditional2)</span>
<span id="cb19-875"><a href="#cb19-875" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>model <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'ground truth dataset'</span>, <span class="st">'classified dataset'</span>), <span class="fu">c</span>(<span class="fu">nrow</span>(conditional1), <span class="fu">nrow</span>(conditional2)))</span>
<span id="cb19-876"><a href="#cb19-876" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>param_f <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>param, </span>
<span id="cb19-877"><a href="#cb19-877" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Pr(lynx present | roe deer and chamois present)"</span>, </span>
<span id="cb19-878"><a href="#cb19-878" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer and chamois absent)"</span>, </span>
<span id="cb19-879"><a href="#cb19-879" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer present and chamois absent)"</span>,</span>
<span id="cb19-880"><a href="#cb19-880" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer absent and chamois present)"</span>)),</span>
<span id="cb19-881"><a href="#cb19-881" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">"Pr(lynx present | roe deer and chamois present)"</span>, </span>
<span id="cb19-882"><a href="#cb19-882" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer and chamois absent)"</span>, </span>
<span id="cb19-883"><a href="#cb19-883" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer present and chamois absent)"</span>,</span>
<span id="cb19-884"><a href="#cb19-884" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"Pr(lynx present | roe deer absent and chamois present)"</span>)))</span>
<span id="cb19-885"><a href="#cb19-885" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-886"><a href="#cb19-886" aria-hidden="true" tabindex="-1"></a>labels <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb19-887"><a href="#cb19-887" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pr(lynx present | roe deer and chamois present)"</span> <span class="ot">=</span> <span class="st">"&lt;img src='img/light_lcc.jpg'</span></span>
<span id="cb19-888"><a href="#cb19-888" aria-hidden="true" tabindex="-1"></a><span class="st">    width='350' /&gt;&lt;br&gt;*Pr(lynx present | roe deer and chamois present)*"</span>,</span>
<span id="cb19-889"><a href="#cb19-889" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pr(lynx present | roe deer and chamois absent)"</span> <span class="ot">=</span> <span class="st">"&lt;img src='img/light_lnotcnotc.jpg'</span></span>
<span id="cb19-890"><a href="#cb19-890" aria-hidden="true" tabindex="-1"></a><span class="st">    width='350' /&gt;&lt;br&gt;*Pr(lynx present | roe deer and chamois absent)*"</span>,</span>
<span id="cb19-891"><a href="#cb19-891" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pr(lynx present | roe deer present and chamois absent)"</span> <span class="ot">=</span> <span class="st">"&lt;img src='img/light_lcnotc.jpg'</span></span>
<span id="cb19-892"><a href="#cb19-892" aria-hidden="true" tabindex="-1"></a><span class="st">    width='350' /&gt;&lt;br&gt;*Pr(lynx present | roe deer present and chamois absent)*"</span>,</span>
<span id="cb19-893"><a href="#cb19-893" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pr(lynx present | roe deer absent and chamois present)"</span> <span class="ot">=</span> <span class="st">"&lt;img src='img/light_lnotcc.jpg'</span></span>
<span id="cb19-894"><a href="#cb19-894" aria-hidden="true" tabindex="-1"></a><span class="st">    width='350' /&gt;&lt;br&gt;*Pr(lynx present | roe deer absent and chamois present)*"</span></span>
<span id="cb19-895"><a href="#cb19-895" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-896"><a href="#cb19-896" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-897"><a href="#cb19-897" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span></span>
<span id="cb19-898"><a href="#cb19-898" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb19-899"><a href="#cb19-899" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_pointrange</span>(<span class="fu">aes</span>(<span class="at">x =</span> est, </span>
<span id="cb19-900"><a href="#cb19-900" aria-hidden="true" tabindex="-1"></a>                      <span class="at">y =</span> param_f, </span>
<span id="cb19-901"><a href="#cb19-901" aria-hidden="true" tabindex="-1"></a>                      <span class="at">color =</span> model, </span>
<span id="cb19-902"><a href="#cb19-902" aria-hidden="true" tabindex="-1"></a>                      <span class="at">fill =</span> model, </span>
<span id="cb19-903"><a href="#cb19-903" aria-hidden="true" tabindex="-1"></a>                      <span class="at">xmax =</span> up2<span class="fl">.5</span>, </span>
<span id="cb19-904"><a href="#cb19-904" aria-hidden="true" tabindex="-1"></a>                      <span class="at">xmin =</span> low2<span class="fl">.5</span>), </span>
<span id="cb19-905"><a href="#cb19-905" aria-hidden="true" tabindex="-1"></a>                  <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> .<span class="dv">8</span>),</span>
<span id="cb19-906"><a href="#cb19-906" aria-hidden="true" tabindex="-1"></a>                  <span class="at">size =</span> <span class="fl">1.5</span>,</span>
<span id="cb19-907"><a href="#cb19-907" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pch =</span> <span class="dv">21</span>,</span>
<span id="cb19-908"><a href="#cb19-908" aria-hidden="true" tabindex="-1"></a>                  <span class="at">fill =</span> <span class="st">"white"</span>,</span>
<span id="cb19-909"><a href="#cb19-909" aria-hidden="true" tabindex="-1"></a>                  <span class="at">stroke =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb19-910"><a href="#cb19-910" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">wes_palette</span>(<span class="st">"Royal1"</span>)) <span class="sc">+</span> </span>
<span id="cb19-911"><a href="#cb19-911" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">name =</span> <span class="cn">NULL</span>,</span>
<span id="cb19-912"><a href="#cb19-912" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> labels) <span class="sc">+</span> </span>
<span id="cb19-913"><a href="#cb19-913" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb19-914"><a href="#cb19-914" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>, <span class="at">color =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb19-915"><a href="#cb19-915" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">reverse=</span>T)) <span class="sc">+</span> </span>
<span id="cb19-916"><a href="#cb19-916" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_markdown</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb19-917"><a href="#cb19-917" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>()) <span class="ot">-&gt;</span> conditional</span>
<span id="cb19-918"><a href="#cb19-918" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-919"><a href="#cb19-919" aria-hidden="true" tabindex="-1"></a><span class="co"># Dataviz marginal probabilities</span></span>
<span id="cb19-920"><a href="#cb19-920" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(marginal1, marginal2)</span>
<span id="cb19-921"><a href="#cb19-921" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>model <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">'ground truth dataset'</span>, <span class="st">'classified dataset'</span>), <span class="fu">c</span>(<span class="fu">nrow</span>(marginal1), <span class="fu">nrow</span>(marginal2)))</span>
<span id="cb19-922"><a href="#cb19-922" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>param_f <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>param, </span>
<span id="cb19-923"><a href="#cb19-923" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">'Pr(lynx present)'</span>, <span class="st">'Pr(roe deer present)'</span>, <span class="st">'Pr(chamois present)'</span>)),</span>
<span id="cb19-924"><a href="#cb19-924" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">'Pr(lynx present)'</span>, <span class="st">'Pr(roe deer present)'</span>, <span class="st">'Pr(chamois present)'</span>)))</span>
<span id="cb19-925"><a href="#cb19-925" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-926"><a href="#cb19-926" aria-hidden="true" tabindex="-1"></a>labels <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb19-927"><a href="#cb19-927" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pr(lynx present)"</span> <span class="ot">=</span> <span class="st">"&lt;img src='img/lynx.jpg'</span></span>
<span id="cb19-928"><a href="#cb19-928" aria-hidden="true" tabindex="-1"></a><span class="st">    width='200' /&gt;&lt;br&gt;*Pr(lynx present)*"</span>,</span>
<span id="cb19-929"><a href="#cb19-929" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pr(roe deer present)"</span> <span class="ot">=</span> <span class="st">"&lt;img src='img/deer.jpg'</span></span>
<span id="cb19-930"><a href="#cb19-930" aria-hidden="true" tabindex="-1"></a><span class="st">    width='200' /&gt;&lt;br&gt;*Pr(roe deer present)*"</span>,</span>
<span id="cb19-931"><a href="#cb19-931" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pr(chamois present)"</span> <span class="ot">=</span> <span class="st">"&lt;img src='img/chamois.jpg'</span></span>
<span id="cb19-932"><a href="#cb19-932" aria-hidden="true" tabindex="-1"></a><span class="st">    width='200' /&gt;&lt;br&gt;*Pr(chamois present)*"</span></span>
<span id="cb19-933"><a href="#cb19-933" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-934"><a href="#cb19-934" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-935"><a href="#cb19-935" aria-hidden="true" tabindex="-1"></a><span class="co"># change JPG into jpg</span></span>
<span id="cb19-936"><a href="#cb19-936" aria-hidden="true" tabindex="-1"></a><span class="co">#files &lt;- list.files(path = "pix/pixJura/", pattern = "*.JPG", full.names = TRUE)</span></span>
<span id="cb19-937"><a href="#cb19-937" aria-hidden="true" tabindex="-1"></a><span class="co">#newfiles &lt;- gsub(".JPG$", ".jpg", files)</span></span>
<span id="cb19-938"><a href="#cb19-938" aria-hidden="true" tabindex="-1"></a><span class="co">#file.rename(files, newfiles)</span></span>
<span id="cb19-939"><a href="#cb19-939" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-940"><a href="#cb19-940" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-941"><a href="#cb19-941" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span></span>
<span id="cb19-942"><a href="#cb19-942" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb19-943"><a href="#cb19-943" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_pointrange</span>(<span class="fu">aes</span>(<span class="at">x =</span> est, </span>
<span id="cb19-944"><a href="#cb19-944" aria-hidden="true" tabindex="-1"></a>                      <span class="at">y =</span> param_f, </span>
<span id="cb19-945"><a href="#cb19-945" aria-hidden="true" tabindex="-1"></a>                      <span class="at">color =</span> model, </span>
<span id="cb19-946"><a href="#cb19-946" aria-hidden="true" tabindex="-1"></a>                      <span class="at">fill =</span> model, </span>
<span id="cb19-947"><a href="#cb19-947" aria-hidden="true" tabindex="-1"></a>                      <span class="at">xmax =</span> up2<span class="fl">.5</span>, </span>
<span id="cb19-948"><a href="#cb19-948" aria-hidden="true" tabindex="-1"></a>                      <span class="at">xmin =</span> low2<span class="fl">.5</span>), </span>
<span id="cb19-949"><a href="#cb19-949" aria-hidden="true" tabindex="-1"></a>                  <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> .<span class="dv">8</span>),</span>
<span id="cb19-950"><a href="#cb19-950" aria-hidden="true" tabindex="-1"></a>                  <span class="at">size =</span> <span class="fl">1.5</span>,</span>
<span id="cb19-951"><a href="#cb19-951" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pch =</span> <span class="dv">21</span>,</span>
<span id="cb19-952"><a href="#cb19-952" aria-hidden="true" tabindex="-1"></a>                  <span class="at">fill =</span> <span class="st">"white"</span>,</span>
<span id="cb19-953"><a href="#cb19-953" aria-hidden="true" tabindex="-1"></a>                  <span class="at">stroke =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb19-954"><a href="#cb19-954" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> wesanderson<span class="sc">::</span><span class="fu">wes_palette</span>(<span class="st">"Royal1"</span>)) <span class="sc">+</span> </span>
<span id="cb19-955"><a href="#cb19-955" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">name =</span> <span class="cn">NULL</span>,</span>
<span id="cb19-956"><a href="#cb19-956" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> labels) <span class="sc">+</span> </span>
<span id="cb19-957"><a href="#cb19-957" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="fl">0.15</span>, <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb19-958"><a href="#cb19-958" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>, <span class="at">color =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb19-959"><a href="#cb19-959" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">reverse=</span>T)) <span class="sc">+</span> </span>
<span id="cb19-960"><a href="#cb19-960" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_markdown</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">20</span>),</span>
<span id="cb19-961"><a href="#cb19-961" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb19-962"><a href="#cb19-962" aria-hidden="true" tabindex="-1"></a>  ) <span class="ot">-&gt;</span> marginal</span>
<span id="cb19-963"><a href="#cb19-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-964"><a href="#cb19-964" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"img/conditional_occ_penalized.png"</span>, </span>
<span id="cb19-965"><a href="#cb19-965" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> conditional,</span>
<span id="cb19-966"><a href="#cb19-966" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">15</span>,</span>
<span id="cb19-967"><a href="#cb19-967" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb19-968"><a href="#cb19-968" aria-hidden="true" tabindex="-1"></a>       <span class="at">dpi =</span> <span class="dv">600</span>)</span>
<span id="cb19-969"><a href="#cb19-969" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-970"><a href="#cb19-970" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"img/marginal_occ_penalized.png"</span>, </span>
<span id="cb19-971"><a href="#cb19-971" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> marginal,</span>
<span id="cb19-972"><a href="#cb19-972" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">15</span>,</span>
<span id="cb19-973"><a href="#cb19-973" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb19-974"><a href="#cb19-974" aria-hidden="true" tabindex="-1"></a>       <span class="at">dpi =</span> <span class="dv">600</span>)</span>
<span id="cb19-975"><a href="#cb19-975" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-976"><a href="#cb19-976" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-977"><a href="#cb19-977" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-978"><a href="#cb19-978" aria-hidden="true" tabindex="-1"></a>Detection probabilities were indistinguishable (at the third decimal) whether we used the ground truth or the classified dataset, with $p_{\text{lynx}} = 0.51 (0.45, 0.58)$, $p_{\text{roe deer}} = 0.63 (0.57, 0.68)$ and $p_{\text{chamois}} = 0.61 (0.55, 0.67)$.</span>
<span id="cb19-979"><a href="#cb19-979" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-980"><a href="#cb19-980" aria-hidden="true" tabindex="-1"></a>We also found that occupancy probability estimates were similar whether we used the ground truth or the classified dataset (@fig-four). Roe deer was the most prevalent species, but lynx and chamois were also occurring with high probability (@fig-four). Note that, despite chamois being often misclassified (@fig-confusion), its marginal occupancy tends to be higher when estimated with the classified dataset. Ecologically speaking, this might well be the case if the correctly classified detections are spread over all camera traps. The difference in marginal occupancy seems however non-significant judging by the overlap between the two confidence intervals.</span>
<span id="cb19-981"><a href="#cb19-981" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-982"><a href="#cb19-982" aria-hidden="true" tabindex="-1"></a>:::{.column-page}</span>
<span id="cb19-983"><a href="#cb19-983" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-984"><a href="#cb19-984" aria-hidden="true" tabindex="-1"></a><span class="in">```{r figure4}</span></span>
<span id="cb19-985"><a href="#cb19-985" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-four</span></span>
<span id="cb19-986"><a href="#cb19-986" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Marginal occupancy probabilities for all three species (lynx, roe deer and chamois). Parameter estimates are from a multispecies occupancy model using either the ground truth dataset (in red) or the classified dataset (in blue-grey). Note that marginal occupancy probabilities are estimated with high precision for roe deer, which explain why the associated confidence intervals do not show."</span></span>
<span id="cb19-987"><a href="#cb19-987" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"img/marginal_occ_penalized.png"</span>)</span>
<span id="cb19-988"><a href="#cb19-988" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-989"><a href="#cb19-989" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-990"><a href="#cb19-990" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb19-991"><a href="#cb19-991" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-992"><a href="#cb19-992" aria-hidden="true" tabindex="-1"></a>Because marginal occupancy probabilities were high, probabilities of co-occurrence were also estimated high (@fig-five). Our results should be interpreted bearing in mind that co-occurrence is a necessary but not sufficient condition for actual interaction. When both preys were present, lynx was more present than when they were both absent (@fig-five). Lynx was more sensitive to the presence of roe deer than that of chamois (@fig-five). </span>
<span id="cb19-993"><a href="#cb19-993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-994"><a href="#cb19-994" aria-hidden="true" tabindex="-1"></a>:::{.column-page}</span>
<span id="cb19-995"><a href="#cb19-995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-996"><a href="#cb19-996" aria-hidden="true" tabindex="-1"></a><span class="in">```{r figure5}</span></span>
<span id="cb19-997"><a href="#cb19-997" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-five</span></span>
<span id="cb19-998"><a href="#cb19-998" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Lynx occupancy probability conditional on the presence or absence of its preys (roe deer and chamois). Parameter estimates are from a multispecies occupancy model using either the ground truth dataset (in red) or the classified dataset (in blue-grey)."</span></span>
<span id="cb19-999"><a href="#cb19-999" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"img/conditional_occ_penalized.png"</span>)</span>
<span id="cb19-1000"><a href="#cb19-1000" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1001"><a href="#cb19-1001" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1002"><a href="#cb19-1002" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb19-1003"><a href="#cb19-1003" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1004"><a href="#cb19-1004" aria-hidden="true" tabindex="-1"></a>Overall, we found similar or higher uncertainty in estimates obtained from the classified dataset (@fig-four and @fig-five). Sample size being similar for both datasets, we do not have a solid explanation for this pattern.  </span>
<span id="cb19-1005"><a href="#cb19-1005" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1006"><a href="#cb19-1006" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion</span></span>
<span id="cb19-1007"><a href="#cb19-1007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1008"><a href="#cb19-1008" aria-hidden="true" tabindex="-1"></a>In this paper, we aimed at illustrating a reproducible workflow for studying the structure of an animal community and species spatial co-occurrence (*why*) using images acquired from camera traps and automatically labelled with deep learning (*what*) which we analysed with statistical occupancy models accounting for imperfect species detection (*how*). Overall, we found that, even though model transferability could be improved, inference about the co-occurrence of lynx and its preys was similar whether we analysed the ground truth data or classified data. </span>
<span id="cb19-1009"><a href="#cb19-1009" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1010"><a href="#cb19-1010" aria-hidden="true" tabindex="-1"></a>This result calls for further work on the trade-offs between time and resources allocated to train models with deep learning and our ability to correctly answer key ecological questions with camera-trap surveys. In other words, while a computer scientist might be keen on spending time training models to achieve top performances, an ecologist would rather rely on a model showing average performances and use this time to proceed with statistical analyses if, of course, errors in computer-annotated images do not make ecological inference flawed. The right balance may be found with collaborative projects in which scientists from artificial intelligence, statistics and ecology agree on a common objective, and identify research questions that can pick the interest of all parties. </span>
<span id="cb19-1011"><a href="#cb19-1011" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1012"><a href="#cb19-1012" aria-hidden="true" tabindex="-1"></a>Our demonstration remains however empirical, and ecological inference might no longer be robust to misclassification if detection and non-detections were pooled weekly or daily, or if more complex models, e.g. including time-varying detection probabilities and/or habitat-specific occupancy probabilities, were fitted to the data. Therefore, we encourage others to try and replicate our results. In that spirit, we praise previous work on plants which used deep learning to produce occurrence data and tested the sensitivity of species distribution models to image classification errors <span class="co">[</span><span class="ot">@botella2018</span><span class="co">]</span>. We also see two avenues of research that could benefit the integration of deep learning and ecological statistics. First, a simulation study could be conducted to evaluate bias and precision in ecological parameter estimators with regard to errors in image annotation by computers. The outcome of this exercise could be, for example, guidelines informing on the confidence an investigator may place in ecological inference as a function of the amount of false negatives and false positives. Second, annotation errors could be accomodated directly in statistical models. For example, single-species occupancy models account for false negatives when a species is not detected by the camera at a site where it is present, as well as false positives when a species is detected at a site where it is not present due to species misidentification by the observer <span class="co">[</span><span class="ot">@miller2011</span><span class="co">]</span>. Pending a careful distinction between ecological vs. computer-generated false negatives and false positives, error rates could be added to multispecies occupancy models <span class="co">[</span><span class="ot">@chambert2018</span><span class="co">]</span> and informed by recall and precision metrics obtained during model training <span class="co">[</span><span class="ot">@tabak_improving_2020</span><span class="co">]</span>. An alternative quick and dirty approach would consist in adopting a Monte Carlo approach by sampling the species detected or non-detected in each picture according to its predicted probability of belonging to a given class, then building the corresponding dataset and fitting occupancy models to it for each sample. </span>
<span id="cb19-1013"><a href="#cb19-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1014"><a href="#cb19-1014" aria-hidden="true" tabindex="-1"></a>When it comes to the case study, our results should be discussed with regard to the sensitivity of co-occurrence estimates to errors in automatic species classification. In particular, we expected that confusions between the two prey species might artificially increase the estimated probability of co-occurrence with lynx. This was illustrated by $\Pr(\text{lynx present} | \text{roe deer present and chamois absent})$ (resp. $\Pr(\text{lynx present} | \text{roe deer absent and chamois present})$) being estimated higher (resp. lower) with the classified than the ground truth dataset (@fig-five). This pattern could be explained by chamois being often classified as (and confused with) roe deer (@fig-confusion). </span>
<span id="cb19-1015"><a href="#cb19-1015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1016"><a href="#cb19-1016" aria-hidden="true" tabindex="-1"></a>Our results are only preliminary and we see several perspectives to our work. First, we focused our analysis on lynx and its main prey, while other species should be included to get a better understanding of the community structure. For example, both lynx and fox prey on small rodents and birds and a model including co-occurrence between these two predators showed better support by the data (AIC was 1544 when co-occurrence was included vs. 1557 when it was not). Second, we aim at quantifying the relative contribution of biotic (lynx predation on chamois and roe deer) and abiotic (habitat quality) processes to the composition and dynamic of this ecological community. Third, to benefit future camera trap studies of lynx in the Jura mountains, we plan to train a model again using more manually annotated images from both the Jura and the Ain study sites. These perspectives are the object of ongoing work. </span>
<span id="cb19-1017"><a href="#cb19-1017" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1018"><a href="#cb19-1018" aria-hidden="true" tabindex="-1"></a>With the rapid advances in technologies for biodiversity monitoring <span class="co">[</span><span class="ot">@lahoz-monfort_comprehensive_2021</span><span class="co">]</span>, the possibility of analysing large amounts of images makes deep learning appealing to ecologists. We hope that our proposal of a reproducible <span class="in">`R`</span> workflow for deep learning and statistical ecology will encourage further studies in the integration of these disciplines, and contribute to the adoption of computer vision by ecologists.</span>
<span id="cb19-1019"><a href="#cb19-1019" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1020"><a href="#cb19-1020" aria-hidden="true" tabindex="-1"></a><span class="fu"># Appendix: Reproducible example of species identification on camera trap images with CPU</span></span>
<span id="cb19-1021"><a href="#cb19-1021" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1022"><a href="#cb19-1022" aria-hidden="true" tabindex="-1"></a>In this section, we go through a reproducible example of the entire deep learning workflow, including data preparation, model training, and automatic labeling of new images. We used a subsample of 467 images from the original dataset in the Jura county to allow the training of our model with CPU on a personal computer. We also used 14 images from the original dataset in the Ain county to illustrate prediction.</span>
<span id="cb19-1023"><a href="#cb19-1023" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1024"><a href="#cb19-1024" aria-hidden="true" tabindex="-1"></a><span class="fu">## Training and validation datasets</span></span>
<span id="cb19-1025"><a href="#cb19-1025" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1026"><a href="#cb19-1026" aria-hidden="true" tabindex="-1"></a>We first split the dataset of Jura images in two datasets, a dataset for training, and the other one for validation. We use the <span class="in">`exifr`</span> package to extract metadata from images, get a list of images names and extract the species from these.</span>
<span id="cb19-1027"><a href="#cb19-1027" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1028"><a href="#cb19-1028" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo=TRUE}</span></span>
<span id="cb19-1029"><a href="#cb19-1029" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-datasets</span></span>
<span id="cb19-1030"><a href="#cb19-1030" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Species considered, and number of images with these species in them."</span></span>
<span id="cb19-1031"><a href="#cb19-1031" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(exifr)</span>
<span id="cb19-1032"><a href="#cb19-1032" aria-hidden="true" tabindex="-1"></a>pix_folder <span class="ot">&lt;-</span> <span class="st">'pix/pixJura/'</span></span>
<span id="cb19-1033"><a href="#cb19-1033" aria-hidden="true" tabindex="-1"></a>file_list <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> pix_folder,</span>
<span id="cb19-1034"><a href="#cb19-1034" aria-hidden="true" tabindex="-1"></a>                        <span class="at">recursive =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-1035"><a href="#cb19-1035" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pattern =</span> <span class="st">"*.jpg"</span>,</span>
<span id="cb19-1036"><a href="#cb19-1036" aria-hidden="true" tabindex="-1"></a>                        <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-1037"><a href="#cb19-1037" aria-hidden="true" tabindex="-1"></a>labels <span class="ot">&lt;-</span></span>
<span id="cb19-1038"><a href="#cb19-1038" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_exif</span>(file_list) <span class="sc">%&gt;%</span></span>
<span id="cb19-1039"><a href="#cb19-1039" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-1040"><a href="#cb19-1040" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(Keywords, <span class="at">keep_empty =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="co"># keep_empty = TRUE keeps pix with no labels (empty pix)</span></span>
<span id="cb19-1041"><a href="#cb19-1041" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SourceFile) <span class="sc">%&gt;%</span></span>
<span id="cb19-1042"><a href="#cb19-1042" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>() <span class="sc">%&gt;%</span> <span class="co"># when several labels in a pix, keep first only</span></span>
<span id="cb19-1043"><a href="#cb19-1043" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-1044"><a href="#cb19-1044" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> <span class="fu">as_factor</span>(Keywords)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1045"><a href="#cb19-1045" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> <span class="fu">fct_explicit_na</span>(Keywords, <span class="st">"wo_tag"</span>)) <span class="sc">%&gt;%</span> <span class="co"># when pix has no tag</span></span>
<span id="cb19-1046"><a href="#cb19-1046" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SourceFile, FileName, Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb19-1047"><a href="#cb19-1047" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Keywords =</span> <span class="fu">fct_recode</span>(Keywords,</span>
<span id="cb19-1048"><a href="#cb19-1048" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"chat"</span> <span class="ot">=</span> <span class="st">"chat forestier"</span>,</span>
<span id="cb19-1049"><a href="#cb19-1049" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"lievre"</span> <span class="ot">=</span> <span class="st">"lièvre"</span>,</span>
<span id="cb19-1050"><a href="#cb19-1050" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"vehicule"</span> <span class="ot">=</span> <span class="st">"véhicule"</span>,</span>
<span id="cb19-1051"><a href="#cb19-1051" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"ni"</span> <span class="ot">=</span> <span class="st">"Non identifié"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1052"><a href="#cb19-1052" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(Keywords <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ni"</span>, <span class="st">"wo_tag"</span>)))</span>
<span id="cb19-1053"><a href="#cb19-1053" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1054"><a href="#cb19-1054" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo=FALSE}</span></span>
<span id="cb19-1055"><a href="#cb19-1055" aria-hidden="true" tabindex="-1"></a>labels <span class="sc">%&gt;%</span></span>
<span id="cb19-1056"><a href="#cb19-1056" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Keywords, <span class="at">sort =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1057"><a href="#cb19-1057" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-1058"><a href="#cb19-1058" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span>
<span id="cb19-1059"><a href="#cb19-1059" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1060"><a href="#cb19-1060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1061"><a href="#cb19-1061" aria-hidden="true" tabindex="-1"></a>Then we pick 80$\%$ of the images for training in each category, the rest being used for validation.</span>
<span id="cb19-1062"><a href="#cb19-1062" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1063"><a href="#cb19-1063" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo = TRUE}</span></span>
<span id="cb19-1064"><a href="#cb19-1064" aria-hidden="true" tabindex="-1"></a><span class="co"># training dataset</span></span>
<span id="cb19-1065"><a href="#cb19-1065" aria-hidden="true" tabindex="-1"></a>pix_train <span class="ot">&lt;-</span> labels <span class="sc">%&gt;%</span></span>
<span id="cb19-1066"><a href="#cb19-1066" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SourceFile, FileName, Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb19-1067"><a href="#cb19-1067" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb19-1068"><a href="#cb19-1068" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(<span class="fu">row_number</span>(), <span class="dv">1</span>, <span class="fu">floor</span>(<span class="fu">n</span>()<span class="sc">*</span><span class="dv">80</span><span class="sc">/</span><span class="dv">100</span>))) <span class="co"># 80% per category</span></span>
<span id="cb19-1069"><a href="#cb19-1069" aria-hidden="true" tabindex="-1"></a><span class="co"># validation dataset</span></span>
<span id="cb19-1070"><a href="#cb19-1070" aria-hidden="true" tabindex="-1"></a>pix_valid <span class="ot">&lt;-</span> labels <span class="sc">%&gt;%</span></span>
<span id="cb19-1071"><a href="#cb19-1071" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb19-1072"><a href="#cb19-1072" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(<span class="fu">row_number</span>(), <span class="fu">floor</span>(<span class="fu">n</span>()<span class="sc">*</span><span class="dv">80</span><span class="sc">/</span><span class="dv">100</span>) <span class="sc">+</span> <span class="dv">1</span>, <span class="fu">n</span>()))</span>
<span id="cb19-1073"><a href="#cb19-1073" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1074"><a href="#cb19-1074" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1075"><a href="#cb19-1075" aria-hidden="true" tabindex="-1"></a>Eventually, we store these images in two distinct directories named <span class="in">`train`</span> and <span class="in">`valid`</span>.</span>
<span id="cb19-1076"><a href="#cb19-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1077"><a href="#cb19-1077" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo=TRUE}</span></span>
<span id="cb19-1078"><a href="#cb19-1078" aria-hidden="true" tabindex="-1"></a><span class="co"># create dir train/ and copy pix there, organised by categories</span></span>
<span id="cb19-1079"><a href="#cb19-1079" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">'pix/train'</span>) <span class="co"># create training directory</span></span>
<span id="cb19-1080"><a href="#cb19-1080" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">levels</span>(<span class="fu">fct_drop</span>(pix_train<span class="sc">$</span>Keywords))) <span class="fu">dir.create</span>(<span class="fu">paste0</span>(<span class="st">'pix/train/'</span>,i)) <span class="co"># create dir for labels</span></span>
<span id="cb19-1081"><a href="#cb19-1081" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pix_train)){</span>
<span id="cb19-1082"><a href="#cb19-1082" aria-hidden="true" tabindex="-1"></a>    <span class="fu">file.copy</span>(<span class="fu">as.character</span>(pix_train<span class="sc">$</span>SourceFile[i]),</span>
<span id="cb19-1083"><a href="#cb19-1083" aria-hidden="true" tabindex="-1"></a>              <span class="fu">paste0</span>(<span class="st">'pix/train/'</span>, <span class="fu">as.character</span>(pix_train<span class="sc">$</span>Keywords[i]))) <span class="co"># copy pix in corresp dir</span></span>
<span id="cb19-1084"><a href="#cb19-1084" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-1085"><a href="#cb19-1085" aria-hidden="true" tabindex="-1"></a><span class="co"># create dir valid/ and copy pix there, organised by categories.</span></span>
<span id="cb19-1086"><a href="#cb19-1086" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">'pix/valid'</span>) <span class="co"># create validation dir</span></span>
<span id="cb19-1087"><a href="#cb19-1087" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">levels</span>(<span class="fu">fct_drop</span>(pix_train<span class="sc">$</span>Keywords))) <span class="fu">dir.create</span>(<span class="fu">paste0</span>(<span class="st">'pix/valid/'</span>,i)) <span class="co"># create dir for labels</span></span>
<span id="cb19-1088"><a href="#cb19-1088" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pix_valid)){</span>
<span id="cb19-1089"><a href="#cb19-1089" aria-hidden="true" tabindex="-1"></a>    <span class="fu">file.copy</span>(<span class="fu">as.character</span>(pix_valid<span class="sc">$</span>SourceFile[i]),</span>
<span id="cb19-1090"><a href="#cb19-1090" aria-hidden="true" tabindex="-1"></a>              <span class="fu">paste0</span>(<span class="st">'pix/valid/'</span>, <span class="fu">as.character</span>(pix_valid<span class="sc">$</span>Keywords[i]))) <span class="co"># copy pix in corresp dir</span></span>
<span id="cb19-1091"><a href="#cb19-1091" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-1092"><a href="#cb19-1092" aria-hidden="true" tabindex="-1"></a><span class="co"># delete pictures in valid/ directory for which we did not train the model</span></span>
<span id="cb19-1093"><a href="#cb19-1093" aria-hidden="true" tabindex="-1"></a>to_be_deleted <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">levels</span>(<span class="fu">fct_drop</span>(pix_valid<span class="sc">$</span>Keywords)), <span class="fu">levels</span>(<span class="fu">fct_drop</span>(pix_train<span class="sc">$</span>Keywords)))</span>
<span id="cb19-1094"><a href="#cb19-1094" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">is_empty</span>(to_be_deleted)) {</span>
<span id="cb19-1095"><a href="#cb19-1095" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(to_be_deleted)){</span>
<span id="cb19-1096"><a href="#cb19-1096" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unlink</span>(<span class="fu">paste0</span>(<span class="st">'pix/valid/'</span>, to_be_deleted[i]))</span>
<span id="cb19-1097"><a href="#cb19-1097" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb19-1098"><a href="#cb19-1098" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-1099"><a href="#cb19-1099" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1100"><a href="#cb19-1100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1101"><a href="#cb19-1101" aria-hidden="true" tabindex="-1"></a>What is the sample size of these two datasets?</span>
<span id="cb19-1102"><a href="#cb19-1102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1103"><a href="#cb19-1103" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo=TRUE}</span></span>
<span id="cb19-1104"><a href="#cb19-1104" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-samplesize</span></span>
<span id="cb19-1105"><a href="#cb19-1105" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Sample size (n) for the training and validation datasets."</span></span>
<span id="cb19-1106"><a href="#cb19-1106" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(<span class="st">"training"</span> <span class="ot">=</span> pix_train, <span class="st">"validation"</span> <span class="ot">=</span> pix_valid, <span class="at">.id =</span> <span class="st">"dataset"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1107"><a href="#cb19-1107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb19-1108"><a href="#cb19-1108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb19-1109"><a href="#cb19-1109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">category =</span> Keywords) <span class="sc">%&gt;%</span></span>
<span id="cb19-1110"><a href="#cb19-1110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-1111"><a href="#cb19-1111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span>
<span id="cb19-1112"><a href="#cb19-1112" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1113"><a href="#cb19-1113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1114"><a href="#cb19-1114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1115"><a href="#cb19-1115" aria-hidden="true" tabindex="-1"></a><span class="fu">## Transfer learning</span></span>
<span id="cb19-1116"><a href="#cb19-1116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1117"><a href="#cb19-1117" aria-hidden="true" tabindex="-1"></a>We proceed with transfer learning using images from the Jura county (or a subsample more exactly). We first load images and apply standard transformations to improve training (flip, rotate, zoom, rotate, ligth transform).</span>
<span id="cb19-1118"><a href="#cb19-1118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1119"><a href="#cb19-1119" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo = TRUE}</span></span>
<span id="cb19-1120"><a href="#cb19-1120" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb19-1121"><a href="#cb19-1121" aria-hidden="true" tabindex="-1"></a><span class="co">#reticulate::use_condaenv("computo")</span></span>
<span id="cb19-1122"><a href="#cb19-1122" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fastai)</span>
<span id="cb19-1123"><a href="#cb19-1123" aria-hidden="true" tabindex="-1"></a>dls <span class="ot">&lt;-</span> <span class="fu">ImageDataLoaders_from_folder</span>(</span>
<span id="cb19-1124"><a href="#cb19-1124" aria-hidden="true" tabindex="-1"></a>  <span class="at">path =</span> <span class="st">"pix/"</span>,</span>
<span id="cb19-1125"><a href="#cb19-1125" aria-hidden="true" tabindex="-1"></a>  <span class="at">train =</span> <span class="st">"train"</span>,</span>
<span id="cb19-1126"><a href="#cb19-1126" aria-hidden="true" tabindex="-1"></a>  <span class="at">valid =</span> <span class="st">"valid"</span>,</span>
<span id="cb19-1127"><a href="#cb19-1127" aria-hidden="true" tabindex="-1"></a>  <span class="at">item_tfms =</span> <span class="fu">Resize</span>(<span class="at">size =</span> <span class="dv">460</span>),</span>
<span id="cb19-1128"><a href="#cb19-1128" aria-hidden="true" tabindex="-1"></a>  <span class="at">bs =</span> <span class="dv">10</span>,</span>
<span id="cb19-1129"><a href="#cb19-1129" aria-hidden="true" tabindex="-1"></a>  <span class="at">batch_tfms =</span> <span class="fu">list</span>(<span class="fu">aug_transforms</span>(<span class="at">size =</span> <span class="dv">224</span>,</span>
<span id="cb19-1130"><a href="#cb19-1130" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">min_scale =</span> <span class="fl">0.75</span>), <span class="co"># transformation</span></span>
<span id="cb19-1131"><a href="#cb19-1131" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">Normalize_from_stats</span>( <span class="fu">imagenet_stats</span>() )),</span>
<span id="cb19-1132"><a href="#cb19-1132" aria-hidden="true" tabindex="-1"></a>  <span class="at">num_workers =</span> <span class="dv">0</span>,</span>
<span id="cb19-1133"><a href="#cb19-1133" aria-hidden="true" tabindex="-1"></a>  <span class="at">ImageFile.LOAD_TRUNCATED_IMAGES =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-1134"><a href="#cb19-1134" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1135"><a href="#cb19-1135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1136"><a href="#cb19-1136" aria-hidden="true" tabindex="-1"></a>Then we get the model architecture. For the sake of illustration, we use a resnet18 here, but we used a resnet50 to get the full results presented in the main text.</span>
<span id="cb19-1137"><a href="#cb19-1137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1138"><a href="#cb19-1138" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo = TRUE}</span></span>
<span id="cb19-1139"><a href="#cb19-1139" aria-hidden="true" tabindex="-1"></a>learn <span class="ot">&lt;-</span> <span class="fu">cnn_learner</span>(<span class="at">dls =</span> dls,</span>
<span id="cb19-1140"><a href="#cb19-1140" aria-hidden="true" tabindex="-1"></a>                     <span class="at">arch =</span> <span class="fu">resnet18</span>(),</span>
<span id="cb19-1141"><a href="#cb19-1141" aria-hidden="true" tabindex="-1"></a>                     <span class="at">metrics =</span> <span class="fu">list</span>(accuracy, error_rate))</span>
<span id="cb19-1142"><a href="#cb19-1142" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1143"><a href="#cb19-1143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1144"><a href="#cb19-1144" aria-hidden="true" tabindex="-1"></a>Now we are ready to train our model. Again, for the sake of illustration, we use only 2 epochs here, but used 20 epochs to get the full results presented in the main text. With all pictures and a resnet50, it took 75 minutes per epoch approximatively on a Mac with a 2.4Ghz processor and 64Go memory, and less than half an hour on a machine with GPU. On this reduced dataset, it took a bit more than a minute per epoch on the same Mac. Note that we save the model after each epoch for later use.</span>
<span id="cb19-1145"><a href="#cb19-1145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1146"><a href="#cb19-1146" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, echo = TRUE, cache = FALSE}</span></span>
<span id="cb19-1147"><a href="#cb19-1147" aria-hidden="true" tabindex="-1"></a>one_cycle <span class="ot">&lt;-</span> learn <span class="sc">%&gt;%</span></span>
<span id="cb19-1148"><a href="#cb19-1148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fit_one_cycle</span>(<span class="dv">2</span>, <span class="at">cbs =</span> <span class="fu">SaveModelCallback</span>(<span class="at">every_epoch =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-1149"><a href="#cb19-1149" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">fname =</span> <span class="st">'model'</span>))</span>
<span id="cb19-1150"><a href="#cb19-1150" aria-hidden="true" tabindex="-1"></a>one_cycle</span>
<span id="cb19-1151"><a href="#cb19-1151" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1152"><a href="#cb19-1152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1153"><a href="#cb19-1153" aria-hidden="true" tabindex="-1"></a>We may dig a bit deeper in training performances by loading the best model, here <span class="in">`model_1.pth`</span>, and display some metrics for each species.</span>
<span id="cb19-1154"><a href="#cb19-1154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1155"><a href="#cb19-1155" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo = TRUE}</span></span>
<span id="cb19-1156"><a href="#cb19-1156" aria-hidden="true" tabindex="-1"></a>learn<span class="sc">$</span><span class="fu">load</span>(<span class="st">"model_1"</span>)</span>
<span id="cb19-1157"><a href="#cb19-1157" aria-hidden="true" tabindex="-1"></a>interp <span class="ot">&lt;-</span> <span class="fu">ClassificationInterpretation_from_learner</span>(learn)</span>
<span id="cb19-1158"><a href="#cb19-1158" aria-hidden="true" tabindex="-1"></a>interp<span class="sc">$</span><span class="fu">print_classification_report</span>()</span>
<span id="cb19-1159"><a href="#cb19-1159" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1160"><a href="#cb19-1160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1161"><a href="#cb19-1161" aria-hidden="true" tabindex="-1"></a>We may extract the categories that get the most confused.</span>
<span id="cb19-1162"><a href="#cb19-1162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1163"><a href="#cb19-1163" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo = TRUE}</span></span>
<span id="cb19-1164"><a href="#cb19-1164" aria-hidden="true" tabindex="-1"></a>interp <span class="sc">%&gt;%</span> <span class="fu">most_confused</span>()</span>
<span id="cb19-1165"><a href="#cb19-1165" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1166"><a href="#cb19-1166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1167"><a href="#cb19-1167" aria-hidden="true" tabindex="-1"></a><span class="fu">## Transferability</span></span>
<span id="cb19-1168"><a href="#cb19-1168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1169"><a href="#cb19-1169" aria-hidden="true" tabindex="-1"></a>In this section, we show how to use our freshly trained model to label images that were taken in another study site in the Ain county, and not used to train our model. First, we get the path to the images.</span>
<span id="cb19-1170"><a href="#cb19-1170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1171"><a href="#cb19-1171" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo = TRUE}</span></span>
<span id="cb19-1172"><a href="#cb19-1172" aria-hidden="true" tabindex="-1"></a>fls <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">"pix/pixAin"</span>,</span>
<span id="cb19-1173"><a href="#cb19-1173" aria-hidden="true" tabindex="-1"></a>                  <span class="at">full.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-1174"><a href="#cb19-1174" aria-hidden="true" tabindex="-1"></a>                  <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-1175"><a href="#cb19-1175" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1176"><a href="#cb19-1176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1177"><a href="#cb19-1177" aria-hidden="true" tabindex="-1"></a>Then we carry out prediction, and compare to the truth.</span>
<span id="cb19-1178"><a href="#cb19-1178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1179"><a href="#cb19-1179" aria-hidden="true" tabindex="-1"></a><span class="in">```{r echo = TRUE}</span></span>
<span id="cb19-1180"><a href="#cb19-1180" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-prediction</span></span>
<span id="cb19-1181"><a href="#cb19-1181" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Comparison of the predictions vs. ground truth."</span></span>
<span id="cb19-1182"><a href="#cb19-1182" aria-hidden="true" tabindex="-1"></a>predicted <span class="ot">&lt;-</span> <span class="fu">character</span>(<span class="dv">3</span>)</span>
<span id="cb19-1183"><a href="#cb19-1183" aria-hidden="true" tabindex="-1"></a>categories <span class="ot">&lt;-</span> interp<span class="sc">$</span>vocab <span class="sc">%&gt;%</span> <span class="fu">as.character</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb19-1184"><a href="#cb19-1184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace_all</span>(<span class="st">"[[:punct:]]"</span>, <span class="st">" "</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1185"><a href="#cb19-1185" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_trim</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-1186"><a href="#cb19-1186" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_split</span>(<span class="st">"   "</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1187"><a href="#cb19-1187" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>()</span>
<span id="cb19-1188"><a href="#cb19-1188" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fls)){</span>
<span id="cb19-1189"><a href="#cb19-1189" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> learn <span class="sc">%&gt;%</span> <span class="fu">predict</span>(fls[i]) <span class="co"># make prediction</span></span>
<span id="cb19-1190"><a href="#cb19-1190" aria-hidden="true" tabindex="-1"></a>  result[[<span class="dv">3</span>]] <span class="sc">%&gt;%</span> <span class="fu">as.character</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb19-1191"><a href="#cb19-1191" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_extract</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1192"><a href="#cb19-1192" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.integer</span>() <span class="ot">-&gt;</span> index <span class="co"># extract relevant info</span></span>
<span id="cb19-1193"><a href="#cb19-1193" aria-hidden="true" tabindex="-1"></a>  predicted[i] <span class="ot">&lt;-</span> categories[index <span class="sc">+</span> <span class="dv">1</span>] <span class="co"># match it with categories</span></span>
<span id="cb19-1194"><a href="#cb19-1194" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-1195"><a href="#cb19-1195" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">truth =</span> <span class="fu">c</span>(<span class="st">"lynx"</span>, <span class="st">"roe deer"</span>, <span class="st">"wild boar"</span>),</span>
<span id="cb19-1196"><a href="#cb19-1196" aria-hidden="true" tabindex="-1"></a>           <span class="at">prediction =</span> predicted) <span class="sc">%&gt;%</span></span>
<span id="cb19-1197"><a href="#cb19-1197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-1198"><a href="#cb19-1198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span>
<span id="cb19-1199"><a href="#cb19-1199" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1200"><a href="#cb19-1200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1201"><a href="#cb19-1201" aria-hidden="true" tabindex="-1"></a><span class="fu"># Session information {.appendix .unnumbered}</span></span>
<span id="cb19-1202"><a href="#cb19-1202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1203"><a href="#cb19-1203" aria-hidden="true" tabindex="-1"></a><span class="in">```{r session-info}</span></span>
<span id="cb19-1204"><a href="#cb19-1204" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb19-1205"><a href="#cb19-1205" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1206"><a href="#cb19-1206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1207"><a href="#cb19-1207" aria-hidden="true" tabindex="-1"></a><span class="fu"># Acknowledgments {.appendix .unnumbered}</span></span>
<span id="cb19-1208"><a href="#cb19-1208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1209"><a href="#cb19-1209" aria-hidden="true" tabindex="-1"></a>We warmly thank Mathieu Massaviol, Remy Dernat and Khalid Belkhir for their help in using GPU machines on the Montpellier Bioinformatics Biodiversity platform, Julien Renoult for helpful discussions, Delphine Dinouart and Chloé Quillard for their precious help in manually tagging the images, and Vincent Miele for having inspired this work, and his help and support along the way. We also thank the staff of the Federations of Hunters from the Jura and Ain counties, hunters who helped to find locations for camera traps and volunteers who contributed in collecting data. Our thanks also go to Hannah Clipp, Chris Rota and Ken Kellner for sharing a development version of unmarked, and an unpublished version of their paper. The Lynx Predator Prey Program was funded by Auvergne-Rhône-Alpes Region, Ain and Jura departmental Councils, The French National Federation of Hunters, French Environmental Ministry based in Auvergne-Rhone-Alpes and Bourgogne Franche-Comté Region and the French Office for Biodiversity. Our work was also partly funded by the French National Research Agency (grant ANR-16-CE02-0007). </span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>